// @storybook/core/dist/core-events/index.js
// src/core-events/index.ts
var R = /* @__PURE__ */ ((E) => (E.CHANNEL_WS_DISCONNECT = "channelWSDisconnect", E.CHANNEL_CREATED = "channelCreated", E.CONFIG_ERROR = "co\
nfigError", E.STORY_INDEX_INVALIDATED = "storyIndexInvalidated", E.STORY_SPECIFIED = "storySpecified", E.SET_CONFIG = "setConfig", E.SET_STORIES =
"setStories", E.SET_INDEX = "setIndex", E.SET_CURRENT_STORY = "setCurrentStory", E.CURRENT_STORY_WAS_SET = "currentStoryWasSet", E.FORCE_RE_RENDER =
"forceReRender", E.FORCE_REMOUNT = "forceRemount", E.PRELOAD_ENTRIES = "preloadStories", E.STORY_PREPARED = "storyPrepared", E.DOCS_PREPARED =
"docsPrepared", E.STORY_CHANGED = "storyChanged", E.STORY_UNCHANGED = "storyUnchanged", E.STORY_RENDERED = "storyRendered", E.STORY_MISSING =
"storyMissing", E.STORY_ERRORED = "storyErrored", E.STORY_THREW_EXCEPTION = "storyThrewException", E.STORY_RENDER_PHASE_CHANGED = "storyRend\
erPhaseChanged", E.PLAY_FUNCTION_THREW_EXCEPTION = "playFunctionThrewException", E.UNHANDLED_ERRORS_WHILE_PLAYING = "unhandledErrorsWhilePla\
ying", E.UPDATE_STORY_ARGS = "updateStoryArgs", E.STORY_ARGS_UPDATED = "storyArgsUpdated", E.RESET_STORY_ARGS = "resetStoryArgs", E.SET_FILTER =
"setFilter", E.SET_GLOBALS = "setGlobals", E.UPDATE_GLOBALS = "updateGlobals", E.GLOBALS_UPDATED = "globalsUpdated", E.REGISTER_SUBSCRIPTION =
"registerSubscription", E.PREVIEW_KEYDOWN = "previewKeydown", E.PREVIEW_BUILDER_PROGRESS = "preview_builder_progress", E.SELECT_STORY = "sel\
ectStory", E.STORIES_COLLAPSE_ALL = "storiesCollapseAll", E.STORIES_EXPAND_ALL = "storiesExpandAll", E.DOCS_RENDERED = "docsRendered", E.SHARED_STATE_CHANGED =
"sharedStateChanged", E.SHARED_STATE_SET = "sharedStateSet", E.NAVIGATE_URL = "navigateUrl", E.UPDATE_QUERY_PARAMS = "updateQueryParams", E.
REQUEST_WHATS_NEW_DATA = "requestWhatsNewData", E.RESULT_WHATS_NEW_DATA = "resultWhatsNewData", E.SET_WHATS_NEW_CACHE = "setWhatsNewCache", E.
TOGGLE_WHATS_NEW_NOTIFICATIONS = "toggleWhatsNewNotifications", E.TELEMETRY_ERROR = "telemetryError", E.FILE_COMPONENT_SEARCH_REQUEST = "fil\
eComponentSearchRequest", E.FILE_COMPONENT_SEARCH_RESPONSE = "fileComponentSearchResponse", E.SAVE_STORY_REQUEST = "saveStoryRequest", E.SAVE_STORY_RESPONSE =
"saveStoryResponse", E.ARGTYPES_INFO_REQUEST = "argtypesInfoRequest", E.ARGTYPES_INFO_RESPONSE = "argtypesInfoResponse", E.CREATE_NEW_STORYFILE_REQUEST =
"createNewStoryfileRequest", E.CREATE_NEW_STORYFILE_RESPONSE = "createNewStoryfileResponse", E))(R || {}), S = R, {
  CHANNEL_WS_DISCONNECT: _,
  CHANNEL_CREATED: T,
  CONFIG_ERROR: A,
  CREATE_NEW_STORYFILE_REQUEST: O,
  CREATE_NEW_STORYFILE_RESPONSE: N,
  CURRENT_STORY_WAS_SET: r,
  DOCS_PREPARED: D,
  DOCS_RENDERED: I,
  FILE_COMPONENT_SEARCH_REQUEST: C,
  FILE_COMPONENT_SEARCH_RESPONSE: o,
  FORCE_RE_RENDER: P,
  FORCE_REMOUNT: L,
  GLOBALS_UPDATED: a,
  NAVIGATE_URL: Y,
  PLAY_FUNCTION_THREW_EXCEPTION: U,
  UNHANDLED_ERRORS_WHILE_PLAYING: G,
  PRELOAD_ENTRIES: W,
  PREVIEW_BUILDER_PROGRESS: H,
  PREVIEW_KEYDOWN: d,
  REGISTER_SUBSCRIPTION: t,
  RESET_STORY_ARGS: p,
  SELECT_STORY: i,
  SET_CONFIG: l,
  SET_CURRENT_STORY: F,
  SET_FILTER: e,
  SET_GLOBALS: y,
  SET_INDEX: c,
  SET_STORIES: h,
  SHARED_STATE_CHANGED: f,
  SHARED_STATE_SET: g,
  STORIES_COLLAPSE_ALL: u,
  STORIES_EXPAND_ALL: s,
  STORY_ARGS_UPDATED: x,
  STORY_CHANGED: M,
  STORY_ERRORED: Q,
  STORY_INDEX_INVALIDATED: m,
  STORY_MISSING: V,
  STORY_PREPARED: w,
  STORY_RENDER_PHASE_CHANGED: B,
  STORY_RENDERED: X,
  STORY_SPECIFIED: b,
  STORY_THREW_EXCEPTION: q,
  STORY_UNCHANGED: K,
  UPDATE_GLOBALS: j,
  UPDATE_QUERY_PARAMS: k,
  UPDATE_STORY_ARGS: z,
  REQUEST_WHATS_NEW_DATA: J,
  RESULT_WHATS_NEW_DATA: Z,
  SET_WHATS_NEW_CACHE: $,
  TOGGLE_WHATS_NEW_NOTIFICATIONS: n,
  TELEMETRY_ERROR: v,
  SAVE_STORY_REQUEST: EE,
  SAVE_STORY_RESPONSE: RE,
  ARGTYPES_INFO_REQUEST: SE,
  ARGTYPES_INFO_RESPONSE: _E
} = R;
export {
  SE as ARGTYPES_INFO_REQUEST,
  _E as ARGTYPES_INFO_RESPONSE,
  T as CHANNEL_CREATED,
  _ as CHANNEL_WS_DISCONNECT,
  A as CONFIG_ERROR,
  O as CREATE_NEW_STORYFILE_REQUEST,
  N as CREATE_NEW_STORYFILE_RESPONSE,
  r as CURRENT_STORY_WAS_SET,
  D as DOCS_PREPARED,
  I as DOCS_RENDERED,
  C as FILE_COMPONENT_SEARCH_REQUEST,
  o as FILE_COMPONENT_SEARCH_RESPONSE,
  L as FORCE_REMOUNT,
  P as FORCE_RE_RENDER,
  a as GLOBALS_UPDATED,
  Y as NAVIGATE_URL,
  U as PLAY_FUNCTION_THREW_EXCEPTION,
  W as PRELOAD_ENTRIES,
  H as PREVIEW_BUILDER_PROGRESS,
  d as PREVIEW_KEYDOWN,
  t as REGISTER_SUBSCRIPTION,
  J as REQUEST_WHATS_NEW_DATA,
  p as RESET_STORY_ARGS,
  Z as RESULT_WHATS_NEW_DATA,
  EE as SAVE_STORY_REQUEST,
  RE as SAVE_STORY_RESPONSE,
  i as SELECT_STORY,
  l as SET_CONFIG,
  F as SET_CURRENT_STORY,
  e as SET_FILTER,
  y as SET_GLOBALS,
  c as SET_INDEX,
  h as SET_STORIES,
  $ as SET_WHATS_NEW_CACHE,
  f as SHARED_STATE_CHANGED,
  g as SHARED_STATE_SET,
  u as STORIES_COLLAPSE_ALL,
  s as STORIES_EXPAND_ALL,
  x as STORY_ARGS_UPDATED,
  M as STORY_CHANGED,
  Q as STORY_ERRORED,
  m as STORY_INDEX_INVALIDATED,
  V as STORY_MISSING,
  w as STORY_PREPARED,
  X as STORY_RENDERED,
  B as STORY_RENDER_PHASE_CHANGED,
  b as STORY_SPECIFIED,
  q as STORY_THREW_EXCEPTION,
  K as STORY_UNCHANGED,
  v as TELEMETRY_ERROR,
  n as TOGGLE_WHATS_NEW_NOTIFICATIONS,
  G as UNHANDLED_ERRORS_WHILE_PLAYING,
  j as UPDATE_GLOBALS,
  k as UPDATE_QUERY_PARAMS,
  z as UPDATE_STORY_ARGS,
  S as default
};
