// @storybook/core/dist/docs-tools/index.js
var ii = Object.create;
var he = Object.defineProperty;
var si = Object.getOwnPropertyDescriptor;
var ui = Object.getOwnPropertyNames;
var pi = Object.getPrototypeOf, ci = Object.prototype.hasOwnProperty;
var t = (e, r) => he(e, "name", { value: r, configurable: !0 });
var i = (e, r) => () => (r || e((r = { exports: {} }).exports, r), r.exports);
var li = (e, r, n, o) => {
  if (r && typeof r == "object" || typeof r == "function")
    for (let a of ui(r))
      !ci.call(e, a) && a !== n && he(e, a, { get: () => r[a], enumerable: !(o = si(r, a)) || o.enumerable });
  return e;
};
var fi = (e, r, n) => (n = e != null ? ii(pi(e)) : {}, li(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  r || !e || !e.__esModule ? he(n, "default", { value: e, enumerable: !0 }) : n,
  e
));

// ../node_modules/lodash/_freeGlobal.js
var Te = i((wg, rr) => {
  var hi = typeof global == "object" && global && global.Object === Object && global;
  rr.exports = hi;
});

// ../node_modules/lodash/_root.js
var D = i((Ag, tr) => {
  var Ti = Te(), xi = typeof self == "object" && self && self.Object === Object && self, bi = Ti || xi || Function("return this")();
  tr.exports = bi;
});

// ../node_modules/lodash/_Symbol.js
var B = i((Eg, nr) => {
  var vi = D(), Di = vi.Symbol;
  nr.exports = Di;
});

// ../node_modules/lodash/_getRawTag.js
var sr = i((jg, ir) => {
  var or = B(), ar = Object.prototype, Pi = ar.hasOwnProperty, qi = ar.toString, z = or ? or.toStringTag : void 0;
  function Si(e) {
    var r = Pi.call(e, z), n = e[z];
    try {
      e[z] = void 0;
      var o = !0;
    } catch {
    }
    var a = qi.call(e);
    return o && (r ? e[z] = n : delete e[z]), a;
  }
  t(Si, "getRawTag");
  ir.exports = Si;
});

// ../node_modules/lodash/_objectToString.js
var pr = i((Ig, ur) => {
  var _i = Object.prototype, Oi = _i.toString;
  function wi(e) {
    return Oi.call(e);
  }
  t(wi, "objectToString");
  ur.exports = wi;
});

// ../node_modules/lodash/_baseGetTag.js
var j = i((Ng, fr) => {
  var cr = B(), Ai = sr(), Ei = pr(), ji = "[object Null]", Ci = "[object Undefined]", lr = cr ? cr.toStringTag : void 0;
  function Ii(e) {
    return e == null ? e === void 0 ? Ci : ji : lr && lr in Object(e) ? Ai(e) : Ei(e);
  }
  t(Ii, "baseGetTag");
  fr.exports = Ii;
});

// ../node_modules/lodash/isObject.js
var re = i((Lg, mr) => {
  function Fi(e) {
    var r = typeof e;
    return e != null && (r == "object" || r == "function");
  }
  t(Fi, "isObject");
  mr.exports = Fi;
});

// ../node_modules/lodash/isFunction.js
var xe = i((Jg, gr) => {
  var Ni = j(), Ri = re(), Li = "[object AsyncFunction]", Mi = "[object Function]", Ji = "[object GeneratorFunction]", Ui = "[object Proxy]";
  function Gi(e) {
    if (!Ri(e))
      return !1;
    var r = Ni(e);
    return r == Mi || r == Ji || r == Li || r == Ui;
  }
  t(Gi, "isFunction");
  gr.exports = Gi;
});

// ../node_modules/lodash/_coreJsData.js
var dr = i((Gg, yr) => {
  var Vi = D(), Bi = Vi["__core-js_shared__"];
  yr.exports = Bi;
});

// ../node_modules/lodash/_isMasked.js
var xr = i((Vg, Tr) => {
  var be = dr(), hr = function() {
    var e = /[^.]+$/.exec(be && be.keys && be.keys.IE_PROTO || "");
    return e ? "Symbol(src)_1." + e : "";
  }();
  function zi(e) {
    return !!hr && hr in e;
  }
  t(zi, "isMasked");
  Tr.exports = zi;
});

// ../node_modules/lodash/_toSource.js
var ve = i((zg, br) => {
  var ki = Function.prototype, Hi = ki.toString;
  function Ki(e) {
    if (e != null) {
      try {
        return Hi.call(e);
      } catch {
      }
      try {
        return e + "";
      } catch {
      }
    }
    return "";
  }
  t(Ki, "toSource");
  br.exports = Ki;
});

// ../node_modules/lodash/_baseIsNative.js
var Dr = i((Hg, vr) => {
  var $i = xe(), Wi = xr(), Yi = re(), Xi = ve(), Qi = /[\\^$.*+?()[\]{}|]/g, Zi = /^\[object .+?Constructor\]$/, es = Function.prototype, rs = Object.
  prototype, ts = es.toString, ns = rs.hasOwnProperty, os = RegExp(
    "^" + ts.call(ns).replace(Qi, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
  );
  function as(e) {
    if (!Yi(e) || Wi(e))
      return !1;
    var r = $i(e) ? os : Zi;
    return r.test(Xi(e));
  }
  t(as, "baseIsNative");
  vr.exports = as;
});

// ../node_modules/lodash/_getValue.js
var qr = i(($g, Pr) => {
  function is(e, r) {
    return e?.[r];
  }
  t(is, "getValue");
  Pr.exports = is;
});

// ../node_modules/lodash/_getNative.js
var S = i((Yg, Sr) => {
  var ss = Dr(), us = qr();
  function ps(e, r) {
    var n = us(e, r);
    return ss(n) ? n : void 0;
  }
  t(ps, "getNative");
  Sr.exports = ps;
});

// ../node_modules/lodash/_defineProperty.js
var Or = i((Qg, _r) => {
  var cs = S(), ls = function() {
    try {
      var e = cs(Object, "defineProperty");
      return e({}, "", {}), e;
    } catch {
    }
  }();
  _r.exports = ls;
});

// ../node_modules/lodash/_baseAssignValue.js
var Er = i((Zg, Ar) => {
  var wr = Or();
  function fs(e, r, n) {
    r == "__proto__" && wr ? wr(e, r, {
      configurable: !0,
      enumerable: !0,
      value: n,
      writable: !0
    }) : e[r] = n;
  }
  t(fs, "baseAssignValue");
  Ar.exports = fs;
});

// ../node_modules/lodash/_createBaseFor.js
var Cr = i((ry, jr) => {
  function ms(e) {
    return function(r, n, o) {
      for (var a = -1, s = Object(r), u = o(r), p = u.length; p--; ) {
        var c = u[e ? p : ++a];
        if (n(s[c], c, s) === !1)
          break;
      }
      return r;
    };
  }
  t(ms, "createBaseFor");
  jr.exports = ms;
});

// ../node_modules/lodash/_baseFor.js
var Fr = i((ny, Ir) => {
  var gs = Cr(), ys = gs();
  Ir.exports = ys;
});

// ../node_modules/lodash/_baseTimes.js
var Rr = i((oy, Nr) => {
  function ds(e, r) {
    for (var n = -1, o = Array(e); ++n < e; )
      o[n] = r(n);
    return o;
  }
  t(ds, "baseTimes");
  Nr.exports = ds;
});

// ../node_modules/lodash/isObjectLike.js
var C = i((iy, Lr) => {
  function hs(e) {
    return e != null && typeof e == "object";
  }
  t(hs, "isObjectLike");
  Lr.exports = hs;
});

// ../node_modules/lodash/_baseIsArguments.js
var Jr = i((uy, Mr) => {
  var Ts = j(), xs = C(), bs = "[object Arguments]";
  function vs(e) {
    return xs(e) && Ts(e) == bs;
  }
  t(vs, "baseIsArguments");
  Mr.exports = vs;
});

// ../node_modules/lodash/isArguments.js
var De = i((cy, Vr) => {
  var Ur = Jr(), Ds = C(), Gr = Object.prototype, Ps = Gr.hasOwnProperty, qs = Gr.propertyIsEnumerable, Ss = Ur(/* @__PURE__ */ function() {
    return arguments;
  }()) ? Ur : function(e) {
    return Ds(e) && Ps.call(e, "callee") && !qs.call(e, "callee");
  };
  Vr.exports = Ss;
});

// ../node_modules/lodash/isArray.js
var P = i((ly, Br) => {
  var _s = Array.isArray;
  Br.exports = _s;
});

// ../node_modules/lodash/stubFalse.js
var kr = i((fy, zr) => {
  function Os() {
    return !1;
  }
  t(Os, "stubFalse");
  zr.exports = Os;
});

// ../node_modules/lodash/isBuffer.js
var Pe = i((k, I) => {
  var ws = D(), As = kr(), $r = typeof k == "object" && k && !k.nodeType && k, Hr = $r && typeof I == "object" && I && !I.nodeType && I, Es = Hr &&
  Hr.exports === $r, Kr = Es ? ws.Buffer : void 0, js = Kr ? Kr.isBuffer : void 0, Cs = js || As;
  I.exports = Cs;
});

// ../node_modules/lodash/_isIndex.js
var qe = i((gy, Wr) => {
  var Is = 9007199254740991, Fs = /^(?:0|[1-9]\d*)$/;
  function Ns(e, r) {
    var n = typeof e;
    return r = r ?? Is, !!r && (n == "number" || n != "symbol" && Fs.test(e)) && e > -1 && e % 1 == 0 && e < r;
  }
  t(Ns, "isIndex");
  Wr.exports = Ns;
});

// ../node_modules/lodash/isLength.js
var te = i((dy, Yr) => {
  var Rs = 9007199254740991;
  function Ls(e) {
    return typeof e == "number" && e > -1 && e % 1 == 0 && e <= Rs;
  }
  t(Ls, "isLength");
  Yr.exports = Ls;
});

// ../node_modules/lodash/_baseIsTypedArray.js
var Qr = i((Ty, Xr) => {
  var Ms = j(), Js = te(), Us = C(), Gs = "[object Arguments]", Vs = "[object Array]", Bs = "[object Boolean]", zs = "[object Date]", ks = "\
[object Error]", Hs = "[object Function]", Ks = "[object Map]", $s = "[object Number]", Ws = "[object Object]", Ys = "[object RegExp]", Xs = "\
[object Set]", Qs = "[object String]", Zs = "[object WeakMap]", eu = "[object ArrayBuffer]", ru = "[object DataView]", tu = "[object Float32\
Array]", nu = "[object Float64Array]", ou = "[object Int8Array]", au = "[object Int16Array]", iu = "[object Int32Array]", su = "[object Uint\
8Array]", uu = "[object Uint8ClampedArray]", pu = "[object Uint16Array]", cu = "[object Uint32Array]", m = {};
  m[tu] = m[nu] = m[ou] = m[au] = m[iu] = m[su] = m[uu] = m[pu] = m[cu] = !0;
  m[Gs] = m[Vs] = m[eu] = m[Bs] = m[ru] = m[zs] = m[ks] = m[Hs] = m[Ks] = m[$s] = m[Ws] = m[Ys] = m[Xs] = m[Qs] = m[Zs] = !1;
  function lu(e) {
    return Us(e) && Js(e.length) && !!m[Ms(e)];
  }
  t(lu, "baseIsTypedArray");
  Xr.exports = lu;
});

// ../node_modules/lodash/_baseUnary.js
var et = i((by, Zr) => {
  function fu(e) {
    return function(r) {
      return e(r);
    };
  }
  t(fu, "baseUnary");
  Zr.exports = fu;
});

// ../node_modules/lodash/_nodeUtil.js
var tt = i((H, F) => {
  var mu = Te(), rt = typeof H == "object" && H && !H.nodeType && H, K = rt && typeof F == "object" && F && !F.nodeType && F, gu = K && K.exports ===
  rt, Se = gu && mu.process, yu = function() {
    try {
      var e = K && K.require && K.require("util").types;
      return e || Se && Se.binding && Se.binding("util");
    } catch {
    }
  }();
  F.exports = yu;
});

// ../node_modules/lodash/isTypedArray.js
var _e = i((Dy, at) => {
  var du = Qr(), hu = et(), nt = tt(), ot = nt && nt.isTypedArray, Tu = ot ? hu(ot) : du;
  at.exports = Tu;
});

// ../node_modules/lodash/_arrayLikeKeys.js
var st = i((Py, it) => {
  var xu = Rr(), bu = De(), vu = P(), Du = Pe(), Pu = qe(), qu = _e(), Su = Object.prototype, _u = Su.hasOwnProperty;
  function Ou(e, r) {
    var n = vu(e), o = !n && bu(e), a = !n && !o && Du(e), s = !n && !o && !a && qu(e), u = n || o || a || s, p = u ? xu(e.length, String) :
    [], c = p.length;
    for (var l in e)
      (r || _u.call(e, l)) && !(u && // Safari 9 has enumerable `arguments.length` in strict mode.
      (l == "length" || // Node.js 0.10 has enumerable non-index properties on buffers.
      a && (l == "offset" || l == "parent") || // PhantomJS 2 has enumerable non-index properties on typed arrays.
      s && (l == "buffer" || l == "byteLength" || l == "byteOffset") || // Skip index properties.
      Pu(l, c))) && p.push(l);
    return p;
  }
  t(Ou, "arrayLikeKeys");
  it.exports = Ou;
});

// ../node_modules/lodash/_isPrototype.js
var pt = i((Sy, ut) => {
  var wu = Object.prototype;
  function Au(e) {
    var r = e && e.constructor, n = typeof r == "function" && r.prototype || wu;
    return e === n;
  }
  t(Au, "isPrototype");
  ut.exports = Au;
});

// ../node_modules/lodash/_overArg.js
var lt = i((Oy, ct) => {
  function Eu(e, r) {
    return function(n) {
      return e(r(n));
    };
  }
  t(Eu, "overArg");
  ct.exports = Eu;
});

// ../node_modules/lodash/_nativeKeys.js
var mt = i((Ay, ft) => {
  var ju = lt(), Cu = ju(Object.keys, Object);
  ft.exports = Cu;
});

// ../node_modules/lodash/_baseKeys.js
var yt = i((Ey, gt) => {
  var Iu = pt(), Fu = mt(), Nu = Object.prototype, Ru = Nu.hasOwnProperty;
  function Lu(e) {
    if (!Iu(e))
      return Fu(e);
    var r = [];
    for (var n in Object(e))
      Ru.call(e, n) && n != "constructor" && r.push(n);
    return r;
  }
  t(Lu, "baseKeys");
  gt.exports = Lu;
});

// ../node_modules/lodash/isArrayLike.js
var ht = i((Cy, dt) => {
  var Mu = xe(), Ju = te();
  function Uu(e) {
    return e != null && Ju(e.length) && !Mu(e);
  }
  t(Uu, "isArrayLike");
  dt.exports = Uu;
});

// ../node_modules/lodash/keys.js
var ne = i((Fy, Tt) => {
  var Gu = st(), Vu = yt(), Bu = ht();
  function zu(e) {
    return Bu(e) ? Gu(e) : Vu(e);
  }
  t(zu, "keys");
  Tt.exports = zu;
});

// ../node_modules/lodash/_baseForOwn.js
var bt = i((Ry, xt) => {
  var ku = Fr(), Hu = ne();
  function Ku(e, r) {
    return e && ku(e, r, Hu);
  }
  t(Ku, "baseForOwn");
  xt.exports = Ku;
});

// ../node_modules/lodash/_listCacheClear.js
var Dt = i((My, vt) => {
  function $u() {
    this.__data__ = [], this.size = 0;
  }
  t($u, "listCacheClear");
  vt.exports = $u;
});

// ../node_modules/lodash/eq.js
var Oe = i((Uy, Pt) => {
  function Wu(e, r) {
    return e === r || e !== e && r !== r;
  }
  t(Wu, "eq");
  Pt.exports = Wu;
});

// ../node_modules/lodash/_assocIndexOf.js
var $ = i((Vy, qt) => {
  var Yu = Oe();
  function Xu(e, r) {
    for (var n = e.length; n--; )
      if (Yu(e[n][0], r))
        return n;
    return -1;
  }
  t(Xu, "assocIndexOf");
  qt.exports = Xu;
});

// ../node_modules/lodash/_listCacheDelete.js
var _t = i((zy, St) => {
  var Qu = $(), Zu = Array.prototype, ep = Zu.splice;
  function rp(e) {
    var r = this.__data__, n = Qu(r, e);
    if (n < 0)
      return !1;
    var o = r.length - 1;
    return n == o ? r.pop() : ep.call(r, n, 1), --this.size, !0;
  }
  t(rp, "listCacheDelete");
  St.exports = rp;
});

// ../node_modules/lodash/_listCacheGet.js
var wt = i((Hy, Ot) => {
  var tp = $();
  function np(e) {
    var r = this.__data__, n = tp(r, e);
    return n < 0 ? void 0 : r[n][1];
  }
  t(np, "listCacheGet");
  Ot.exports = np;
});

// ../node_modules/lodash/_listCacheHas.js
var Et = i(($y, At) => {
  var op = $();
  function ap(e) {
    return op(this.__data__, e) > -1;
  }
  t(ap, "listCacheHas");
  At.exports = ap;
});

// ../node_modules/lodash/_listCacheSet.js
var Ct = i((Yy, jt) => {
  var ip = $();
  function sp(e, r) {
    var n = this.__data__, o = ip(n, e);
    return o < 0 ? (++this.size, n.push([e, r])) : n[o][1] = r, this;
  }
  t(sp, "listCacheSet");
  jt.exports = sp;
});

// ../node_modules/lodash/_ListCache.js
var W = i((Qy, It) => {
  var up = Dt(), pp = _t(), cp = wt(), lp = Et(), fp = Ct();
  function N(e) {
    var r = -1, n = e == null ? 0 : e.length;
    for (this.clear(); ++r < n; ) {
      var o = e[r];
      this.set(o[0], o[1]);
    }
  }
  t(N, "ListCache");
  N.prototype.clear = up;
  N.prototype.delete = pp;
  N.prototype.get = cp;
  N.prototype.has = lp;
  N.prototype.set = fp;
  It.exports = N;
});

// ../node_modules/lodash/_stackClear.js
var Nt = i((ed, Ft) => {
  var mp = W();
  function gp() {
    this.__data__ = new mp(), this.size = 0;
  }
  t(gp, "stackClear");
  Ft.exports = gp;
});

// ../node_modules/lodash/_stackDelete.js
var Lt = i((td, Rt) => {
  function yp(e) {
    var r = this.__data__, n = r.delete(e);
    return this.size = r.size, n;
  }
  t(yp, "stackDelete");
  Rt.exports = yp;
});

// ../node_modules/lodash/_stackGet.js
var Jt = i((od, Mt) => {
  function dp(e) {
    return this.__data__.get(e);
  }
  t(dp, "stackGet");
  Mt.exports = dp;
});

// ../node_modules/lodash/_stackHas.js
var Gt = i((id, Ut) => {
  function hp(e) {
    return this.__data__.has(e);
  }
  t(hp, "stackHas");
  Ut.exports = hp;
});

// ../node_modules/lodash/_Map.js
var oe = i((ud, Vt) => {
  var Tp = S(), xp = D(), bp = Tp(xp, "Map");
  Vt.exports = bp;
});

// ../node_modules/lodash/_nativeCreate.js
var Y = i((pd, Bt) => {
  var vp = S(), Dp = vp(Object, "create");
  Bt.exports = Dp;
});

// ../node_modules/lodash/_hashClear.js
var Ht = i((cd, kt) => {
  var zt = Y();
  function Pp() {
    this.__data__ = zt ? zt(null) : {}, this.size = 0;
  }
  t(Pp, "hashClear");
  kt.exports = Pp;
});

// ../node_modules/lodash/_hashDelete.js
var $t = i((fd, Kt) => {
  function qp(e) {
    var r = this.has(e) && delete this.__data__[e];
    return this.size -= r ? 1 : 0, r;
  }
  t(qp, "hashDelete");
  Kt.exports = qp;
});

// ../node_modules/lodash/_hashGet.js
var Yt = i((gd, Wt) => {
  var Sp = Y(), _p = "__lodash_hash_undefined__", Op = Object.prototype, wp = Op.hasOwnProperty;
  function Ap(e) {
    var r = this.__data__;
    if (Sp) {
      var n = r[e];
      return n === _p ? void 0 : n;
    }
    return wp.call(r, e) ? r[e] : void 0;
  }
  t(Ap, "hashGet");
  Wt.exports = Ap;
});

// ../node_modules/lodash/_hashHas.js
var Qt = i((dd, Xt) => {
  var Ep = Y(), jp = Object.prototype, Cp = jp.hasOwnProperty;
  function Ip(e) {
    var r = this.__data__;
    return Ep ? r[e] !== void 0 : Cp.call(r, e);
  }
  t(Ip, "hashHas");
  Xt.exports = Ip;
});

// ../node_modules/lodash/_hashSet.js
var en = i((Td, Zt) => {
  var Fp = Y(), Np = "__lodash_hash_undefined__";
  function Rp(e, r) {
    var n = this.__data__;
    return this.size += this.has(e) ? 0 : 1, n[e] = Fp && r === void 0 ? Np : r, this;
  }
  t(Rp, "hashSet");
  Zt.exports = Rp;
});

// ../node_modules/lodash/_Hash.js
var tn = i((bd, rn) => {
  var Lp = Ht(), Mp = $t(), Jp = Yt(), Up = Qt(), Gp = en();
  function R(e) {
    var r = -1, n = e == null ? 0 : e.length;
    for (this.clear(); ++r < n; ) {
      var o = e[r];
      this.set(o[0], o[1]);
    }
  }
  t(R, "Hash");
  R.prototype.clear = Lp;
  R.prototype.delete = Mp;
  R.prototype.get = Jp;
  R.prototype.has = Up;
  R.prototype.set = Gp;
  rn.exports = R;
});

// ../node_modules/lodash/_mapCacheClear.js
var an = i((Dd, on) => {
  var nn = tn(), Vp = W(), Bp = oe();
  function zp() {
    this.size = 0, this.__data__ = {
      hash: new nn(),
      map: new (Bp || Vp)(),
      string: new nn()
    };
  }
  t(zp, "mapCacheClear");
  on.exports = zp;
});

// ../node_modules/lodash/_isKeyable.js
var un = i((qd, sn) => {
  function kp(e) {
    var r = typeof e;
    return r == "string" || r == "number" || r == "symbol" || r == "boolean" ? e !== "__proto__" : e === null;
  }
  t(kp, "isKeyable");
  sn.exports = kp;
});

// ../node_modules/lodash/_getMapData.js
var X = i((_d, pn) => {
  var Hp = un();
  function Kp(e, r) {
    var n = e.__data__;
    return Hp(r) ? n[typeof r == "string" ? "string" : "hash"] : n.map;
  }
  t(Kp, "getMapData");
  pn.exports = Kp;
});

// ../node_modules/lodash/_mapCacheDelete.js
var ln = i((wd, cn) => {
  var $p = X();
  function Wp(e) {
    var r = $p(this, e).delete(e);
    return this.size -= r ? 1 : 0, r;
  }
  t(Wp, "mapCacheDelete");
  cn.exports = Wp;
});

// ../node_modules/lodash/_mapCacheGet.js
var mn = i((Ed, fn) => {
  var Yp = X();
  function Xp(e) {
    return Yp(this, e).get(e);
  }
  t(Xp, "mapCacheGet");
  fn.exports = Xp;
});

// ../node_modules/lodash/_mapCacheHas.js
var yn = i((Cd, gn) => {
  var Qp = X();
  function Zp(e) {
    return Qp(this, e).has(e);
  }
  t(Zp, "mapCacheHas");
  gn.exports = Zp;
});

// ../node_modules/lodash/_mapCacheSet.js
var hn = i((Fd, dn) => {
  var ec = X();
  function rc(e, r) {
    var n = ec(this, e), o = n.size;
    return n.set(e, r), this.size += n.size == o ? 0 : 1, this;
  }
  t(rc, "mapCacheSet");
  dn.exports = rc;
});

// ../node_modules/lodash/_MapCache.js
var ae = i((Rd, Tn) => {
  var tc = an(), nc = ln(), oc = mn(), ac = yn(), ic = hn();
  function L(e) {
    var r = -1, n = e == null ? 0 : e.length;
    for (this.clear(); ++r < n; ) {
      var o = e[r];
      this.set(o[0], o[1]);
    }
  }
  t(L, "MapCache");
  L.prototype.clear = tc;
  L.prototype.delete = nc;
  L.prototype.get = oc;
  L.prototype.has = ac;
  L.prototype.set = ic;
  Tn.exports = L;
});

// ../node_modules/lodash/_stackSet.js
var bn = i((Md, xn) => {
  var sc = W(), uc = oe(), pc = ae(), cc = 200;
  function lc(e, r) {
    var n = this.__data__;
    if (n instanceof sc) {
      var o = n.__data__;
      if (!uc || o.length < cc - 1)
        return o.push([e, r]), this.size = ++n.size, this;
      n = this.__data__ = new pc(o);
    }
    return n.set(e, r), this.size = n.size, this;
  }
  t(lc, "stackSet");
  xn.exports = lc;
});

// ../node_modules/lodash/_Stack.js
var we = i((Ud, vn) => {
  var fc = W(), mc = Nt(), gc = Lt(), yc = Jt(), dc = Gt(), hc = bn();
  function M(e) {
    var r = this.__data__ = new fc(e);
    this.size = r.size;
  }
  t(M, "Stack");
  M.prototype.clear = mc;
  M.prototype.delete = gc;
  M.prototype.get = yc;
  M.prototype.has = dc;
  M.prototype.set = hc;
  vn.exports = M;
});

// ../node_modules/lodash/_setCacheAdd.js
var Pn = i((Vd, Dn) => {
  var Tc = "__lodash_hash_undefined__";
  function xc(e) {
    return this.__data__.set(e, Tc), this;
  }
  t(xc, "setCacheAdd");
  Dn.exports = xc;
});

// ../node_modules/lodash/_setCacheHas.js
var Sn = i((zd, qn) => {
  function bc(e) {
    return this.__data__.has(e);
  }
  t(bc, "setCacheHas");
  qn.exports = bc;
});

// ../node_modules/lodash/_SetCache.js
var On = i((Hd, _n) => {
  var vc = ae(), Dc = Pn(), Pc = Sn();
  function ie(e) {
    var r = -1, n = e == null ? 0 : e.length;
    for (this.__data__ = new vc(); ++r < n; )
      this.add(e[r]);
  }
  t(ie, "SetCache");
  ie.prototype.add = ie.prototype.push = Dc;
  ie.prototype.has = Pc;
  _n.exports = ie;
});

// ../node_modules/lodash/_arraySome.js
var An = i(($d, wn) => {
  function qc(e, r) {
    for (var n = -1, o = e == null ? 0 : e.length; ++n < o; )
      if (r(e[n], n, e))
        return !0;
    return !1;
  }
  t(qc, "arraySome");
  wn.exports = qc;
});

// ../node_modules/lodash/_cacheHas.js
var jn = i((Yd, En) => {
  function Sc(e, r) {
    return e.has(r);
  }
  t(Sc, "cacheHas");
  En.exports = Sc;
});

// ../node_modules/lodash/_equalArrays.js
var Ae = i((Qd, Cn) => {
  var _c = On(), Oc = An(), wc = jn(), Ac = 1, Ec = 2;
  function jc(e, r, n, o, a, s) {
    var u = n & Ac, p = e.length, c = r.length;
    if (p != c && !(u && c > p))
      return !1;
    var l = s.get(e), f = s.get(r);
    if (l && f)
      return l == r && f == e;
    var d = -1, g = !0, x = n & Ec ? new _c() : void 0;
    for (s.set(e, r), s.set(r, e); ++d < p; ) {
      var h = e[d], T = r[d];
      if (o)
        var v = u ? o(T, h, d, r, e, s) : o(h, T, d, e, r, s);
      if (v !== void 0) {
        if (v)
          continue;
        g = !1;
        break;
      }
      if (x) {
        if (!Oc(r, function(O, w) {
          if (!wc(x, w) && (h === O || a(h, O, n, o, s)))
            return x.push(w);
        })) {
          g = !1;
          break;
        }
      } else if (!(h === T || a(h, T, n, o, s))) {
        g = !1;
        break;
      }
    }
    return s.delete(e), s.delete(r), g;
  }
  t(jc, "equalArrays");
  Cn.exports = jc;
});

// ../node_modules/lodash/_Uint8Array.js
var Fn = i((eh, In) => {
  var Cc = D(), Ic = Cc.Uint8Array;
  In.exports = Ic;
});

// ../node_modules/lodash/_mapToArray.js
var Rn = i((rh, Nn) => {
  function Fc(e) {
    var r = -1, n = Array(e.size);
    return e.forEach(function(o, a) {
      n[++r] = [a, o];
    }), n;
  }
  t(Fc, "mapToArray");
  Nn.exports = Fc;
});

// ../node_modules/lodash/_setToArray.js
var Mn = i((nh, Ln) => {
  function Nc(e) {
    var r = -1, n = Array(e.size);
    return e.forEach(function(o) {
      n[++r] = o;
    }), n;
  }
  t(Nc, "setToArray");
  Ln.exports = Nc;
});

// ../node_modules/lodash/_equalByTag.js
var Bn = i((ah, Vn) => {
  var Jn = B(), Un = Fn(), Rc = Oe(), Lc = Ae(), Mc = Rn(), Jc = Mn(), Uc = 1, Gc = 2, Vc = "[object Boolean]", Bc = "[object Date]", zc = "\
[object Error]", kc = "[object Map]", Hc = "[object Number]", Kc = "[object RegExp]", $c = "[object Set]", Wc = "[object String]", Yc = "[ob\
ject Symbol]", Xc = "[object ArrayBuffer]", Qc = "[object DataView]", Gn = Jn ? Jn.prototype : void 0, Ee = Gn ? Gn.valueOf : void 0;
  function Zc(e, r, n, o, a, s, u) {
    switch (n) {
      case Qc:
        if (e.byteLength != r.byteLength || e.byteOffset != r.byteOffset)
          return !1;
        e = e.buffer, r = r.buffer;
      case Xc:
        return !(e.byteLength != r.byteLength || !s(new Un(e), new Un(r)));
      case Vc:
      case Bc:
      case Hc:
        return Rc(+e, +r);
      case zc:
        return e.name == r.name && e.message == r.message;
      case Kc:
      case Wc:
        return e == r + "";
      case kc:
        var p = Mc;
      case $c:
        var c = o & Uc;
        if (p || (p = Jc), e.size != r.size && !c)
          return !1;
        var l = u.get(e);
        if (l)
          return l == r;
        o |= Gc, u.set(e, r);
        var f = Lc(p(e), p(r), o, a, s, u);
        return u.delete(e), f;
      case Yc:
        if (Ee)
          return Ee.call(e) == Ee.call(r);
    }
    return !1;
  }
  t(Zc, "equalByTag");
  Vn.exports = Zc;
});

// ../node_modules/lodash/_arrayPush.js
var kn = i((sh, zn) => {
  function el(e, r) {
    for (var n = -1, o = r.length, a = e.length; ++n < o; )
      e[a + n] = r[n];
    return e;
  }
  t(el, "arrayPush");
  zn.exports = el;
});

// ../node_modules/lodash/_baseGetAllKeys.js
var Kn = i((ph, Hn) => {
  var rl = kn(), tl = P();
  function nl(e, r, n) {
    var o = r(e);
    return tl(e) ? o : rl(o, n(e));
  }
  t(nl, "baseGetAllKeys");
  Hn.exports = nl;
});

// ../node_modules/lodash/_arrayFilter.js
var Wn = i((lh, $n) => {
  function ol(e, r) {
    for (var n = -1, o = e == null ? 0 : e.length, a = 0, s = []; ++n < o; ) {
      var u = e[n];
      r(u, n, e) && (s[a++] = u);
    }
    return s;
  }
  t(ol, "arrayFilter");
  $n.exports = ol;
});

// ../node_modules/lodash/stubArray.js
var Xn = i((mh, Yn) => {
  function al() {
    return [];
  }
  t(al, "stubArray");
  Yn.exports = al;
});

// ../node_modules/lodash/_getSymbols.js
var eo = i((yh, Zn) => {
  var il = Wn(), sl = Xn(), ul = Object.prototype, pl = ul.propertyIsEnumerable, Qn = Object.getOwnPropertySymbols, cl = Qn ? function(e) {
    return e == null ? [] : (e = Object(e), il(Qn(e), function(r) {
      return pl.call(e, r);
    }));
  } : sl;
  Zn.exports = cl;
});

// ../node_modules/lodash/_getAllKeys.js
var to = i((dh, ro) => {
  var ll = Kn(), fl = eo(), ml = ne();
  function gl(e) {
    return ll(e, ml, fl);
  }
  t(gl, "getAllKeys");
  ro.exports = gl;
});

// ../node_modules/lodash/_equalObjects.js
var ao = i((Th, oo) => {
  var no = to(), yl = 1, dl = Object.prototype, hl = dl.hasOwnProperty;
  function Tl(e, r, n, o, a, s) {
    var u = n & yl, p = no(e), c = p.length, l = no(r), f = l.length;
    if (c != f && !u)
      return !1;
    for (var d = c; d--; ) {
      var g = p[d];
      if (!(u ? g in r : hl.call(r, g)))
        return !1;
    }
    var x = s.get(e), h = s.get(r);
    if (x && h)
      return x == r && h == e;
    var T = !0;
    s.set(e, r), s.set(r, e);
    for (var v = u; ++d < c; ) {
      g = p[d];
      var O = e[g], w = r[g];
      if (o)
        var er = u ? o(w, O, g, r, e, s) : o(O, w, g, e, r, s);
      if (!(er === void 0 ? O === w || a(O, w, n, o, s) : er)) {
        T = !1;
        break;
      }
      v || (v = g == "constructor");
    }
    if (T && !v) {
      var Z = e.constructor, ee = r.constructor;
      Z != ee && "constructor" in e && "constructor" in r && !(typeof Z == "function" && Z instanceof Z && typeof ee == "function" && ee instanceof
      ee) && (T = !1);
    }
    return s.delete(e), s.delete(r), T;
  }
  t(Tl, "equalObjects");
  oo.exports = Tl;
});

// ../node_modules/lodash/_DataView.js
var so = i((bh, io) => {
  var xl = S(), bl = D(), vl = xl(bl, "DataView");
  io.exports = vl;
});

// ../node_modules/lodash/_Promise.js
var po = i((vh, uo) => {
  var Dl = S(), Pl = D(), ql = Dl(Pl, "Promise");
  uo.exports = ql;
});

// ../node_modules/lodash/_Set.js
var lo = i((Dh, co) => {
  var Sl = S(), _l = D(), Ol = Sl(_l, "Set");
  co.exports = Ol;
});

// ../node_modules/lodash/_WeakMap.js
var mo = i((Ph, fo) => {
  var wl = S(), Al = D(), El = wl(Al, "WeakMap");
  fo.exports = El;
});

// ../node_modules/lodash/_getTag.js
var Do = i((qh, vo) => {
  var je = so(), Ce = oe(), Ie = po(), Fe = lo(), Ne = mo(), bo = j(), J = ve(), go = "[object Map]", jl = "[object Object]", yo = "[object \
Promise]", ho = "[object Set]", To = "[object WeakMap]", xo = "[object DataView]", Cl = J(je), Il = J(Ce), Fl = J(Ie), Nl = J(Fe), Rl = J(Ne),
  A = bo;
  (je && A(new je(new ArrayBuffer(1))) != xo || Ce && A(new Ce()) != go || Ie && A(Ie.resolve()) != yo || Fe && A(new Fe()) != ho || Ne && A(
  new Ne()) != To) && (A = /* @__PURE__ */ t(function(e) {
    var r = bo(e), n = r == jl ? e.constructor : void 0, o = n ? J(n) : "";
    if (o)
      switch (o) {
        case Cl:
          return xo;
        case Il:
          return go;
        case Fl:
          return yo;
        case Nl:
          return ho;
        case Rl:
          return To;
      }
    return r;
  }, "getTag"));
  vo.exports = A;
});

// ../node_modules/lodash/_baseIsEqualDeep.js
var Eo = i((_h, Ao) => {
  var Re = we(), Ll = Ae(), Ml = Bn(), Jl = ao(), Po = Do(), qo = P(), So = Pe(), Ul = _e(), Gl = 1, _o = "[object Arguments]", Oo = "[objec\
t Array]", se = "[object Object]", Vl = Object.prototype, wo = Vl.hasOwnProperty;
  function Bl(e, r, n, o, a, s) {
    var u = qo(e), p = qo(r), c = u ? Oo : Po(e), l = p ? Oo : Po(r);
    c = c == _o ? se : c, l = l == _o ? se : l;
    var f = c == se, d = l == se, g = c == l;
    if (g && So(e)) {
      if (!So(r))
        return !1;
      u = !0, f = !1;
    }
    if (g && !f)
      return s || (s = new Re()), u || Ul(e) ? Ll(e, r, n, o, a, s) : Ml(e, r, c, n, o, a, s);
    if (!(n & Gl)) {
      var x = f && wo.call(e, "__wrapped__"), h = d && wo.call(r, "__wrapped__");
      if (x || h) {
        var T = x ? e.value() : e, v = h ? r.value() : r;
        return s || (s = new Re()), a(T, v, n, o, s);
      }
    }
    return g ? (s || (s = new Re()), Jl(e, r, n, o, a, s)) : !1;
  }
  t(Bl, "baseIsEqualDeep");
  Ao.exports = Bl;
});

// ../node_modules/lodash/_baseIsEqual.js
var Le = i((wh, Io) => {
  var zl = Eo(), jo = C();
  function Co(e, r, n, o, a) {
    return e === r ? !0 : e == null || r == null || !jo(e) && !jo(r) ? e !== e && r !== r : zl(e, r, n, o, Co, a);
  }
  t(Co, "baseIsEqual");
  Io.exports = Co;
});

// ../node_modules/lodash/_baseIsMatch.js
var No = i((Eh, Fo) => {
  var kl = we(), Hl = Le(), Kl = 1, $l = 2;
  function Wl(e, r, n, o) {
    var a = n.length, s = a, u = !o;
    if (e == null)
      return !s;
    for (e = Object(e); a--; ) {
      var p = n[a];
      if (u && p[2] ? p[1] !== e[p[0]] : !(p[0] in e))
        return !1;
    }
    for (; ++a < s; ) {
      p = n[a];
      var c = p[0], l = e[c], f = p[1];
      if (u && p[2]) {
        if (l === void 0 && !(c in e))
          return !1;
      } else {
        var d = new kl();
        if (o)
          var g = o(l, f, c, e, r, d);
        if (!(g === void 0 ? Hl(f, l, Kl | $l, o, d) : g))
          return !1;
      }
    }
    return !0;
  }
  t(Wl, "baseIsMatch");
  Fo.exports = Wl;
});

// ../node_modules/lodash/_isStrictComparable.js
var Me = i((Ch, Ro) => {
  var Yl = re();
  function Xl(e) {
    return e === e && !Yl(e);
  }
  t(Xl, "isStrictComparable");
  Ro.exports = Xl;
});

// ../node_modules/lodash/_getMatchData.js
var Mo = i((Fh, Lo) => {
  var Ql = Me(), Zl = ne();
  function ef(e) {
    for (var r = Zl(e), n = r.length; n--; ) {
      var o = r[n], a = e[o];
      r[n] = [o, a, Ql(a)];
    }
    return r;
  }
  t(ef, "getMatchData");
  Lo.exports = ef;
});

// ../node_modules/lodash/_matchesStrictComparable.js
var Je = i((Rh, Jo) => {
  function rf(e, r) {
    return function(n) {
      return n == null ? !1 : n[e] === r && (r !== void 0 || e in Object(n));
    };
  }
  t(rf, "matchesStrictComparable");
  Jo.exports = rf;
});

// ../node_modules/lodash/_baseMatches.js
var Go = i((Mh, Uo) => {
  var tf = No(), nf = Mo(), of = Je();
  function af(e) {
    var r = nf(e);
    return r.length == 1 && r[0][2] ? of(r[0][0], r[0][1]) : function(n) {
      return n === e || tf(n, e, r);
    };
  }
  t(af, "baseMatches");
  Uo.exports = af;
});

// ../node_modules/lodash/isSymbol.js
var ue = i((Uh, Vo) => {
  var sf = j(), uf = C(), pf = "[object Symbol]";
  function cf(e) {
    return typeof e == "symbol" || uf(e) && sf(e) == pf;
  }
  t(cf, "isSymbol");
  Vo.exports = cf;
});

// ../node_modules/lodash/_isKey.js
var pe = i((Vh, Bo) => {
  var lf = P(), ff = ue(), mf = /\.|\[(?:[^[\]]*|(["'])(?:(?!\1)[^\\]|\\.)*?\1)\]/, gf = /^\w*$/;
  function yf(e, r) {
    if (lf(e))
      return !1;
    var n = typeof e;
    return n == "number" || n == "symbol" || n == "boolean" || e == null || ff(e) ? !0 : gf.test(e) || !mf.test(e) || r != null && e in Object(
    r);
  }
  t(yf, "isKey");
  Bo.exports = yf;
});

// ../node_modules/lodash/memoize.js
var Ho = i((zh, ko) => {
  var zo = ae(), df = "Expected a function";
  function Ue(e, r) {
    if (typeof e != "function" || r != null && typeof r != "function")
      throw new TypeError(df);
    var n = /* @__PURE__ */ t(function() {
      var o = arguments, a = r ? r.apply(this, o) : o[0], s = n.cache;
      if (s.has(a))
        return s.get(a);
      var u = e.apply(this, o);
      return n.cache = s.set(a, u) || s, u;
    }, "memoized");
    return n.cache = new (Ue.Cache || zo)(), n;
  }
  t(Ue, "memoize");
  Ue.Cache = zo;
  ko.exports = Ue;
});

// ../node_modules/lodash/_memoizeCapped.js
var $o = i((Hh, Ko) => {
  var hf = Ho(), Tf = 500;
  function xf(e) {
    var r = hf(e, function(o) {
      return n.size === Tf && n.clear(), o;
    }), n = r.cache;
    return r;
  }
  t(xf, "memoizeCapped");
  Ko.exports = xf;
});

// ../node_modules/lodash/_stringToPath.js
var Yo = i(($h, Wo) => {
  var bf = $o(), vf = /[^.[\]]+|\[(?:(-?\d+(?:\.\d+)?)|(["'])((?:(?!\2)[^\\]|\\.)*?)\2)\]|(?=(?:\.|\[\])(?:\.|\[\]|$))/g, Df = /\\(\\)?/g, Pf = bf(
  function(e) {
    var r = [];
    return e.charCodeAt(0) === 46 && r.push(""), e.replace(vf, function(n, o, a, s) {
      r.push(a ? s.replace(Df, "$1") : o || n);
    }), r;
  });
  Wo.exports = Pf;
});

// ../node_modules/lodash/_arrayMap.js
var Qo = i((Wh, Xo) => {
  function qf(e, r) {
    for (var n = -1, o = e == null ? 0 : e.length, a = Array(o); ++n < o; )
      a[n] = r(e[n], n, e);
    return a;
  }
  t(qf, "arrayMap");
  Xo.exports = qf;
});

// ../node_modules/lodash/_baseToString.js
var oa = i((Xh, na) => {
  var Zo = B(), Sf = Qo(), _f = P(), Of = ue(), wf = 1 / 0, ea = Zo ? Zo.prototype : void 0, ra = ea ? ea.toString : void 0;
  function ta(e) {
    if (typeof e == "string")
      return e;
    if (_f(e))
      return Sf(e, ta) + "";
    if (Of(e))
      return ra ? ra.call(e) : "";
    var r = e + "";
    return r == "0" && 1 / e == -wf ? "-0" : r;
  }
  t(ta, "baseToString");
  na.exports = ta;
});

// ../node_modules/lodash/toString.js
var ia = i((Zh, aa) => {
  var Af = oa();
  function Ef(e) {
    return e == null ? "" : Af(e);
  }
  t(Ef, "toString");
  aa.exports = Ef;
});

// ../node_modules/lodash/_castPath.js
var Ge = i((rT, sa) => {
  var jf = P(), Cf = pe(), If = Yo(), Ff = ia();
  function Nf(e, r) {
    return jf(e) ? e : Cf(e, r) ? [e] : If(Ff(e));
  }
  t(Nf, "castPath");
  sa.exports = Nf;
});

// ../node_modules/lodash/_toKey.js
var Q = i((nT, ua) => {
  var Rf = ue(), Lf = 1 / 0;
  function Mf(e) {
    if (typeof e == "string" || Rf(e))
      return e;
    var r = e + "";
    return r == "0" && 1 / e == -Lf ? "-0" : r;
  }
  t(Mf, "toKey");
  ua.exports = Mf;
});

// ../node_modules/lodash/_baseGet.js
var Ve = i((aT, pa) => {
  var Jf = Ge(), Uf = Q();
  function Gf(e, r) {
    r = Jf(r, e);
    for (var n = 0, o = r.length; e != null && n < o; )
      e = e[Uf(r[n++])];
    return n && n == o ? e : void 0;
  }
  t(Gf, "baseGet");
  pa.exports = Gf;
});

// ../node_modules/lodash/get.js
var la = i((sT, ca) => {
  var Vf = Ve();
  function Bf(e, r, n) {
    var o = e == null ? void 0 : Vf(e, r);
    return o === void 0 ? n : o;
  }
  t(Bf, "get");
  ca.exports = Bf;
});

// ../node_modules/lodash/_baseHasIn.js
var ma = i((pT, fa) => {
  function zf(e, r) {
    return e != null && r in Object(e);
  }
  t(zf, "baseHasIn");
  fa.exports = zf;
});

// ../node_modules/lodash/_hasPath.js
var ya = i((lT, ga) => {
  var kf = Ge(), Hf = De(), Kf = P(), $f = qe(), Wf = te(), Yf = Q();
  function Xf(e, r, n) {
    r = kf(r, e);
    for (var o = -1, a = r.length, s = !1; ++o < a; ) {
      var u = Yf(r[o]);
      if (!(s = e != null && n(e, u)))
        break;
      e = e[u];
    }
    return s || ++o != a ? s : (a = e == null ? 0 : e.length, !!a && Wf(a) && $f(u, a) && (Kf(e) || Hf(e)));
  }
  t(Xf, "hasPath");
  ga.exports = Xf;
});

// ../node_modules/lodash/hasIn.js
var ha = i((mT, da) => {
  var Qf = ma(), Zf = ya();
  function em(e, r) {
    return e != null && Zf(e, r, Qf);
  }
  t(em, "hasIn");
  da.exports = em;
});

// ../node_modules/lodash/_baseMatchesProperty.js
var xa = i((yT, Ta) => {
  var rm = Le(), tm = la(), nm = ha(), om = pe(), am = Me(), im = Je(), sm = Q(), um = 1, pm = 2;
  function cm(e, r) {
    return om(e) && am(r) ? im(sm(e), r) : function(n) {
      var o = tm(n, e);
      return o === void 0 && o === r ? nm(n, e) : rm(r, o, um | pm);
    };
  }
  t(cm, "baseMatchesProperty");
  Ta.exports = cm;
});

// ../node_modules/lodash/identity.js
var va = i((hT, ba) => {
  function lm(e) {
    return e;
  }
  t(lm, "identity");
  ba.exports = lm;
});

// ../node_modules/lodash/_baseProperty.js
var Pa = i((xT, Da) => {
  function fm(e) {
    return function(r) {
      return r?.[e];
    };
  }
  t(fm, "baseProperty");
  Da.exports = fm;
});

// ../node_modules/lodash/_basePropertyDeep.js
var Sa = i((vT, qa) => {
  var mm = Ve();
  function gm(e) {
    return function(r) {
      return mm(r, e);
    };
  }
  t(gm, "basePropertyDeep");
  qa.exports = gm;
});

// ../node_modules/lodash/property.js
var Oa = i((PT, _a) => {
  var ym = Pa(), dm = Sa(), hm = pe(), Tm = Q();
  function xm(e) {
    return hm(e) ? ym(Tm(e)) : dm(e);
  }
  t(xm, "property");
  _a.exports = xm;
});

// ../node_modules/lodash/_baseIteratee.js
var Aa = i((ST, wa) => {
  var bm = Go(), vm = xa(), Dm = va(), Pm = P(), qm = Oa();
  function Sm(e) {
    return typeof e == "function" ? e : e == null ? Dm : typeof e == "object" ? Pm(e) ? vm(e[0], e[1]) : bm(e) : qm(e);
  }
  t(Sm, "baseIteratee");
  wa.exports = Sm;
});

// ../node_modules/lodash/mapValues.js
var ja = i((OT, Ea) => {
  var _m = Er(), Om = bt(), wm = Aa();
  function Am(e, r) {
    var n = {};
    return r = wm(r, 3), Om(e, function(o, a, s) {
      _m(n, a, r(o, a, s));
    }), n;
  }
  t(Am, "mapValues");
  Ea.exports = Am;
});

// src/docs-tools/argTypes/convert/flow/convert.ts
import { UnknownArgTypesError as mi } from "@storybook/core/preview-errors";
var gi = /* @__PURE__ */ t((e) => e.name === "literal", "isLiteral"), yi = /* @__PURE__ */ t((e) => e.value.replace(/['|"]/g, ""), "toEnumOp\
tion"), di = /* @__PURE__ */ t((e) => {
  switch (e.type) {
    case "function":
      return { name: "function" };
    case "object":
      let r = {};
      return e.signature.properties.forEach((n) => {
        r[n.key] = E(n.value);
      }), {
        name: "object",
        value: r
      };
    default:
      throw new mi({ type: e, language: "Flow" });
  }
}, "convertSig"), E = /* @__PURE__ */ t((e) => {
  let { name: r, raw: n } = e, o = {};
  switch (typeof n < "u" && (o.raw = n), e.name) {
    case "literal":
      return { ...o, name: "other", value: e.value };
    case "string":
    case "number":
    case "symbol":
    case "boolean":
      return { ...o, name: r };
    case "Array":
      return { ...o, name: "array", value: e.elements.map(E) };
    case "signature":
      return { ...o, ...di(e) };
    case "union":
      return e.elements?.every(gi) ? { ...o, name: "enum", value: e.elements?.map(yi) } : { ...o, name: r, value: e.elements?.map(E) };
    case "intersection":
      return { ...o, name: r, value: e.elements?.map(E) };
    default:
      return { ...o, name: "other", value: r };
  }
}, "convert");

// src/docs-tools/argTypes/convert/proptypes/convert.ts
var Ia = fi(ja(), 1);

// src/docs-tools/argTypes/convert/utils.ts
var Ca = /^['"]|['"]$/g, Em = /* @__PURE__ */ t((e) => e.replace(Ca, ""), "trimQuotes"), jm = /* @__PURE__ */ t((e) => Ca.test(e), "includes\
Quotes"), ce = /* @__PURE__ */ t((e) => {
  let r = Em(e);
  return jm(e) || Number.isNaN(Number(r)) ? r : Number(r);
}, "parseLiteral");

// src/docs-tools/argTypes/convert/proptypes/convert.ts
var Cm = /^\(.*\) => /, U = /* @__PURE__ */ t((e) => {
  let { name: r, raw: n, computed: o, value: a } = e, s = {};
  switch (typeof n < "u" && (s.raw = n), r) {
    case "enum": {
      let p = o ? a : a.map((c) => ce(c.value));
      return { ...s, name: r, value: p };
    }
    case "string":
    case "number":
    case "symbol":
      return { ...s, name: r };
    case "func":
      return { ...s, name: "function" };
    case "bool":
    case "boolean":
      return { ...s, name: "boolean" };
    case "arrayOf":
    case "array":
      return { ...s, name: "array", value: a && U(a) };
    case "object":
      return { ...s, name: r };
    case "objectOf":
      return { ...s, name: r, value: U(a) };
    case "shape":
    case "exact":
      let u = (0, Ia.default)(a, (p) => U(p));
      return { ...s, name: "object", value: u };
    case "union":
      return { ...s, name: "union", value: a.map((p) => U(p)) };
    case "instanceOf":
    case "element":
    case "elementType":
    default: {
      if (r?.indexOf("|") > 0)
        try {
          let l = r.split("|").map((f) => JSON.parse(f));
          return { ...s, name: "enum", value: l };
        } catch {
        }
      let p = a ? `${r}(${a})` : r, c = Cm.test(r) ? "function" : "other";
      return { ...s, name: c, value: p };
    }
  }
}, "convert");

// src/docs-tools/argTypes/convert/typescript/convert.ts
import { UnknownArgTypesError as Im } from "@storybook/core/preview-errors";
var Fm = /* @__PURE__ */ t((e) => {
  switch (e.type) {
    case "function":
      return { name: "function" };
    case "object":
      let r = {};
      return e.signature.properties.forEach((n) => {
        r[n.key] = G(n.value);
      }), {
        name: "object",
        value: r
      };
    default:
      throw new Im({ type: e, language: "Typescript" });
  }
}, "convertSig"), G = /* @__PURE__ */ t((e) => {
  let { name: r, raw: n } = e, o = {};
  switch (typeof n < "u" && (o.raw = n), e.name) {
    case "string":
    case "number":
    case "symbol":
    case "boolean":
      return { ...o, name: r };
    case "Array":
      return { ...o, name: "array", value: e.elements.map(G) };
    case "signature":
      return { ...o, ...Fm(e) };
    case "union":
      let a;
      return e.elements?.every((s) => s.name === "literal") ? a = {
        ...o,
        name: "enum",
        // @ts-expect-error fix types
        value: e.elements?.map((s) => ce(s.value))
      } : a = { ...o, name: r, value: e.elements?.map(G) }, a;
    case "intersection":
      return { ...o, name: r, value: e.elements?.map(G) };
    default:
      return { ...o, name: "other", value: r };
  }
}, "convert");

// src/docs-tools/argTypes/convert/index.ts
var le = /* @__PURE__ */ t((e) => {
  let { type: r, tsType: n, flowType: o } = e;
  try {
    if (r != null)
      return U(r);
    if (n != null)
      return G(n);
    if (o != null)
      return E(o);
  } catch (a) {
    console.error(a);
  }
  return null;
}, "convert");

// src/docs-tools/argTypes/docgen/types.ts
var Nm = /* @__PURE__ */ ((a) => (a.JAVASCRIPT = "JavaScript", a.FLOW = "Flow", a.TYPESCRIPT = "TypeScript", a.UNKNOWN = "Unknown", a))(Nm ||
{});

// src/docs-tools/argTypes/docgen/utils/defaultValue.ts
var Rm = ["null", "undefined"];
function V(e) {
  return Rm.some((r) => r === e);
}
t(V, "isDefaultValueBlacklisted");

// src/docs-tools/argTypes/docgen/utils/string.ts
var Fa = /* @__PURE__ */ t((e) => {
  if (!e)
    return "";
  if (typeof e == "string")
    return e;
  throw new Error(`Description: expected string, got: ${JSON.stringify(e)}`);
}, "str");

// src/docs-tools/argTypes/docgen/utils/docgenInfo.ts
function Na(e) {
  return !!e.__docgenInfo;
}
t(Na, "hasDocgen");
function Ra(e) {
  return e != null && Object.keys(e).length > 0;
}
t(Ra, "isValidDocgenSection");
function La(e, r) {
  return Na(e) ? e.__docgenInfo[r] : null;
}
t(La, "getDocgenSection");
function Ma(e) {
  return Na(e) ? Fa(e.__docgenInfo.description) : "";
}
t(Ma, "getDocgenDescription");

// ../node_modules/comment-parser/es6/primitives.js
var b;
(function(e) {
  e.start = "/**", e.nostart = "/***", e.delim = "*", e.end = "*/";
})(b = b || (b = {}));

// ../node_modules/comment-parser/es6/util.js
function Be(e) {
  return /^\s+$/.test(e);
}
t(Be, "isSpace");
function Ja(e) {
  let r = e.match(/\r+$/);
  return r == null ? ["", e] : [e.slice(-r[0].length), e.slice(0, -r[0].length)];
}
t(Ja, "splitCR");
function q(e) {
  let r = e.match(/^\s+/);
  return r == null ? ["", e] : [e.slice(0, r[0].length), e.slice(r[0].length)];
}
t(q, "splitSpace");
function Ua(e) {
  return e.split(/\n/);
}
t(Ua, "splitLines");
function Ga(e = {}) {
  return Object.assign({ tag: "", name: "", type: "", optional: !1, description: "", problems: [], source: [] }, e);
}
t(Ga, "seedSpec");
function ze(e = {}) {
  return Object.assign({ start: "", delimiter: "", postDelimiter: "", tag: "", postTag: "", name: "", postName: "", type: "", postType: "", description: "",
  end: "", lineEnd: "" }, e);
}
t(ze, "seedTokens");

// ../node_modules/comment-parser/es6/parser/block-parser.js
var Lm = /^@\S+/;
function ke({ fence: e = "```" } = {}) {
  let r = Mm(e), n = /* @__PURE__ */ t((o, a) => r(o) ? !a : a, "toggleFence");
  return /* @__PURE__ */ t(function(a) {
    let s = [[]], u = !1;
    for (let p of a)
      Lm.test(p.tokens.description) && !u ? s.push([p]) : s[s.length - 1].push(p), u = n(p.tokens.description, u);
    return s;
  }, "parseBlock");
}
t(ke, "getParser");
function Mm(e) {
  return typeof e == "string" ? (r) => r.split(e).length % 2 === 0 : e;
}
t(Mm, "getFencer");

// ../node_modules/comment-parser/es6/parser/source-parser.js
function He({ startLine: e = 0, markers: r = b } = {}) {
  let n = null, o = e;
  return /* @__PURE__ */ t(function(s) {
    let u = s, p = ze();
    if ([p.lineEnd, u] = Ja(u), [p.start, u] = q(u), n === null && u.startsWith(r.start) && !u.startsWith(r.nostart) && (n = [], p.delimiter =
    u.slice(0, r.start.length), u = u.slice(r.start.length), [p.postDelimiter, u] = q(u)), n === null)
      return o++, null;
    let c = u.trimRight().endsWith(r.end);
    if (p.delimiter === "" && u.startsWith(r.delim) && !u.startsWith(r.end) && (p.delimiter = r.delim, u = u.slice(r.delim.length), [p.postDelimiter,
    u] = q(u)), c) {
      let l = u.trimRight();
      p.end = u.slice(l.length - r.end.length), u = l.slice(0, -r.end.length);
    }
    if (p.description = u, n.push({ number: o, source: s, tokens: p }), o++, c) {
      let l = n.slice();
      return n = null, l;
    }
    return null;
  }, "parseSource");
}
t(He, "getParser");

// ../node_modules/comment-parser/es6/parser/spec-parser.js
function Ke({ tokenizers: e }) {
  return /* @__PURE__ */ t(function(n) {
    var o;
    let a = Ga({ source: n });
    for (let s of e)
      if (a = s(a), !((o = a.problems[a.problems.length - 1]) === null || o === void 0) && o.critical)
        break;
    return a;
  }, "parseSpec");
}
t(Ke, "getParser");

// ../node_modules/comment-parser/es6/parser/tokenizers/tag.js
function fe() {
  return (e) => {
    let { tokens: r } = e.source[0], n = r.description.match(/\s*(@(\S+))(\s*)/);
    return n === null ? (e.problems.push({
      code: "spec:tag:prefix",
      message: 'tag should start with "@" symbol',
      line: e.source[0].number,
      critical: !0
    }), e) : (r.tag = n[1], r.postTag = n[3], r.description = r.description.slice(n[0].length), e.tag = n[2], e);
  };
}
t(fe, "tagTokenizer");

// ../node_modules/comment-parser/es6/parser/tokenizers/type.js
function me(e = "compact") {
  let r = Um(e);
  return (n) => {
    let o = 0, a = [];
    for (let [p, { tokens: c }] of n.source.entries()) {
      let l = "";
      if (p === 0 && c.description[0] !== "{")
        return n;
      for (let f of c.description)
        if (f === "{" && o++, f === "}" && o--, l += f, o === 0)
          break;
      if (a.push([c, l]), o === 0)
        break;
    }
    if (o !== 0)
      return n.problems.push({
        code: "spec:type:unpaired-curlies",
        message: "unpaired curlies",
        line: n.source[0].number,
        critical: !0
      }), n;
    let s = [], u = a[0][0].postDelimiter.length;
    for (let [p, [c, l]] of a.entries())
      c.type = l, p > 0 && (c.type = c.postDelimiter.slice(u) + l, c.postDelimiter = c.postDelimiter.slice(0, u)), [c.postType, c.description] =
      q(c.description.slice(l.length)), s.push(c.type);
    return s[0] = s[0].slice(1), s[s.length - 1] = s[s.length - 1].slice(0, -1), n.type = r(s), n;
  };
}
t(me, "typeTokenizer");
var Jm = /* @__PURE__ */ t((e) => e.trim(), "trim");
function Um(e) {
  return e === "compact" ? (r) => r.map(Jm).join("") : e === "preserve" ? (r) => r.join(`
`) : e;
}
t(Um, "getJoiner");

// ../node_modules/comment-parser/es6/parser/tokenizers/name.js
var Gm = /* @__PURE__ */ t((e) => e && e.startsWith('"') && e.endsWith('"'), "isQuoted");
function ge() {
  let e = /* @__PURE__ */ t((r, { tokens: n }, o) => n.type === "" ? r : o, "typeEnd");
  return (r) => {
    let { tokens: n } = r.source[r.source.reduce(e, 0)], o = n.description.trimLeft(), a = o.split('"');
    if (a.length > 1 && a[0] === "" && a.length % 2 === 1)
      return r.name = a[1], n.name = `"${a[1]}"`, [n.postName, n.description] = q(o.slice(n.name.length)), r;
    let s = 0, u = "", p = !1, c;
    for (let f of o) {
      if (s === 0 && Be(f))
        break;
      f === "[" && s++, f === "]" && s--, u += f;
    }
    if (s !== 0)
      return r.problems.push({
        code: "spec:name:unpaired-brackets",
        message: "unpaired brackets",
        line: r.source[0].number,
        critical: !0
      }), r;
    let l = u;
    if (u[0] === "[" && u[u.length - 1] === "]") {
      p = !0, u = u.slice(1, -1);
      let f = u.split("=");
      if (u = f[0].trim(), f[1] !== void 0 && (c = f.slice(1).join("=").trim()), u === "")
        return r.problems.push({
          code: "spec:name:empty-name",
          message: "empty name",
          line: r.source[0].number,
          critical: !0
        }), r;
      if (c === "")
        return r.problems.push({
          code: "spec:name:empty-default",
          message: "empty default value",
          line: r.source[0].number,
          critical: !0
        }), r;
      if (!Gm(c) && /=(?!>)/.test(c))
        return r.problems.push({
          code: "spec:name:invalid-default",
          message: "invalid default value syntax",
          line: r.source[0].number,
          critical: !0
        }), r;
    }
    return r.optional = p, r.name = u, n.name = l, c !== void 0 && (r.default = c), [n.postName, n.description] = q(o.slice(n.name.length)),
    r;
  };
}
t(ge, "nameTokenizer");

// ../node_modules/comment-parser/es6/parser/tokenizers/description.js
function ye(e = "compact", r = b) {
  let n = $e(e);
  return (o) => (o.description = n(o.source, r), o);
}
t(ye, "descriptionTokenizer");
function $e(e) {
  return e === "compact" ? Vm : e === "preserve" ? km : e;
}
t($e, "getJoiner");
function Vm(e, r = b) {
  return e.map(({ tokens: { description: n } }) => n.trim()).filter((n) => n !== "").join(" ");
}
t(Vm, "compactJoiner");
var Bm = /* @__PURE__ */ t((e, { tokens: r }, n) => r.type === "" ? e : n, "lineNo"), zm = /* @__PURE__ */ t(({ tokens: e }) => (e.delimiter ===
"" ? e.start : e.postDelimiter.slice(1)) + e.description, "getDescription");
function km(e, r = b) {
  if (e.length === 0)
    return "";
  e[0].tokens.description === "" && e[0].tokens.delimiter === r.start && (e = e.slice(1));
  let n = e[e.length - 1];
  return n !== void 0 && n.tokens.description === "" && n.tokens.end.endsWith(r.end) && (e = e.slice(0, -1)), e = e.slice(e.reduce(Bm, 0)), e.
  map(zm).join(`
`);
}
t(km, "preserveJoiner");

// ../node_modules/comment-parser/es6/parser/index.js
function We({ startLine: e = 0, fence: r = "```", spacing: n = "compact", markers: o = b, tokenizers: a = [
  fe(),
  me(n),
  ge(),
  ye(n)
] } = {}) {
  if (e < 0 || e % 1 > 0)
    throw new Error("Invalid startLine");
  let s = He({ startLine: e, markers: o }), u = ke({ fence: r }), p = Ke({ tokenizers: a }), c = $e(n);
  return function(l) {
    let f = [];
    for (let d of Ua(l)) {
      let g = s(d);
      if (g === null)
        continue;
      let x = u(g), h = x.slice(1).map(p);
      f.push({
        description: c(x[0], o),
        tags: h,
        source: g,
        problems: h.reduce((T, v) => T.concat(v.problems), [])
      });
    }
    return f;
  };
}
t(We, "getParser");

// ../node_modules/comment-parser/es6/stringifier/index.js
function Hm(e) {
  return e.start + e.delimiter + e.postDelimiter + e.tag + e.postTag + e.type + e.postType + e.name + e.postName + e.description + e.end + e.
  lineEnd;
}
t(Hm, "join");
function Ye() {
  return (e) => e.source.map(({ tokens: r }) => Hm(r)).join(`
`);
}
t(Ye, "getStringifier");

// ../node_modules/comment-parser/es6/stringifier/inspect.js
var Km = {
  line: 0,
  start: 0,
  delimiter: 0,
  postDelimiter: 0,
  tag: 0,
  postTag: 0,
  name: 0,
  postName: 0,
  type: 0,
  postType: 0,
  description: 0,
  end: 0,
  lineEnd: 0
};
var Hx = Object.keys(Km);

// ../node_modules/comment-parser/es6/index.js
function Va(e, r = {}) {
  return We(r)(e);
}
t(Va, "parse");
var mb = Ye();

// src/docs-tools/argTypes/jsdocParser.ts
import {
  parse as $m,
  stringifyRules as Wm,
  transform as Ym
} from "jsdoc-type-pratt-parser";
function Xm(e) {
  return e != null && e.includes("@");
}
t(Xm, "containsJsDoc");
function Qm(e) {
  let o = `/**
` + (e ?? "").split(`
`).map((s) => ` * ${s}`).join(`
`) + `
*/`, a = Va(o, {
    spacing: "preserve"
  });
  if (!a || a.length === 0)
    throw new Error("Cannot parse JSDoc tags.");
  return a[0];
}
t(Qm, "parse");
var Zm = {
  tags: ["param", "arg", "argument", "returns", "ignore", "deprecated"]
}, Ba = /* @__PURE__ */ t((e, r = Zm) => {
  if (!Xm(e))
    return {
      includesJsDoc: !1,
      ignore: !1
    };
  let n = Qm(e), o = eg(n, r.tags);
  return o.ignore ? {
    includesJsDoc: !0,
    ignore: !0
  } : {
    includesJsDoc: !0,
    ignore: !1,
    // Always use the parsed description to ensure JSDoc is removed from the description.
    description: n.description.trim(),
    extractedTags: o
  };
}, "parseJsDoc");
function eg(e, r) {
  let n = {
    params: null,
    deprecated: null,
    returns: null,
    ignore: !1
  };
  for (let o of e.tags)
    if (!(r !== void 0 && !r.includes(o.tag)))
      if (o.tag === "ignore") {
        n.ignore = !0;
        break;
      } else
        switch (o.tag) {
          case "param":
          case "arg":
          case "argument": {
            let a = tg(o);
            a != null && (n.params == null && (n.params = []), n.params.push(a));
            break;
          }
          case "deprecated": {
            let a = ng(o);
            a != null && (n.deprecated = a);
            break;
          }
          case "returns": {
            let a = og(o);
            a != null && (n.returns = a);
            break;
          }
          default:
            break;
        }
  return n;
}
t(eg, "extractJsDocTags");
function rg(e) {
  return e.replace(/[\.-]$/, "");
}
t(rg, "normaliseParamName");
function tg(e) {
  if (!e.name || e.name === "-")
    return null;
  let r = Ha(e.type);
  return {
    name: e.name,
    type: r,
    description: ka(e.description),
    getPrettyName: /* @__PURE__ */ t(() => rg(e.name), "getPrettyName"),
    getTypeName: /* @__PURE__ */ t(() => r ? Ka(r) : null, "getTypeName")
  };
}
t(tg, "extractParam");
function ng(e) {
  return e.name ? za(e.name, e.description) : null;
}
t(ng, "extractDeprecated");
function za(e, r) {
  let n = e === "" ? r : `${e} ${r}`;
  return ka(n);
}
t(za, "joinNameAndDescription");
function ka(e) {
  let r = e.replace(/^- /g, "").trim();
  return r === "" ? null : r;
}
t(ka, "normaliseDescription");
function og(e) {
  let r = Ha(e.type);
  return r ? {
    type: r,
    description: za(e.name, e.description),
    getTypeName: /* @__PURE__ */ t(() => Ka(r), "getTypeName")
  } : null;
}
t(og, "extractReturns");
var _ = Wm(), ag = _.JsdocTypeObject;
_.JsdocTypeAny = () => "any";
_.JsdocTypeObject = (e, r) => `(${ag(e, r)})`;
_.JsdocTypeOptional = (e, r) => r(e.element);
_.JsdocTypeNullable = (e, r) => r(e.element);
_.JsdocTypeNotNullable = (e, r) => r(e.element);
_.JsdocTypeUnion = (e, r) => e.elements.map(r).join("|");
function Ha(e) {
  try {
    return $m(e, "typescript");
  } catch {
    return null;
  }
}
t(Ha, "extractType");
function Ka(e) {
  return Ym(_, e);
}
t(Ka, "extractTypeName");

// src/docs-tools/argTypes/utils.ts
var Db = 90, Pb = 50;
function Qe(e) {
  return e.length > 90;
}
t(Qe, "isTooLongForTypeSummary");
function $a(e) {
  return e.length > 50;
}
t($a, "isTooLongForDefaultValueSummary");
function y(e, r) {
  return e === r ? { summary: e } : { summary: e, detail: r };
}
t(y, "createSummaryValue");
var qb = /* @__PURE__ */ t((e) => e.replace(/\\r\\n/g, "\\n"), "normalizeNewlines");

// src/docs-tools/argTypes/docgen/flow/createDefaultValue.ts
function Wa(e, r) {
  if (e != null) {
    let { value: n } = e;
    if (!V(n))
      return $a(n) ? y(r?.name, n) : y(n);
  }
  return null;
}
t(Wa, "createDefaultValue");

// src/docs-tools/argTypes/docgen/flow/createType.ts
function Ya({ name: e, value: r, elements: n, raw: o }) {
  return r ?? (n != null ? n.map(Ya).join(" | ") : o ?? e);
}
t(Ya, "generateUnionElement");
function ig({ name: e, raw: r, elements: n }) {
  return n != null ? y(n.map(Ya).join(" | ")) : r != null ? y(r.replace(/^\|\s*/, "")) : y(e);
}
t(ig, "generateUnion");
function sg({ type: e, raw: r }) {
  return r != null ? y(r) : y(e);
}
t(sg, "generateFuncSignature");
function ug({ type: e, raw: r }) {
  return r != null ? Qe(r) ? y(e, r) : y(r) : y(e);
}
t(ug, "generateObjectSignature");
function pg(e) {
  let { type: r } = e;
  return r === "object" ? ug(e) : sg(e);
}
t(pg, "generateSignature");
function cg({ name: e, raw: r }) {
  return r != null ? Qe(r) ? y(e, r) : y(r) : y(e);
}
t(cg, "generateDefault");
function Xa(e) {
  if (e == null)
    return null;
  switch (e.name) {
    case "union":
      return ig(e);
    case "signature":
      return pg(e);
    default:
      return cg(e);
  }
}
t(Xa, "createType");

// src/docs-tools/argTypes/docgen/flow/createPropDef.ts
var Qa = /* @__PURE__ */ t((e, r) => {
  let { flowType: n, description: o, required: a, defaultValue: s } = r;
  return {
    name: e,
    type: Xa(n),
    required: a,
    description: o,
    defaultValue: Wa(s ?? null, n ?? null)
  };
}, "createFlowPropDef");

// src/docs-tools/argTypes/docgen/typeScript/createDefaultValue.ts
function Za({ defaultValue: e }) {
  if (e != null) {
    let { value: r } = e;
    if (!V(r))
      return y(r);
  }
  return null;
}
t(Za, "createDefaultValue");

// src/docs-tools/argTypes/docgen/typeScript/createType.ts
function ei({ tsType: e, required: r }) {
  if (e == null)
    return null;
  let n = e.name;
  return r || (n = n.replace(" | undefined", "")), y(
    ["Array", "Record", "signature"].includes(e.name) ? e.raw : n
  );
}
t(ei, "createType");

// src/docs-tools/argTypes/docgen/typeScript/createPropDef.ts
var ri = /* @__PURE__ */ t((e, r) => {
  let { description: n, required: o } = r;
  return {
    name: e,
    type: ei(r),
    required: o,
    description: n,
    defaultValue: Za(r)
  };
}, "createTsPropDef");

// src/docs-tools/argTypes/docgen/createPropDef.ts
function lg(e) {
  return e != null ? y(e.name) : null;
}
t(lg, "createType");
function fg(e) {
  let { computed: r, func: n } = e;
  return typeof r > "u" && typeof n > "u";
}
t(fg, "isReactDocgenTypescript");
function mg(e) {
  return e ? e.name === "string" ? !0 : e.name === "enum" ? Array.isArray(e.value) && e.value.every(
    ({ value: r }) => typeof r == "string" && r[0] === '"' && r[r.length - 1] === '"'
  ) : !1 : !1;
}
t(mg, "isStringValued");
function gg(e, r) {
  if (e != null) {
    let { value: n } = e;
    if (!V(n))
      return fg(e) && mg(r) ? y(JSON.stringify(n)) : y(n);
  }
  return null;
}
t(gg, "createDefaultValue");
function ti(e, r, n) {
  let { description: o, required: a, defaultValue: s } = n;
  return {
    name: e,
    type: lg(r),
    required: a,
    description: o,
    defaultValue: gg(s, r)
  };
}
t(ti, "createBasicPropDef");
function de(e, r) {
  if (r?.includesJsDoc) {
    let { description: n, extractedTags: o } = r;
    n != null && (e.description = r.description);
    let a = {
      ...o,
      params: o?.params?.map(
        (s) => ({
          name: s.getPrettyName(),
          description: s.description
        })
      )
    };
    Object.values(a).filter(Boolean).length > 0 && (e.jsDocTags = a);
  }
  return e;
}
t(de, "applyJsDocResult");
var yg = /* @__PURE__ */ t((e, r, n) => {
  let o = ti(e, r.type, r);
  return o.sbType = le(r), de(o, n);
}, "javaScriptFactory"), dg = /* @__PURE__ */ t((e, r, n) => {
  let o = ri(e, r);
  return o.sbType = le(r), de(o, n);
}, "tsFactory"), hg = /* @__PURE__ */ t((e, r, n) => {
  let o = Qa(e, r);
  return o.sbType = le(r), de(o, n);
}, "flowFactory"), Tg = /* @__PURE__ */ t((e, r, n) => {
  let o = ti(e, { name: "unknown" }, r);
  return de(o, n);
}, "unknownFactory"), Ze = /* @__PURE__ */ t((e) => {
  switch (e) {
    case "JavaScript":
      return yg;
    case "TypeScript":
      return dg;
    case "Flow":
      return hg;
    default:
      return Tg;
  }
}, "getPropDefFactory");

// src/docs-tools/argTypes/docgen/extractDocgenProps.ts
var ni = /* @__PURE__ */ t((e) => e.type != null ? "JavaScript" : e.flowType != null ? "Flow" : e.tsType != null ? "TypeScript" : "Unknown",
"getTypeSystem"), xg = /* @__PURE__ */ t((e) => {
  let r = ni(e[0]), n = Ze(r);
  return e.map((o) => {
    let a = o;
    return o.type?.elements && (a = {
      ...o,
      type: {
        ...o.type,
        value: o.type.elements
      }
    }), oi(a.name, a, r, n);
  });
}, "extractComponentSectionArray"), bg = /* @__PURE__ */ t((e) => {
  let r = Object.keys(e), n = ni(e[r[0]]), o = Ze(n);
  return r.map((a) => {
    let s = e[a];
    return s != null ? oi(a, s, n, o) : null;
  }).filter(Boolean);
}, "extractComponentSectionObject"), av = /* @__PURE__ */ t((e, r) => {
  let n = La(e, r);
  return Ra(n) ? Array.isArray(n) ? xg(n) : bg(n) : [];
}, "extractComponentProps");
function oi(e, r, n, o) {
  let a = Ba(r.description);
  return a.includesJsDoc && a.ignore ? null : {
    propDef: o(e, r, a),
    jsDocTags: a.extractedTags,
    docgenInfo: r,
    typeSystem: n
  };
}
t(oi, "extractProp");
function iv(e) {
  return e != null ? Ma(e) : "";
}
t(iv, "extractComponentDescription");

// src/docs-tools/argTypes/enhanceArgTypes.ts
import { combineParameters as vg } from "@storybook/core/preview-api";
var cv = /* @__PURE__ */ t((e) => {
  let {
    component: r,
    argTypes: n,
    parameters: { docs: o = {} }
  } = e, { extractArgTypes: a } = o, s = a && r ? a(r) : {};
  return s ? vg(s, n) : n;
}, "enhanceArgTypes");

// src/docs-tools/shared.ts
var ai = "storybook/docs", mv = `${ai}/panel`, gv = "docs", yv = `${ai}/snippet-rendered`, Dg = /* @__PURE__ */ ((o) => (o.AUTO = "auto", o.
CODE = "code", o.DYNAMIC = "dynamic", o))(Dg || {});

// src/docs-tools/hasDocsOrControls.ts
var Pg = /(addons\/|addon-|addon-essentials\/)(docs|controls)/, hv = /* @__PURE__ */ t((e) => e.presetsList?.some((r) => Pg.test(r.name)), "\
hasDocsOrControls");
export {
  ai as ADDON_ID,
  Pb as MAX_DEFAULT_VALUE_SUMMARY_LENGTH,
  Db as MAX_TYPE_SUMMARY_LENGTH,
  mv as PANEL_ID,
  gv as PARAM_KEY,
  yv as SNIPPET_RENDERED,
  Dg as SourceType,
  Nm as TypeSystem,
  le as convert,
  y as createSummaryValue,
  cv as enhanceArgTypes,
  iv as extractComponentDescription,
  av as extractComponentProps,
  xg as extractComponentSectionArray,
  bg as extractComponentSectionObject,
  Ma as getDocgenDescription,
  La as getDocgenSection,
  Na as hasDocgen,
  hv as hasDocsOrControls,
  V as isDefaultValueBlacklisted,
  $a as isTooLongForDefaultValueSummary,
  Qe as isTooLongForTypeSummary,
  Ra as isValidDocgenSection,
  qb as normalizeNewlines,
  Ba as parseJsDoc,
  Fa as str
};
