// @storybook/core/dist/docs-tools/index.cjs
"use strict";
var yi = Object.create;
var z = Object.defineProperty;
var di = Object.getOwnPropertyDescriptor;
var hi = Object.getOwnPropertyNames;
var Ti = Object.getPrototypeOf, xi = Object.prototype.hasOwnProperty;
var t = (e, r) => z(e, "name", { value: r, configurable: !0 });
var i = (e, r) => () => (r || e((r = { exports: {} }).exports, r), r.exports), bi = (e, r) => {
  for (var n in r)
    z(e, n, { get: r[n], enumerable: !0 });
}, cr = (e, r, n, o) => {
  if (r && typeof r == "object" || typeof r == "function")
    for (let a of hi(r))
      !xi.call(e, a) && a !== n && z(e, a, { get: () => r[a], enumerable: !(o = di(r, a)) || o.enumerable });
  return e;
};
var vi = (e, r, n) => (n = e != null ? yi(Ti(e)) : {}, cr(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  r || !e || !e.__esModule ? z(n, "default", { value: e, enumerable: !0 }) : n,
  e
)), Di = (e) => cr(z({}, "__esModule", { value: !0 }), e);

// ../node_modules/lodash/_freeGlobal.js
var be = i((Mg, fr) => {
  var _i = typeof global == "object" && global && global.Object === Object && global;
  fr.exports = _i;
});

// ../node_modules/lodash/_root.js
var D = i((Jg, mr) => {
  var Oi = be(), wi = typeof self == "object" && self && self.Object === Object && self, Ai = Oi || wi || Function("return this")();
  mr.exports = Ai;
});

// ../node_modules/lodash/_Symbol.js
var k = i((Ug, gr) => {
  var Ei = D(), ji = Ei.Symbol;
  gr.exports = ji;
});

// ../node_modules/lodash/_getRawTag.js
var Tr = i((Gg, hr) => {
  var yr = k(), dr = Object.prototype, Ci = dr.hasOwnProperty, Ii = dr.toString, H = yr ? yr.toStringTag : void 0;
  function Fi(e) {
    var r = Ci.call(e, H), n = e[H];
    try {
      e[H] = void 0;
      var o = !0;
    } catch {
    }
    var a = Ii.call(e);
    return o && (r ? e[H] = n : delete e[H]), a;
  }
  t(Fi, "getRawTag");
  hr.exports = Fi;
});

// ../node_modules/lodash/_objectToString.js
var br = i((Bg, xr) => {
  var Ni = Object.prototype, Ri = Ni.toString;
  function Li(e) {
    return Ri.call(e);
  }
  t(Li, "objectToString");
  xr.exports = Li;
});

// ../node_modules/lodash/_baseGetTag.js
var C = i((kg, Pr) => {
  var vr = k(), Mi = Tr(), Ji = br(), Ui = "[object Null]", Gi = "[object Undefined]", Dr = vr ? vr.toStringTag : void 0;
  function Vi(e) {
    return e == null ? e === void 0 ? Gi : Ui : Dr && Dr in Object(e) ? Mi(e) : Ji(e);
  }
  t(Vi, "baseGetTag");
  Pr.exports = Vi;
});

// ../node_modules/lodash/isObject.js
var oe = i((Kg, qr) => {
  function Bi(e) {
    var r = typeof e;
    return e != null && (r == "object" || r == "function");
  }
  t(Bi, "isObject");
  qr.exports = Bi;
});

// ../node_modules/lodash/isFunction.js
var ve = i((Wg, Sr) => {
  var zi = C(), ki = oe(), Hi = "[object AsyncFunction]", Ki = "[object Function]", $i = "[object GeneratorFunction]", Wi = "[object Proxy]";
  function Yi(e) {
    if (!ki(e))
      return !1;
    var r = zi(e);
    return r == Ki || r == $i || r == Hi || r == Wi;
  }
  t(Yi, "isFunction");
  Sr.exports = Yi;
});

// ../node_modules/lodash/_coreJsData.js
var Or = i((Xg, _r) => {
  var Xi = D(), Qi = Xi["__core-js_shared__"];
  _r.exports = Qi;
});

// ../node_modules/lodash/_isMasked.js
var Er = i((Qg, Ar) => {
  var De = Or(), wr = function() {
    var e = /[^.]+$/.exec(De && De.keys && De.keys.IE_PROTO || "");
    return e ? "Symbol(src)_1." + e : "";
  }();
  function Zi(e) {
    return !!wr && wr in e;
  }
  t(Zi, "isMasked");
  Ar.exports = Zi;
});

// ../node_modules/lodash/_toSource.js
var Pe = i((ey, jr) => {
  var es = Function.prototype, rs = es.toString;
  function ts(e) {
    if (e != null) {
      try {
        return rs.call(e);
      } catch {
      }
      try {
        return e + "";
      } catch {
      }
    }
    return "";
  }
  t(ts, "toSource");
  jr.exports = ts;
});

// ../node_modules/lodash/_baseIsNative.js
var Ir = i((ty, Cr) => {
  var ns = ve(), os = Er(), as = oe(), is = Pe(), ss = /[\\^$.*+?()[\]{}|]/g, us = /^\[object .+?Constructor\]$/, ps = Function.prototype, cs = Object.
  prototype, ls = ps.toString, fs = cs.hasOwnProperty, ms = RegExp(
    "^" + ls.call(fs).replace(ss, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
  );
  function gs(e) {
    if (!as(e) || os(e))
      return !1;
    var r = ns(e) ? ms : us;
    return r.test(is(e));
  }
  t(gs, "baseIsNative");
  Cr.exports = gs;
});

// ../node_modules/lodash/_getValue.js
var Nr = i((oy, Fr) => {
  function ys(e, r) {
    return e?.[r];
  }
  t(ys, "getValue");
  Fr.exports = ys;
});

// ../node_modules/lodash/_getNative.js
var S = i((iy, Rr) => {
  var ds = Ir(), hs = Nr();
  function Ts(e, r) {
    var n = hs(e, r);
    return ds(n) ? n : void 0;
  }
  t(Ts, "getNative");
  Rr.exports = Ts;
});

// ../node_modules/lodash/_defineProperty.js
var Mr = i((uy, Lr) => {
  var xs = S(), bs = function() {
    try {
      var e = xs(Object, "defineProperty");
      return e({}, "", {}), e;
    } catch {
    }
  }();
  Lr.exports = bs;
});

// ../node_modules/lodash/_baseAssignValue.js
var Gr = i((py, Ur) => {
  var Jr = Mr();
  function vs(e, r, n) {
    r == "__proto__" && Jr ? Jr(e, r, {
      configurable: !0,
      enumerable: !0,
      value: n,
      writable: !0
    }) : e[r] = n;
  }
  t(vs, "baseAssignValue");
  Ur.exports = vs;
});

// ../node_modules/lodash/_createBaseFor.js
var Br = i((ly, Vr) => {
  function Ds(e) {
    return function(r, n, o) {
      for (var a = -1, s = Object(r), u = o(r), p = u.length; p--; ) {
        var c = u[e ? p : ++a];
        if (n(s[c], c, s) === !1)
          break;
      }
      return r;
    };
  }
  t(Ds, "createBaseFor");
  Vr.exports = Ds;
});

// ../node_modules/lodash/_baseFor.js
var kr = i((my, zr) => {
  var Ps = Br(), qs = Ps();
  zr.exports = qs;
});

// ../node_modules/lodash/_baseTimes.js
var Kr = i((gy, Hr) => {
  function Ss(e, r) {
    for (var n = -1, o = Array(e); ++n < e; )
      o[n] = r(n);
    return o;
  }
  t(Ss, "baseTimes");
  Hr.exports = Ss;
});

// ../node_modules/lodash/isObjectLike.js
var I = i((dy, $r) => {
  function _s(e) {
    return e != null && typeof e == "object";
  }
  t(_s, "isObjectLike");
  $r.exports = _s;
});

// ../node_modules/lodash/_baseIsArguments.js
var Yr = i((Ty, Wr) => {
  var Os = C(), ws = I(), As = "[object Arguments]";
  function Es(e) {
    return ws(e) && Os(e) == As;
  }
  t(Es, "baseIsArguments");
  Wr.exports = Es;
});

// ../node_modules/lodash/isArguments.js
var qe = i((by, Zr) => {
  var Xr = Yr(), js = I(), Qr = Object.prototype, Cs = Qr.hasOwnProperty, Is = Qr.propertyIsEnumerable, Fs = Xr(/* @__PURE__ */ function() {
    return arguments;
  }()) ? Xr : function(e) {
    return js(e) && Cs.call(e, "callee") && !Is.call(e, "callee");
  };
  Zr.exports = Fs;
});

// ../node_modules/lodash/isArray.js
var P = i((vy, et) => {
  var Ns = Array.isArray;
  et.exports = Ns;
});

// ../node_modules/lodash/stubFalse.js
var tt = i((Dy, rt) => {
  function Rs() {
    return !1;
  }
  t(Rs, "stubFalse");
  rt.exports = Rs;
});

// ../node_modules/lodash/isBuffer.js
var Se = i((K, F) => {
  var Ls = D(), Ms = tt(), at = typeof K == "object" && K && !K.nodeType && K, nt = at && typeof F == "object" && F && !F.nodeType && F, Js = nt &&
  nt.exports === at, ot = Js ? Ls.Buffer : void 0, Us = ot ? ot.isBuffer : void 0, Gs = Us || Ms;
  F.exports = Gs;
});

// ../node_modules/lodash/_isIndex.js
var _e = i((qy, it) => {
  var Vs = 9007199254740991, Bs = /^(?:0|[1-9]\d*)$/;
  function zs(e, r) {
    var n = typeof e;
    return r = r ?? Vs, !!r && (n == "number" || n != "symbol" && Bs.test(e)) && e > -1 && e % 1 == 0 && e < r;
  }
  t(zs, "isIndex");
  it.exports = zs;
});

// ../node_modules/lodash/isLength.js
var ae = i((_y, st) => {
  var ks = 9007199254740991;
  function Hs(e) {
    return typeof e == "number" && e > -1 && e % 1 == 0 && e <= ks;
  }
  t(Hs, "isLength");
  st.exports = Hs;
});

// ../node_modules/lodash/_baseIsTypedArray.js
var pt = i((wy, ut) => {
  var Ks = C(), $s = ae(), Ws = I(), Ys = "[object Arguments]", Xs = "[object Array]", Qs = "[object Boolean]", Zs = "[object Date]", eu = "\
[object Error]", ru = "[object Function]", tu = "[object Map]", nu = "[object Number]", ou = "[object Object]", au = "[object RegExp]", iu = "\
[object Set]", su = "[object String]", uu = "[object WeakMap]", pu = "[object ArrayBuffer]", cu = "[object DataView]", lu = "[object Float32\
Array]", fu = "[object Float64Array]", mu = "[object Int8Array]", gu = "[object Int16Array]", yu = "[object Int32Array]", du = "[object Uint\
8Array]", hu = "[object Uint8ClampedArray]", Tu = "[object Uint16Array]", xu = "[object Uint32Array]", m = {};
  m[lu] = m[fu] = m[mu] = m[gu] = m[yu] = m[du] = m[hu] = m[Tu] = m[xu] = !0;
  m[Ys] = m[Xs] = m[pu] = m[Qs] = m[cu] = m[Zs] = m[eu] = m[ru] = m[tu] = m[nu] = m[ou] = m[au] = m[iu] = m[su] = m[uu] = !1;
  function bu(e) {
    return Ws(e) && $s(e.length) && !!m[Ks(e)];
  }
  t(bu, "baseIsTypedArray");
  ut.exports = bu;
});

// ../node_modules/lodash/_baseUnary.js
var lt = i((Ey, ct) => {
  function vu(e) {
    return function(r) {
      return e(r);
    };
  }
  t(vu, "baseUnary");
  ct.exports = vu;
});

// ../node_modules/lodash/_nodeUtil.js
var mt = i(($, N) => {
  var Du = be(), ft = typeof $ == "object" && $ && !$.nodeType && $, W = ft && typeof N == "object" && N && !N.nodeType && N, Pu = W && W.exports ===
  ft, Oe = Pu && Du.process, qu = function() {
    try {
      var e = W && W.require && W.require("util").types;
      return e || Oe && Oe.binding && Oe.binding("util");
    } catch {
    }
  }();
  N.exports = qu;
});

// ../node_modules/lodash/isTypedArray.js
var we = i((Cy, dt) => {
  var Su = pt(), _u = lt(), gt = mt(), yt = gt && gt.isTypedArray, Ou = yt ? _u(yt) : Su;
  dt.exports = Ou;
});

// ../node_modules/lodash/_arrayLikeKeys.js
var Tt = i((Iy, ht) => {
  var wu = Kr(), Au = qe(), Eu = P(), ju = Se(), Cu = _e(), Iu = we(), Fu = Object.prototype, Nu = Fu.hasOwnProperty;
  function Ru(e, r) {
    var n = Eu(e), o = !n && Au(e), a = !n && !o && ju(e), s = !n && !o && !a && Iu(e), u = n || o || a || s, p = u ? wu(e.length, String) :
    [], c = p.length;
    for (var l in e)
      (r || Nu.call(e, l)) && !(u && // Safari 9 has enumerable `arguments.length` in strict mode.
      (l == "length" || // Node.js 0.10 has enumerable non-index properties on buffers.
      a && (l == "offset" || l == "parent") || // PhantomJS 2 has enumerable non-index properties on typed arrays.
      s && (l == "buffer" || l == "byteLength" || l == "byteOffset") || // Skip index properties.
      Cu(l, c))) && p.push(l);
    return p;
  }
  t(Ru, "arrayLikeKeys");
  ht.exports = Ru;
});

// ../node_modules/lodash/_isPrototype.js
var bt = i((Ny, xt) => {
  var Lu = Object.prototype;
  function Mu(e) {
    var r = e && e.constructor, n = typeof r == "function" && r.prototype || Lu;
    return e === n;
  }
  t(Mu, "isPrototype");
  xt.exports = Mu;
});

// ../node_modules/lodash/_overArg.js
var Dt = i((Ly, vt) => {
  function Ju(e, r) {
    return function(n) {
      return e(r(n));
    };
  }
  t(Ju, "overArg");
  vt.exports = Ju;
});

// ../node_modules/lodash/_nativeKeys.js
var qt = i((Jy, Pt) => {
  var Uu = Dt(), Gu = Uu(Object.keys, Object);
  Pt.exports = Gu;
});

// ../node_modules/lodash/_baseKeys.js
var _t = i((Uy, St) => {
  var Vu = bt(), Bu = qt(), zu = Object.prototype, ku = zu.hasOwnProperty;
  function Hu(e) {
    if (!Vu(e))
      return Bu(e);
    var r = [];
    for (var n in Object(e))
      ku.call(e, n) && n != "constructor" && r.push(n);
    return r;
  }
  t(Hu, "baseKeys");
  St.exports = Hu;
});

// ../node_modules/lodash/isArrayLike.js
var wt = i((Vy, Ot) => {
  var Ku = ve(), $u = ae();
  function Wu(e) {
    return e != null && $u(e.length) && !Ku(e);
  }
  t(Wu, "isArrayLike");
  Ot.exports = Wu;
});

// ../node_modules/lodash/keys.js
var ie = i((zy, At) => {
  var Yu = Tt(), Xu = _t(), Qu = wt();
  function Zu(e) {
    return Qu(e) ? Yu(e) : Xu(e);
  }
  t(Zu, "keys");
  At.exports = Zu;
});

// ../node_modules/lodash/_baseForOwn.js
var jt = i((Hy, Et) => {
  var ep = kr(), rp = ie();
  function tp(e, r) {
    return e && ep(e, r, rp);
  }
  t(tp, "baseForOwn");
  Et.exports = tp;
});

// ../node_modules/lodash/_listCacheClear.js
var It = i(($y, Ct) => {
  function np() {
    this.__data__ = [], this.size = 0;
  }
  t(np, "listCacheClear");
  Ct.exports = np;
});

// ../node_modules/lodash/eq.js
var Ae = i((Yy, Ft) => {
  function op(e, r) {
    return e === r || e !== e && r !== r;
  }
  t(op, "eq");
  Ft.exports = op;
});

// ../node_modules/lodash/_assocIndexOf.js
var Y = i((Qy, Nt) => {
  var ap = Ae();
  function ip(e, r) {
    for (var n = e.length; n--; )
      if (ap(e[n][0], r))
        return n;
    return -1;
  }
  t(ip, "assocIndexOf");
  Nt.exports = ip;
});

// ../node_modules/lodash/_listCacheDelete.js
var Lt = i((ed, Rt) => {
  var sp = Y(), up = Array.prototype, pp = up.splice;
  function cp(e) {
    var r = this.__data__, n = sp(r, e);
    if (n < 0)
      return !1;
    var o = r.length - 1;
    return n == o ? r.pop() : pp.call(r, n, 1), --this.size, !0;
  }
  t(cp, "listCacheDelete");
  Rt.exports = cp;
});

// ../node_modules/lodash/_listCacheGet.js
var Jt = i((td, Mt) => {
  var lp = Y();
  function fp(e) {
    var r = this.__data__, n = lp(r, e);
    return n < 0 ? void 0 : r[n][1];
  }
  t(fp, "listCacheGet");
  Mt.exports = fp;
});

// ../node_modules/lodash/_listCacheHas.js
var Gt = i((od, Ut) => {
  var mp = Y();
  function gp(e) {
    return mp(this.__data__, e) > -1;
  }
  t(gp, "listCacheHas");
  Ut.exports = gp;
});

// ../node_modules/lodash/_listCacheSet.js
var Bt = i((id, Vt) => {
  var yp = Y();
  function dp(e, r) {
    var n = this.__data__, o = yp(n, e);
    return o < 0 ? (++this.size, n.push([e, r])) : n[o][1] = r, this;
  }
  t(dp, "listCacheSet");
  Vt.exports = dp;
});

// ../node_modules/lodash/_ListCache.js
var X = i((ud, zt) => {
  var hp = It(), Tp = Lt(), xp = Jt(), bp = Gt(), vp = Bt();
  function R(e) {
    var r = -1, n = e == null ? 0 : e.length;
    for (this.clear(); ++r < n; ) {
      var o = e[r];
      this.set(o[0], o[1]);
    }
  }
  t(R, "ListCache");
  R.prototype.clear = hp;
  R.prototype.delete = Tp;
  R.prototype.get = xp;
  R.prototype.has = bp;
  R.prototype.set = vp;
  zt.exports = R;
});

// ../node_modules/lodash/_stackClear.js
var Ht = i((cd, kt) => {
  var Dp = X();
  function Pp() {
    this.__data__ = new Dp(), this.size = 0;
  }
  t(Pp, "stackClear");
  kt.exports = Pp;
});

// ../node_modules/lodash/_stackDelete.js
var $t = i((fd, Kt) => {
  function qp(e) {
    var r = this.__data__, n = r.delete(e);
    return this.size = r.size, n;
  }
  t(qp, "stackDelete");
  Kt.exports = qp;
});

// ../node_modules/lodash/_stackGet.js
var Yt = i((gd, Wt) => {
  function Sp(e) {
    return this.__data__.get(e);
  }
  t(Sp, "stackGet");
  Wt.exports = Sp;
});

// ../node_modules/lodash/_stackHas.js
var Qt = i((dd, Xt) => {
  function _p(e) {
    return this.__data__.has(e);
  }
  t(_p, "stackHas");
  Xt.exports = _p;
});

// ../node_modules/lodash/_Map.js
var se = i((Td, Zt) => {
  var Op = S(), wp = D(), Ap = Op(wp, "Map");
  Zt.exports = Ap;
});

// ../node_modules/lodash/_nativeCreate.js
var Q = i((xd, en) => {
  var Ep = S(), jp = Ep(Object, "create");
  en.exports = jp;
});

// ../node_modules/lodash/_hashClear.js
var nn = i((bd, tn) => {
  var rn = Q();
  function Cp() {
    this.__data__ = rn ? rn(null) : {}, this.size = 0;
  }
  t(Cp, "hashClear");
  tn.exports = Cp;
});

// ../node_modules/lodash/_hashDelete.js
var an = i((Dd, on) => {
  function Ip(e) {
    var r = this.has(e) && delete this.__data__[e];
    return this.size -= r ? 1 : 0, r;
  }
  t(Ip, "hashDelete");
  on.exports = Ip;
});

// ../node_modules/lodash/_hashGet.js
var un = i((qd, sn) => {
  var Fp = Q(), Np = "__lodash_hash_undefined__", Rp = Object.prototype, Lp = Rp.hasOwnProperty;
  function Mp(e) {
    var r = this.__data__;
    if (Fp) {
      var n = r[e];
      return n === Np ? void 0 : n;
    }
    return Lp.call(r, e) ? r[e] : void 0;
  }
  t(Mp, "hashGet");
  sn.exports = Mp;
});

// ../node_modules/lodash/_hashHas.js
var cn = i((_d, pn) => {
  var Jp = Q(), Up = Object.prototype, Gp = Up.hasOwnProperty;
  function Vp(e) {
    var r = this.__data__;
    return Jp ? r[e] !== void 0 : Gp.call(r, e);
  }
  t(Vp, "hashHas");
  pn.exports = Vp;
});

// ../node_modules/lodash/_hashSet.js
var fn = i((wd, ln) => {
  var Bp = Q(), zp = "__lodash_hash_undefined__";
  function kp(e, r) {
    var n = this.__data__;
    return this.size += this.has(e) ? 0 : 1, n[e] = Bp && r === void 0 ? zp : r, this;
  }
  t(kp, "hashSet");
  ln.exports = kp;
});

// ../node_modules/lodash/_Hash.js
var gn = i((Ed, mn) => {
  var Hp = nn(), Kp = an(), $p = un(), Wp = cn(), Yp = fn();
  function L(e) {
    var r = -1, n = e == null ? 0 : e.length;
    for (this.clear(); ++r < n; ) {
      var o = e[r];
      this.set(o[0], o[1]);
    }
  }
  t(L, "Hash");
  L.prototype.clear = Hp;
  L.prototype.delete = Kp;
  L.prototype.get = $p;
  L.prototype.has = Wp;
  L.prototype.set = Yp;
  mn.exports = L;
});

// ../node_modules/lodash/_mapCacheClear.js
var hn = i((Cd, dn) => {
  var yn = gn(), Xp = X(), Qp = se();
  function Zp() {
    this.size = 0, this.__data__ = {
      hash: new yn(),
      map: new (Qp || Xp)(),
      string: new yn()
    };
  }
  t(Zp, "mapCacheClear");
  dn.exports = Zp;
});

// ../node_modules/lodash/_isKeyable.js
var xn = i((Fd, Tn) => {
  function ec(e) {
    var r = typeof e;
    return r == "string" || r == "number" || r == "symbol" || r == "boolean" ? e !== "__proto__" : e === null;
  }
  t(ec, "isKeyable");
  Tn.exports = ec;
});

// ../node_modules/lodash/_getMapData.js
var Z = i((Rd, bn) => {
  var rc = xn();
  function tc(e, r) {
    var n = e.__data__;
    return rc(r) ? n[typeof r == "string" ? "string" : "hash"] : n.map;
  }
  t(tc, "getMapData");
  bn.exports = tc;
});

// ../node_modules/lodash/_mapCacheDelete.js
var Dn = i((Md, vn) => {
  var nc = Z();
  function oc(e) {
    var r = nc(this, e).delete(e);
    return this.size -= r ? 1 : 0, r;
  }
  t(oc, "mapCacheDelete");
  vn.exports = oc;
});

// ../node_modules/lodash/_mapCacheGet.js
var qn = i((Ud, Pn) => {
  var ac = Z();
  function ic(e) {
    return ac(this, e).get(e);
  }
  t(ic, "mapCacheGet");
  Pn.exports = ic;
});

// ../node_modules/lodash/_mapCacheHas.js
var _n = i((Vd, Sn) => {
  var sc = Z();
  function uc(e) {
    return sc(this, e).has(e);
  }
  t(uc, "mapCacheHas");
  Sn.exports = uc;
});

// ../node_modules/lodash/_mapCacheSet.js
var wn = i((zd, On) => {
  var pc = Z();
  function cc(e, r) {
    var n = pc(this, e), o = n.size;
    return n.set(e, r), this.size += n.size == o ? 0 : 1, this;
  }
  t(cc, "mapCacheSet");
  On.exports = cc;
});

// ../node_modules/lodash/_MapCache.js
var ue = i((Hd, An) => {
  var lc = hn(), fc = Dn(), mc = qn(), gc = _n(), yc = wn();
  function M(e) {
    var r = -1, n = e == null ? 0 : e.length;
    for (this.clear(); ++r < n; ) {
      var o = e[r];
      this.set(o[0], o[1]);
    }
  }
  t(M, "MapCache");
  M.prototype.clear = lc;
  M.prototype.delete = fc;
  M.prototype.get = mc;
  M.prototype.has = gc;
  M.prototype.set = yc;
  An.exports = M;
});

// ../node_modules/lodash/_stackSet.js
var jn = i(($d, En) => {
  var dc = X(), hc = se(), Tc = ue(), xc = 200;
  function bc(e, r) {
    var n = this.__data__;
    if (n instanceof dc) {
      var o = n.__data__;
      if (!hc || o.length < xc - 1)
        return o.push([e, r]), this.size = ++n.size, this;
      n = this.__data__ = new Tc(o);
    }
    return n.set(e, r), this.size = n.size, this;
  }
  t(bc, "stackSet");
  En.exports = bc;
});

// ../node_modules/lodash/_Stack.js
var Ee = i((Yd, Cn) => {
  var vc = X(), Dc = Ht(), Pc = $t(), qc = Yt(), Sc = Qt(), _c = jn();
  function J(e) {
    var r = this.__data__ = new vc(e);
    this.size = r.size;
  }
  t(J, "Stack");
  J.prototype.clear = Dc;
  J.prototype.delete = Pc;
  J.prototype.get = qc;
  J.prototype.has = Sc;
  J.prototype.set = _c;
  Cn.exports = J;
});

// ../node_modules/lodash/_setCacheAdd.js
var Fn = i((Qd, In) => {
  var Oc = "__lodash_hash_undefined__";
  function wc(e) {
    return this.__data__.set(e, Oc), this;
  }
  t(wc, "setCacheAdd");
  In.exports = wc;
});

// ../node_modules/lodash/_setCacheHas.js
var Rn = i((eh, Nn) => {
  function Ac(e) {
    return this.__data__.has(e);
  }
  t(Ac, "setCacheHas");
  Nn.exports = Ac;
});

// ../node_modules/lodash/_SetCache.js
var Mn = i((th, Ln) => {
  var Ec = ue(), jc = Fn(), Cc = Rn();
  function pe(e) {
    var r = -1, n = e == null ? 0 : e.length;
    for (this.__data__ = new Ec(); ++r < n; )
      this.add(e[r]);
  }
  t(pe, "SetCache");
  pe.prototype.add = pe.prototype.push = jc;
  pe.prototype.has = Cc;
  Ln.exports = pe;
});

// ../node_modules/lodash/_arraySome.js
var Un = i((oh, Jn) => {
  function Ic(e, r) {
    for (var n = -1, o = e == null ? 0 : e.length; ++n < o; )
      if (r(e[n], n, e))
        return !0;
    return !1;
  }
  t(Ic, "arraySome");
  Jn.exports = Ic;
});

// ../node_modules/lodash/_cacheHas.js
var Vn = i((ih, Gn) => {
  function Fc(e, r) {
    return e.has(r);
  }
  t(Fc, "cacheHas");
  Gn.exports = Fc;
});

// ../node_modules/lodash/_equalArrays.js
var je = i((uh, Bn) => {
  var Nc = Mn(), Rc = Un(), Lc = Vn(), Mc = 1, Jc = 2;
  function Uc(e, r, n, o, a, s) {
    var u = n & Mc, p = e.length, c = r.length;
    if (p != c && !(u && c > p))
      return !1;
    var l = s.get(e), f = s.get(r);
    if (l && f)
      return l == r && f == e;
    var d = -1, g = !0, x = n & Jc ? new Nc() : void 0;
    for (s.set(e, r), s.set(r, e); ++d < p; ) {
      var h = e[d], T = r[d];
      if (o)
        var v = u ? o(T, h, d, r, e, s) : o(h, T, d, e, r, s);
      if (v !== void 0) {
        if (v)
          continue;
        g = !1;
        break;
      }
      if (x) {
        if (!Rc(r, function(O, w) {
          if (!Lc(x, w) && (h === O || a(h, O, n, o, s)))
            return x.push(w);
        })) {
          g = !1;
          break;
        }
      } else if (!(h === T || a(h, T, n, o, s))) {
        g = !1;
        break;
      }
    }
    return s.delete(e), s.delete(r), g;
  }
  t(Uc, "equalArrays");
  Bn.exports = Uc;
});

// ../node_modules/lodash/_Uint8Array.js
var kn = i((ch, zn) => {
  var Gc = D(), Vc = Gc.Uint8Array;
  zn.exports = Vc;
});

// ../node_modules/lodash/_mapToArray.js
var Kn = i((lh, Hn) => {
  function Bc(e) {
    var r = -1, n = Array(e.size);
    return e.forEach(function(o, a) {
      n[++r] = [a, o];
    }), n;
  }
  t(Bc, "mapToArray");
  Hn.exports = Bc;
});

// ../node_modules/lodash/_setToArray.js
var Wn = i((mh, $n) => {
  function zc(e) {
    var r = -1, n = Array(e.size);
    return e.forEach(function(o) {
      n[++r] = o;
    }), n;
  }
  t(zc, "setToArray");
  $n.exports = zc;
});

// ../node_modules/lodash/_equalByTag.js
var eo = i((yh, Zn) => {
  var Yn = k(), Xn = kn(), kc = Ae(), Hc = je(), Kc = Kn(), $c = Wn(), Wc = 1, Yc = 2, Xc = "[object Boolean]", Qc = "[object Date]", Zc = "\
[object Error]", el = "[object Map]", rl = "[object Number]", tl = "[object RegExp]", nl = "[object Set]", ol = "[object String]", al = "[ob\
ject Symbol]", il = "[object ArrayBuffer]", sl = "[object DataView]", Qn = Yn ? Yn.prototype : void 0, Ce = Qn ? Qn.valueOf : void 0;
  function ul(e, r, n, o, a, s, u) {
    switch (n) {
      case sl:
        if (e.byteLength != r.byteLength || e.byteOffset != r.byteOffset)
          return !1;
        e = e.buffer, r = r.buffer;
      case il:
        return !(e.byteLength != r.byteLength || !s(new Xn(e), new Xn(r)));
      case Xc:
      case Qc:
      case rl:
        return kc(+e, +r);
      case Zc:
        return e.name == r.name && e.message == r.message;
      case tl:
      case ol:
        return e == r + "";
      case el:
        var p = Kc;
      case nl:
        var c = o & Wc;
        if (p || (p = $c), e.size != r.size && !c)
          return !1;
        var l = u.get(e);
        if (l)
          return l == r;
        o |= Yc, u.set(e, r);
        var f = Hc(p(e), p(r), o, a, s, u);
        return u.delete(e), f;
      case al:
        if (Ce)
          return Ce.call(e) == Ce.call(r);
    }
    return !1;
  }
  t(ul, "equalByTag");
  Zn.exports = ul;
});

// ../node_modules/lodash/_arrayPush.js
var to = i((hh, ro) => {
  function pl(e, r) {
    for (var n = -1, o = r.length, a = e.length; ++n < o; )
      e[a + n] = r[n];
    return e;
  }
  t(pl, "arrayPush");
  ro.exports = pl;
});

// ../node_modules/lodash/_baseGetAllKeys.js
var oo = i((xh, no) => {
  var cl = to(), ll = P();
  function fl(e, r, n) {
    var o = r(e);
    return ll(e) ? o : cl(o, n(e));
  }
  t(fl, "baseGetAllKeys");
  no.exports = fl;
});

// ../node_modules/lodash/_arrayFilter.js
var io = i((vh, ao) => {
  function ml(e, r) {
    for (var n = -1, o = e == null ? 0 : e.length, a = 0, s = []; ++n < o; ) {
      var u = e[n];
      r(u, n, e) && (s[a++] = u);
    }
    return s;
  }
  t(ml, "arrayFilter");
  ao.exports = ml;
});

// ../node_modules/lodash/stubArray.js
var uo = i((Ph, so) => {
  function gl() {
    return [];
  }
  t(gl, "stubArray");
  so.exports = gl;
});

// ../node_modules/lodash/_getSymbols.js
var lo = i((Sh, co) => {
  var yl = io(), dl = uo(), hl = Object.prototype, Tl = hl.propertyIsEnumerable, po = Object.getOwnPropertySymbols, xl = po ? function(e) {
    return e == null ? [] : (e = Object(e), yl(po(e), function(r) {
      return Tl.call(e, r);
    }));
  } : dl;
  co.exports = xl;
});

// ../node_modules/lodash/_getAllKeys.js
var mo = i((_h, fo) => {
  var bl = oo(), vl = lo(), Dl = ie();
  function Pl(e) {
    return bl(e, Dl, vl);
  }
  t(Pl, "getAllKeys");
  fo.exports = Pl;
});

// ../node_modules/lodash/_equalObjects.js
var ho = i((wh, yo) => {
  var go = mo(), ql = 1, Sl = Object.prototype, _l = Sl.hasOwnProperty;
  function Ol(e, r, n, o, a, s) {
    var u = n & ql, p = go(e), c = p.length, l = go(r), f = l.length;
    if (c != f && !u)
      return !1;
    for (var d = c; d--; ) {
      var g = p[d];
      if (!(u ? g in r : _l.call(r, g)))
        return !1;
    }
    var x = s.get(e), h = s.get(r);
    if (x && h)
      return x == r && h == e;
    var T = !0;
    s.set(e, r), s.set(r, e);
    for (var v = u; ++d < c; ) {
      g = p[d];
      var O = e[g], w = r[g];
      if (o)
        var pr = u ? o(w, O, g, r, e, s) : o(O, w, g, e, r, s);
      if (!(pr === void 0 ? O === w || a(O, w, n, o, s) : pr)) {
        T = !1;
        break;
      }
      v || (v = g == "constructor");
    }
    if (T && !v) {
      var te = e.constructor, ne = r.constructor;
      te != ne && "constructor" in e && "constructor" in r && !(typeof te == "function" && te instanceof te && typeof ne == "function" && ne instanceof
      ne) && (T = !1);
    }
    return s.delete(e), s.delete(r), T;
  }
  t(Ol, "equalObjects");
  yo.exports = Ol;
});

// ../node_modules/lodash/_DataView.js
var xo = i((Eh, To) => {
  var wl = S(), Al = D(), El = wl(Al, "DataView");
  To.exports = El;
});

// ../node_modules/lodash/_Promise.js
var vo = i((jh, bo) => {
  var jl = S(), Cl = D(), Il = jl(Cl, "Promise");
  bo.exports = Il;
});

// ../node_modules/lodash/_Set.js
var Po = i((Ch, Do) => {
  var Fl = S(), Nl = D(), Rl = Fl(Nl, "Set");
  Do.exports = Rl;
});

// ../node_modules/lodash/_WeakMap.js
var So = i((Ih, qo) => {
  var Ll = S(), Ml = D(), Jl = Ll(Ml, "WeakMap");
  qo.exports = Jl;
});

// ../node_modules/lodash/_getTag.js
var Io = i((Fh, Co) => {
  var Ie = xo(), Fe = se(), Ne = vo(), Re = Po(), Le = So(), jo = C(), U = Pe(), _o = "[object Map]", Ul = "[object Object]", Oo = "[object \
Promise]", wo = "[object Set]", Ao = "[object WeakMap]", Eo = "[object DataView]", Gl = U(Ie), Vl = U(Fe), Bl = U(Ne), zl = U(Re), kl = U(Le),
  A = jo;
  (Ie && A(new Ie(new ArrayBuffer(1))) != Eo || Fe && A(new Fe()) != _o || Ne && A(Ne.resolve()) != Oo || Re && A(new Re()) != wo || Le && A(
  new Le()) != Ao) && (A = /* @__PURE__ */ t(function(e) {
    var r = jo(e), n = r == Ul ? e.constructor : void 0, o = n ? U(n) : "";
    if (o)
      switch (o) {
        case Gl:
          return Eo;
        case Vl:
          return _o;
        case Bl:
          return Oo;
        case zl:
          return wo;
        case kl:
          return Ao;
      }
    return r;
  }, "getTag"));
  Co.exports = A;
});

// ../node_modules/lodash/_baseIsEqualDeep.js
var Go = i((Rh, Uo) => {
  var Me = Ee(), Hl = je(), Kl = eo(), $l = ho(), Fo = Io(), No = P(), Ro = Se(), Wl = we(), Yl = 1, Lo = "[object Arguments]", Mo = "[objec\
t Array]", ce = "[object Object]", Xl = Object.prototype, Jo = Xl.hasOwnProperty;
  function Ql(e, r, n, o, a, s) {
    var u = No(e), p = No(r), c = u ? Mo : Fo(e), l = p ? Mo : Fo(r);
    c = c == Lo ? ce : c, l = l == Lo ? ce : l;
    var f = c == ce, d = l == ce, g = c == l;
    if (g && Ro(e)) {
      if (!Ro(r))
        return !1;
      u = !0, f = !1;
    }
    if (g && !f)
      return s || (s = new Me()), u || Wl(e) ? Hl(e, r, n, o, a, s) : Kl(e, r, c, n, o, a, s);
    if (!(n & Yl)) {
      var x = f && Jo.call(e, "__wrapped__"), h = d && Jo.call(r, "__wrapped__");
      if (x || h) {
        var T = x ? e.value() : e, v = h ? r.value() : r;
        return s || (s = new Me()), a(T, v, n, o, s);
      }
    }
    return g ? (s || (s = new Me()), $l(e, r, n, o, a, s)) : !1;
  }
  t(Ql, "baseIsEqualDeep");
  Uo.exports = Ql;
});

// ../node_modules/lodash/_baseIsEqual.js
var Je = i((Mh, zo) => {
  var Zl = Go(), Vo = I();
  function Bo(e, r, n, o, a) {
    return e === r ? !0 : e == null || r == null || !Vo(e) && !Vo(r) ? e !== e && r !== r : Zl(e, r, n, o, Bo, a);
  }
  t(Bo, "baseIsEqual");
  zo.exports = Bo;
});

// ../node_modules/lodash/_baseIsMatch.js
var Ho = i((Uh, ko) => {
  var ef = Ee(), rf = Je(), tf = 1, nf = 2;
  function of(e, r, n, o) {
    var a = n.length, s = a, u = !o;
    if (e == null)
      return !s;
    for (e = Object(e); a--; ) {
      var p = n[a];
      if (u && p[2] ? p[1] !== e[p[0]] : !(p[0] in e))
        return !1;
    }
    for (; ++a < s; ) {
      p = n[a];
      var c = p[0], l = e[c], f = p[1];
      if (u && p[2]) {
        if (l === void 0 && !(c in e))
          return !1;
      } else {
        var d = new ef();
        if (o)
          var g = o(l, f, c, e, r, d);
        if (!(g === void 0 ? rf(f, l, tf | nf, o, d) : g))
          return !1;
      }
    }
    return !0;
  }
  t(of, "baseIsMatch");
  ko.exports = of;
});

// ../node_modules/lodash/_isStrictComparable.js
var Ue = i((Vh, Ko) => {
  var af = oe();
  function sf(e) {
    return e === e && !af(e);
  }
  t(sf, "isStrictComparable");
  Ko.exports = sf;
});

// ../node_modules/lodash/_getMatchData.js
var Wo = i((zh, $o) => {
  var uf = Ue(), pf = ie();
  function cf(e) {
    for (var r = pf(e), n = r.length; n--; ) {
      var o = r[n], a = e[o];
      r[n] = [o, a, uf(a)];
    }
    return r;
  }
  t(cf, "getMatchData");
  $o.exports = cf;
});

// ../node_modules/lodash/_matchesStrictComparable.js
var Ge = i((Hh, Yo) => {
  function lf(e, r) {
    return function(n) {
      return n == null ? !1 : n[e] === r && (r !== void 0 || e in Object(n));
    };
  }
  t(lf, "matchesStrictComparable");
  Yo.exports = lf;
});

// ../node_modules/lodash/_baseMatches.js
var Qo = i(($h, Xo) => {
  var ff = Ho(), mf = Wo(), gf = Ge();
  function yf(e) {
    var r = mf(e);
    return r.length == 1 && r[0][2] ? gf(r[0][0], r[0][1]) : function(n) {
      return n === e || ff(n, e, r);
    };
  }
  t(yf, "baseMatches");
  Xo.exports = yf;
});

// ../node_modules/lodash/isSymbol.js
var le = i((Yh, Zo) => {
  var df = C(), hf = I(), Tf = "[object Symbol]";
  function xf(e) {
    return typeof e == "symbol" || hf(e) && df(e) == Tf;
  }
  t(xf, "isSymbol");
  Zo.exports = xf;
});

// ../node_modules/lodash/_isKey.js
var fe = i((Qh, ea) => {
  var bf = P(), vf = le(), Df = /\.|\[(?:[^[\]]*|(["'])(?:(?!\1)[^\\]|\\.)*?\1)\]/, Pf = /^\w*$/;
  function qf(e, r) {
    if (bf(e))
      return !1;
    var n = typeof e;
    return n == "number" || n == "symbol" || n == "boolean" || e == null || vf(e) ? !0 : Pf.test(e) || !Df.test(e) || r != null && e in Object(
    r);
  }
  t(qf, "isKey");
  ea.exports = qf;
});

// ../node_modules/lodash/memoize.js
var na = i((eT, ta) => {
  var ra = ue(), Sf = "Expected a function";
  function Ve(e, r) {
    if (typeof e != "function" || r != null && typeof r != "function")
      throw new TypeError(Sf);
    var n = /* @__PURE__ */ t(function() {
      var o = arguments, a = r ? r.apply(this, o) : o[0], s = n.cache;
      if (s.has(a))
        return s.get(a);
      var u = e.apply(this, o);
      return n.cache = s.set(a, u) || s, u;
    }, "memoized");
    return n.cache = new (Ve.Cache || ra)(), n;
  }
  t(Ve, "memoize");
  Ve.Cache = ra;
  ta.exports = Ve;
});

// ../node_modules/lodash/_memoizeCapped.js
var aa = i((tT, oa) => {
  var _f = na(), Of = 500;
  function wf(e) {
    var r = _f(e, function(o) {
      return n.size === Of && n.clear(), o;
    }), n = r.cache;
    return r;
  }
  t(wf, "memoizeCapped");
  oa.exports = wf;
});

// ../node_modules/lodash/_stringToPath.js
var sa = i((oT, ia) => {
  var Af = aa(), Ef = /[^.[\]]+|\[(?:(-?\d+(?:\.\d+)?)|(["'])((?:(?!\2)[^\\]|\\.)*?)\2)\]|(?=(?:\.|\[\])(?:\.|\[\]|$))/g, jf = /\\(\\)?/g, Cf = Af(
  function(e) {
    var r = [];
    return e.charCodeAt(0) === 46 && r.push(""), e.replace(Ef, function(n, o, a, s) {
      r.push(a ? s.replace(jf, "$1") : o || n);
    }), r;
  });
  ia.exports = Cf;
});

// ../node_modules/lodash/_arrayMap.js
var pa = i((aT, ua) => {
  function If(e, r) {
    for (var n = -1, o = e == null ? 0 : e.length, a = Array(o); ++n < o; )
      a[n] = r(e[n], n, e);
    return a;
  }
  t(If, "arrayMap");
  ua.exports = If;
});

// ../node_modules/lodash/_baseToString.js
var ya = i((sT, ga) => {
  var ca = k(), Ff = pa(), Nf = P(), Rf = le(), Lf = 1 / 0, la = ca ? ca.prototype : void 0, fa = la ? la.toString : void 0;
  function ma(e) {
    if (typeof e == "string")
      return e;
    if (Nf(e))
      return Ff(e, ma) + "";
    if (Rf(e))
      return fa ? fa.call(e) : "";
    var r = e + "";
    return r == "0" && 1 / e == -Lf ? "-0" : r;
  }
  t(ma, "baseToString");
  ga.exports = ma;
});

// ../node_modules/lodash/toString.js
var ha = i((pT, da) => {
  var Mf = ya();
  function Jf(e) {
    return e == null ? "" : Mf(e);
  }
  t(Jf, "toString");
  da.exports = Jf;
});

// ../node_modules/lodash/_castPath.js
var Be = i((lT, Ta) => {
  var Uf = P(), Gf = fe(), Vf = sa(), Bf = ha();
  function zf(e, r) {
    return Uf(e) ? e : Gf(e, r) ? [e] : Vf(Bf(e));
  }
  t(zf, "castPath");
  Ta.exports = zf;
});

// ../node_modules/lodash/_toKey.js
var ee = i((mT, xa) => {
  var kf = le(), Hf = 1 / 0;
  function Kf(e) {
    if (typeof e == "string" || kf(e))
      return e;
    var r = e + "";
    return r == "0" && 1 / e == -Hf ? "-0" : r;
  }
  t(Kf, "toKey");
  xa.exports = Kf;
});

// ../node_modules/lodash/_baseGet.js
var ze = i((yT, ba) => {
  var $f = Be(), Wf = ee();
  function Yf(e, r) {
    r = $f(r, e);
    for (var n = 0, o = r.length; e != null && n < o; )
      e = e[Wf(r[n++])];
    return n && n == o ? e : void 0;
  }
  t(Yf, "baseGet");
  ba.exports = Yf;
});

// ../node_modules/lodash/get.js
var Da = i((hT, va) => {
  var Xf = ze();
  function Qf(e, r, n) {
    var o = e == null ? void 0 : Xf(e, r);
    return o === void 0 ? n : o;
  }
  t(Qf, "get");
  va.exports = Qf;
});

// ../node_modules/lodash/_baseHasIn.js
var qa = i((xT, Pa) => {
  function Zf(e, r) {
    return e != null && r in Object(e);
  }
  t(Zf, "baseHasIn");
  Pa.exports = Zf;
});

// ../node_modules/lodash/_hasPath.js
var _a = i((vT, Sa) => {
  var em = Be(), rm = qe(), tm = P(), nm = _e(), om = ae(), am = ee();
  function im(e, r, n) {
    r = em(r, e);
    for (var o = -1, a = r.length, s = !1; ++o < a; ) {
      var u = am(r[o]);
      if (!(s = e != null && n(e, u)))
        break;
      e = e[u];
    }
    return s || ++o != a ? s : (a = e == null ? 0 : e.length, !!a && om(a) && nm(u, a) && (tm(e) || rm(e)));
  }
  t(im, "hasPath");
  Sa.exports = im;
});

// ../node_modules/lodash/hasIn.js
var wa = i((PT, Oa) => {
  var sm = qa(), um = _a();
  function pm(e, r) {
    return e != null && um(e, r, sm);
  }
  t(pm, "hasIn");
  Oa.exports = pm;
});

// ../node_modules/lodash/_baseMatchesProperty.js
var Ea = i((ST, Aa) => {
  var cm = Je(), lm = Da(), fm = wa(), mm = fe(), gm = Ue(), ym = Ge(), dm = ee(), hm = 1, Tm = 2;
  function xm(e, r) {
    return mm(e) && gm(r) ? ym(dm(e), r) : function(n) {
      var o = lm(n, e);
      return o === void 0 && o === r ? fm(n, e) : cm(r, o, hm | Tm);
    };
  }
  t(xm, "baseMatchesProperty");
  Aa.exports = xm;
});

// ../node_modules/lodash/identity.js
var Ca = i((OT, ja) => {
  function bm(e) {
    return e;
  }
  t(bm, "identity");
  ja.exports = bm;
});

// ../node_modules/lodash/_baseProperty.js
var Fa = i((AT, Ia) => {
  function vm(e) {
    return function(r) {
      return r?.[e];
    };
  }
  t(vm, "baseProperty");
  Ia.exports = vm;
});

// ../node_modules/lodash/_basePropertyDeep.js
var Ra = i((jT, Na) => {
  var Dm = ze();
  function Pm(e) {
    return function(r) {
      return Dm(r, e);
    };
  }
  t(Pm, "basePropertyDeep");
  Na.exports = Pm;
});

// ../node_modules/lodash/property.js
var Ma = i((IT, La) => {
  var qm = Fa(), Sm = Ra(), _m = fe(), Om = ee();
  function wm(e) {
    return _m(e) ? qm(Om(e)) : Sm(e);
  }
  t(wm, "property");
  La.exports = wm;
});

// ../node_modules/lodash/_baseIteratee.js
var Ua = i((NT, Ja) => {
  var Am = Qo(), Em = Ea(), jm = Ca(), Cm = P(), Im = Ma();
  function Fm(e) {
    return typeof e == "function" ? e : e == null ? jm : typeof e == "object" ? Cm(e) ? Em(e[0], e[1]) : Am(e) : Im(e);
  }
  t(Fm, "baseIteratee");
  Ja.exports = Fm;
});

// ../node_modules/lodash/mapValues.js
var Va = i((LT, Ga) => {
  var Nm = Gr(), Rm = jt(), Lm = Ua();
  function Mm(e, r) {
    var n = {};
    return r = Lm(r, 3), Rm(e, function(o, a, s) {
      Nm(n, a, r(o, a, s));
    }), n;
  }
  t(Mm, "mapValues");
  Ga.exports = Mm;
});

// src/docs-tools/index.ts
var Fg = {};
bi(Fg, {
  ADDON_ID: () => ur,
  MAX_DEFAULT_VALUE_SUMMARY_LENGTH: () => lg,
  MAX_TYPE_SUMMARY_LENGTH: () => cg,
  PANEL_ID: () => Ag,
  PARAM_KEY: () => Eg,
  SNIPPET_RENDERED: () => jg,
  SourceType: () => gi,
  TypeSystem: () => Ha,
  convert: () => re,
  createSummaryValue: () => y,
  enhanceArgTypes: () => wg,
  extractComponentDescription: () => Og,
  extractComponentProps: () => _g,
  extractComponentSectionArray: () => ci,
  extractComponentSectionObject: () => li,
  getDocgenDescription: () => We,
  getDocgenSection: () => $e,
  hasDocgen: () => He,
  hasDocsOrControls: () => Ig,
  isDefaultValueBlacklisted: () => E,
  isTooLongForDefaultValueSummary: () => ir,
  isTooLongForTypeSummary: () => Te,
  isValidDocgenSection: () => Ke,
  normalizeNewlines: () => fg,
  parseJsDoc: () => ar,
  str: () => ke
});
module.exports = Di(Fg);

// src/docs-tools/argTypes/convert/flow/convert.ts
var lr = require("@storybook/core/preview-errors");
var Pi = /* @__PURE__ */ t((e) => e.name === "literal", "isLiteral"), qi = /* @__PURE__ */ t((e) => e.value.replace(/['|"]/g, ""), "toEnumOp\
tion"), Si = /* @__PURE__ */ t((e) => {
  switch (e.type) {
    case "function":
      return { name: "function" };
    case "object":
      let r = {};
      return e.signature.properties.forEach((n) => {
        r[n.key] = j(n.value);
      }), {
        name: "object",
        value: r
      };
    default:
      throw new lr.UnknownArgTypesError({ type: e, language: "Flow" });
  }
}, "convertSig"), j = /* @__PURE__ */ t((e) => {
  let { name: r, raw: n } = e, o = {};
  switch (typeof n < "u" && (o.raw = n), e.name) {
    case "literal":
      return { ...o, name: "other", value: e.value };
    case "string":
    case "number":
    case "symbol":
    case "boolean":
      return { ...o, name: r };
    case "Array":
      return { ...o, name: "array", value: e.elements.map(j) };
    case "signature":
      return { ...o, ...Si(e) };
    case "union":
      return e.elements?.every(Pi) ? { ...o, name: "enum", value: e.elements?.map(qi) } : { ...o, name: r, value: e.elements?.map(j) };
    case "intersection":
      return { ...o, name: r, value: e.elements?.map(j) };
    default:
      return { ...o, name: "other", value: r };
  }
}, "convert");

// src/docs-tools/argTypes/convert/proptypes/convert.ts
var za = vi(Va(), 1);

// src/docs-tools/argTypes/convert/utils.ts
var Ba = /^['"]|['"]$/g, Jm = /* @__PURE__ */ t((e) => e.replace(Ba, ""), "trimQuotes"), Um = /* @__PURE__ */ t((e) => Ba.test(e), "includes\
Quotes"), me = /* @__PURE__ */ t((e) => {
  let r = Jm(e);
  return Um(e) || Number.isNaN(Number(r)) ? r : Number(r);
}, "parseLiteral");

// src/docs-tools/argTypes/convert/proptypes/convert.ts
var Gm = /^\(.*\) => /, G = /* @__PURE__ */ t((e) => {
  let { name: r, raw: n, computed: o, value: a } = e, s = {};
  switch (typeof n < "u" && (s.raw = n), r) {
    case "enum": {
      let p = o ? a : a.map((c) => me(c.value));
      return { ...s, name: r, value: p };
    }
    case "string":
    case "number":
    case "symbol":
      return { ...s, name: r };
    case "func":
      return { ...s, name: "function" };
    case "bool":
    case "boolean":
      return { ...s, name: "boolean" };
    case "arrayOf":
    case "array":
      return { ...s, name: "array", value: a && G(a) };
    case "object":
      return { ...s, name: r };
    case "objectOf":
      return { ...s, name: r, value: G(a) };
    case "shape":
    case "exact":
      let u = (0, za.default)(a, (p) => G(p));
      return { ...s, name: "object", value: u };
    case "union":
      return { ...s, name: "union", value: a.map((p) => G(p)) };
    case "instanceOf":
    case "element":
    case "elementType":
    default: {
      if (r?.indexOf("|") > 0)
        try {
          let l = r.split("|").map((f) => JSON.parse(f));
          return { ...s, name: "enum", value: l };
        } catch {
        }
      let p = a ? `${r}(${a})` : r, c = Gm.test(r) ? "function" : "other";
      return { ...s, name: c, value: p };
    }
  }
}, "convert");

// src/docs-tools/argTypes/convert/typescript/convert.ts
var ka = require("@storybook/core/preview-errors");
var Vm = /* @__PURE__ */ t((e) => {
  switch (e.type) {
    case "function":
      return { name: "function" };
    case "object":
      let r = {};
      return e.signature.properties.forEach((n) => {
        r[n.key] = V(n.value);
      }), {
        name: "object",
        value: r
      };
    default:
      throw new ka.UnknownArgTypesError({ type: e, language: "Typescript" });
  }
}, "convertSig"), V = /* @__PURE__ */ t((e) => {
  let { name: r, raw: n } = e, o = {};
  switch (typeof n < "u" && (o.raw = n), e.name) {
    case "string":
    case "number":
    case "symbol":
    case "boolean":
      return { ...o, name: r };
    case "Array":
      return { ...o, name: "array", value: e.elements.map(V) };
    case "signature":
      return { ...o, ...Vm(e) };
    case "union":
      let a;
      return e.elements?.every((s) => s.name === "literal") ? a = {
        ...o,
        name: "enum",
        // @ts-expect-error fix types
        value: e.elements?.map((s) => me(s.value))
      } : a = { ...o, name: r, value: e.elements?.map(V) }, a;
    case "intersection":
      return { ...o, name: r, value: e.elements?.map(V) };
    default:
      return { ...o, name: "other", value: r };
  }
}, "convert");

// src/docs-tools/argTypes/convert/index.ts
var re = /* @__PURE__ */ t((e) => {
  let { type: r, tsType: n, flowType: o } = e;
  try {
    if (r != null)
      return G(r);
    if (n != null)
      return V(n);
    if (o != null)
      return j(o);
  } catch (a) {
    console.error(a);
  }
  return null;
}, "convert");

// src/docs-tools/argTypes/docgen/types.ts
var Ha = /* @__PURE__ */ ((a) => (a.JAVASCRIPT = "JavaScript", a.FLOW = "Flow", a.TYPESCRIPT = "TypeScript", a.UNKNOWN = "Unknown", a))(Ha ||
{});

// src/docs-tools/argTypes/docgen/utils/defaultValue.ts
var Bm = ["null", "undefined"];
function E(e) {
  return Bm.some((r) => r === e);
}
t(E, "isDefaultValueBlacklisted");

// src/docs-tools/argTypes/docgen/utils/string.ts
var ke = /* @__PURE__ */ t((e) => {
  if (!e)
    return "";
  if (typeof e == "string")
    return e;
  throw new Error(`Description: expected string, got: ${JSON.stringify(e)}`);
}, "str");

// src/docs-tools/argTypes/docgen/utils/docgenInfo.ts
function He(e) {
  return !!e.__docgenInfo;
}
t(He, "hasDocgen");
function Ke(e) {
  return e != null && Object.keys(e).length > 0;
}
t(Ke, "isValidDocgenSection");
function $e(e, r) {
  return He(e) ? e.__docgenInfo[r] : null;
}
t($e, "getDocgenSection");
function We(e) {
  return He(e) ? ke(e.__docgenInfo.description) : "";
}
t(We, "getDocgenDescription");

// ../node_modules/comment-parser/es6/primitives.js
var b;
(function(e) {
  e.start = "/**", e.nostart = "/***", e.delim = "*", e.end = "*/";
})(b = b || (b = {}));

// ../node_modules/comment-parser/es6/util.js
function Ye(e) {
  return /^\s+$/.test(e);
}
t(Ye, "isSpace");
function Ka(e) {
  let r = e.match(/\r+$/);
  return r == null ? ["", e] : [e.slice(-r[0].length), e.slice(0, -r[0].length)];
}
t(Ka, "splitCR");
function q(e) {
  let r = e.match(/^\s+/);
  return r == null ? ["", e] : [e.slice(0, r[0].length), e.slice(r[0].length)];
}
t(q, "splitSpace");
function $a(e) {
  return e.split(/\n/);
}
t($a, "splitLines");
function Wa(e = {}) {
  return Object.assign({ tag: "", name: "", type: "", optional: !1, description: "", problems: [], source: [] }, e);
}
t(Wa, "seedSpec");
function Xe(e = {}) {
  return Object.assign({ start: "", delimiter: "", postDelimiter: "", tag: "", postTag: "", name: "", postName: "", type: "", postType: "", description: "",
  end: "", lineEnd: "" }, e);
}
t(Xe, "seedTokens");

// ../node_modules/comment-parser/es6/parser/block-parser.js
var zm = /^@\S+/;
function Qe({ fence: e = "```" } = {}) {
  let r = km(e), n = /* @__PURE__ */ t((o, a) => r(o) ? !a : a, "toggleFence");
  return /* @__PURE__ */ t(function(a) {
    let s = [[]], u = !1;
    for (let p of a)
      zm.test(p.tokens.description) && !u ? s.push([p]) : s[s.length - 1].push(p), u = n(p.tokens.description, u);
    return s;
  }, "parseBlock");
}
t(Qe, "getParser");
function km(e) {
  return typeof e == "string" ? (r) => r.split(e).length % 2 === 0 : e;
}
t(km, "getFencer");

// ../node_modules/comment-parser/es6/parser/source-parser.js
function Ze({ startLine: e = 0, markers: r = b } = {}) {
  let n = null, o = e;
  return /* @__PURE__ */ t(function(s) {
    let u = s, p = Xe();
    if ([p.lineEnd, u] = Ka(u), [p.start, u] = q(u), n === null && u.startsWith(r.start) && !u.startsWith(r.nostart) && (n = [], p.delimiter =
    u.slice(0, r.start.length), u = u.slice(r.start.length), [p.postDelimiter, u] = q(u)), n === null)
      return o++, null;
    let c = u.trimRight().endsWith(r.end);
    if (p.delimiter === "" && u.startsWith(r.delim) && !u.startsWith(r.end) && (p.delimiter = r.delim, u = u.slice(r.delim.length), [p.postDelimiter,
    u] = q(u)), c) {
      let l = u.trimRight();
      p.end = u.slice(l.length - r.end.length), u = l.slice(0, -r.end.length);
    }
    if (p.description = u, n.push({ number: o, source: s, tokens: p }), o++, c) {
      let l = n.slice();
      return n = null, l;
    }
    return null;
  }, "parseSource");
}
t(Ze, "getParser");

// ../node_modules/comment-parser/es6/parser/spec-parser.js
function er({ tokenizers: e }) {
  return /* @__PURE__ */ t(function(n) {
    var o;
    let a = Wa({ source: n });
    for (let s of e)
      if (a = s(a), !((o = a.problems[a.problems.length - 1]) === null || o === void 0) && o.critical)
        break;
    return a;
  }, "parseSpec");
}
t(er, "getParser");

// ../node_modules/comment-parser/es6/parser/tokenizers/tag.js
function ge() {
  return (e) => {
    let { tokens: r } = e.source[0], n = r.description.match(/\s*(@(\S+))(\s*)/);
    return n === null ? (e.problems.push({
      code: "spec:tag:prefix",
      message: 'tag should start with "@" symbol',
      line: e.source[0].number,
      critical: !0
    }), e) : (r.tag = n[1], r.postTag = n[3], r.description = r.description.slice(n[0].length), e.tag = n[2], e);
  };
}
t(ge, "tagTokenizer");

// ../node_modules/comment-parser/es6/parser/tokenizers/type.js
function ye(e = "compact") {
  let r = Km(e);
  return (n) => {
    let o = 0, a = [];
    for (let [p, { tokens: c }] of n.source.entries()) {
      let l = "";
      if (p === 0 && c.description[0] !== "{")
        return n;
      for (let f of c.description)
        if (f === "{" && o++, f === "}" && o--, l += f, o === 0)
          break;
      if (a.push([c, l]), o === 0)
        break;
    }
    if (o !== 0)
      return n.problems.push({
        code: "spec:type:unpaired-curlies",
        message: "unpaired curlies",
        line: n.source[0].number,
        critical: !0
      }), n;
    let s = [], u = a[0][0].postDelimiter.length;
    for (let [p, [c, l]] of a.entries())
      c.type = l, p > 0 && (c.type = c.postDelimiter.slice(u) + l, c.postDelimiter = c.postDelimiter.slice(0, u)), [c.postType, c.description] =
      q(c.description.slice(l.length)), s.push(c.type);
    return s[0] = s[0].slice(1), s[s.length - 1] = s[s.length - 1].slice(0, -1), n.type = r(s), n;
  };
}
t(ye, "typeTokenizer");
var Hm = /* @__PURE__ */ t((e) => e.trim(), "trim");
function Km(e) {
  return e === "compact" ? (r) => r.map(Hm).join("") : e === "preserve" ? (r) => r.join(`
`) : e;
}
t(Km, "getJoiner");

// ../node_modules/comment-parser/es6/parser/tokenizers/name.js
var $m = /* @__PURE__ */ t((e) => e && e.startsWith('"') && e.endsWith('"'), "isQuoted");
function de() {
  let e = /* @__PURE__ */ t((r, { tokens: n }, o) => n.type === "" ? r : o, "typeEnd");
  return (r) => {
    let { tokens: n } = r.source[r.source.reduce(e, 0)], o = n.description.trimLeft(), a = o.split('"');
    if (a.length > 1 && a[0] === "" && a.length % 2 === 1)
      return r.name = a[1], n.name = `"${a[1]}"`, [n.postName, n.description] = q(o.slice(n.name.length)), r;
    let s = 0, u = "", p = !1, c;
    for (let f of o) {
      if (s === 0 && Ye(f))
        break;
      f === "[" && s++, f === "]" && s--, u += f;
    }
    if (s !== 0)
      return r.problems.push({
        code: "spec:name:unpaired-brackets",
        message: "unpaired brackets",
        line: r.source[0].number,
        critical: !0
      }), r;
    let l = u;
    if (u[0] === "[" && u[u.length - 1] === "]") {
      p = !0, u = u.slice(1, -1);
      let f = u.split("=");
      if (u = f[0].trim(), f[1] !== void 0 && (c = f.slice(1).join("=").trim()), u === "")
        return r.problems.push({
          code: "spec:name:empty-name",
          message: "empty name",
          line: r.source[0].number,
          critical: !0
        }), r;
      if (c === "")
        return r.problems.push({
          code: "spec:name:empty-default",
          message: "empty default value",
          line: r.source[0].number,
          critical: !0
        }), r;
      if (!$m(c) && /=(?!>)/.test(c))
        return r.problems.push({
          code: "spec:name:invalid-default",
          message: "invalid default value syntax",
          line: r.source[0].number,
          critical: !0
        }), r;
    }
    return r.optional = p, r.name = u, n.name = l, c !== void 0 && (r.default = c), [n.postName, n.description] = q(o.slice(n.name.length)),
    r;
  };
}
t(de, "nameTokenizer");

// ../node_modules/comment-parser/es6/parser/tokenizers/description.js
function he(e = "compact", r = b) {
  let n = rr(e);
  return (o) => (o.description = n(o.source, r), o);
}
t(he, "descriptionTokenizer");
function rr(e) {
  return e === "compact" ? Wm : e === "preserve" ? Qm : e;
}
t(rr, "getJoiner");
function Wm(e, r = b) {
  return e.map(({ tokens: { description: n } }) => n.trim()).filter((n) => n !== "").join(" ");
}
t(Wm, "compactJoiner");
var Ym = /* @__PURE__ */ t((e, { tokens: r }, n) => r.type === "" ? e : n, "lineNo"), Xm = /* @__PURE__ */ t(({ tokens: e }) => (e.delimiter ===
"" ? e.start : e.postDelimiter.slice(1)) + e.description, "getDescription");
function Qm(e, r = b) {
  if (e.length === 0)
    return "";
  e[0].tokens.description === "" && e[0].tokens.delimiter === r.start && (e = e.slice(1));
  let n = e[e.length - 1];
  return n !== void 0 && n.tokens.description === "" && n.tokens.end.endsWith(r.end) && (e = e.slice(0, -1)), e = e.slice(e.reduce(Ym, 0)), e.
  map(Xm).join(`
`);
}
t(Qm, "preserveJoiner");

// ../node_modules/comment-parser/es6/parser/index.js
function tr({ startLine: e = 0, fence: r = "```", spacing: n = "compact", markers: o = b, tokenizers: a = [
  ge(),
  ye(n),
  de(),
  he(n)
] } = {}) {
  if (e < 0 || e % 1 > 0)
    throw new Error("Invalid startLine");
  let s = Ze({ startLine: e, markers: o }), u = Qe({ fence: r }), p = er({ tokenizers: a }), c = rr(n);
  return function(l) {
    let f = [];
    for (let d of $a(l)) {
      let g = s(d);
      if (g === null)
        continue;
      let x = u(g), h = x.slice(1).map(p);
      f.push({
        description: c(x[0], o),
        tags: h,
        source: g,
        problems: h.reduce((T, v) => T.concat(v.problems), [])
      });
    }
    return f;
  };
}
t(tr, "getParser");

// ../node_modules/comment-parser/es6/stringifier/index.js
function Zm(e) {
  return e.start + e.delimiter + e.postDelimiter + e.tag + e.postTag + e.type + e.postType + e.name + e.postName + e.description + e.end + e.
  lineEnd;
}
t(Zm, "join");
function nr() {
  return (e) => e.source.map(({ tokens: r }) => Zm(r)).join(`
`);
}
t(nr, "getStringifier");

// ../node_modules/comment-parser/es6/stringifier/inspect.js
var eg = {
  line: 0,
  start: 0,
  delimiter: 0,
  postDelimiter: 0,
  tag: 0,
  postTag: 0,
  name: 0,
  postName: 0,
  type: 0,
  postType: 0,
  description: 0,
  end: 0,
  lineEnd: 0
};
var rb = Object.keys(eg);

// ../node_modules/comment-parser/es6/index.js
function Ya(e, r = {}) {
  return tr(r)(e);
}
t(Ya, "parse");
var Db = nr();

// src/docs-tools/argTypes/jsdocParser.ts
var B = require("jsdoc-type-pratt-parser");
function rg(e) {
  return e != null && e.includes("@");
}
t(rg, "containsJsDoc");
function tg(e) {
  let o = `/**
` + (e ?? "").split(`
`).map((s) => ` * ${s}`).join(`
`) + `
*/`, a = Ya(o, {
    spacing: "preserve"
  });
  if (!a || a.length === 0)
    throw new Error("Cannot parse JSDoc tags.");
  return a[0];
}
t(tg, "parse");
var ng = {
  tags: ["param", "arg", "argument", "returns", "ignore", "deprecated"]
}, ar = /* @__PURE__ */ t((e, r = ng) => {
  if (!rg(e))
    return {
      includesJsDoc: !1,
      ignore: !1
    };
  let n = tg(e), o = og(n, r.tags);
  return o.ignore ? {
    includesJsDoc: !0,
    ignore: !0
  } : {
    includesJsDoc: !0,
    ignore: !1,
    // Always use the parsed description to ensure JSDoc is removed from the description.
    description: n.description.trim(),
    extractedTags: o
  };
}, "parseJsDoc");
function og(e, r) {
  let n = {
    params: null,
    deprecated: null,
    returns: null,
    ignore: !1
  };
  for (let o of e.tags)
    if (!(r !== void 0 && !r.includes(o.tag)))
      if (o.tag === "ignore") {
        n.ignore = !0;
        break;
      } else
        switch (o.tag) {
          case "param":
          case "arg":
          case "argument": {
            let a = ig(o);
            a != null && (n.params == null && (n.params = []), n.params.push(a));
            break;
          }
          case "deprecated": {
            let a = sg(o);
            a != null && (n.deprecated = a);
            break;
          }
          case "returns": {
            let a = ug(o);
            a != null && (n.returns = a);
            break;
          }
          default:
            break;
        }
  return n;
}
t(og, "extractJsDocTags");
function ag(e) {
  return e.replace(/[\.-]$/, "");
}
t(ag, "normaliseParamName");
function ig(e) {
  if (!e.name || e.name === "-")
    return null;
  let r = Za(e.type);
  return {
    name: e.name,
    type: r,
    description: Qa(e.description),
    getPrettyName: /* @__PURE__ */ t(() => ag(e.name), "getPrettyName"),
    getTypeName: /* @__PURE__ */ t(() => r ? ei(r) : null, "getTypeName")
  };
}
t(ig, "extractParam");
function sg(e) {
  return e.name ? Xa(e.name, e.description) : null;
}
t(sg, "extractDeprecated");
function Xa(e, r) {
  let n = e === "" ? r : `${e} ${r}`;
  return Qa(n);
}
t(Xa, "joinNameAndDescription");
function Qa(e) {
  let r = e.replace(/^- /g, "").trim();
  return r === "" ? null : r;
}
t(Qa, "normaliseDescription");
function ug(e) {
  let r = Za(e.type);
  return r ? {
    type: r,
    description: Xa(e.name, e.description),
    getTypeName: /* @__PURE__ */ t(() => ei(r), "getTypeName")
  } : null;
}
t(ug, "extractReturns");
var _ = (0, B.stringifyRules)(), pg = _.JsdocTypeObject;
_.JsdocTypeAny = () => "any";
_.JsdocTypeObject = (e, r) => `(${pg(e, r)})`;
_.JsdocTypeOptional = (e, r) => r(e.element);
_.JsdocTypeNullable = (e, r) => r(e.element);
_.JsdocTypeNotNullable = (e, r) => r(e.element);
_.JsdocTypeUnion = (e, r) => e.elements.map(r).join("|");
function Za(e) {
  try {
    return (0, B.parse)(e, "typescript");
  } catch {
    return null;
  }
}
t(Za, "extractType");
function ei(e) {
  return (0, B.transform)(_, e);
}
t(ei, "extractTypeName");

// src/docs-tools/argTypes/utils.ts
var cg = 90, lg = 50;
function Te(e) {
  return e.length > 90;
}
t(Te, "isTooLongForTypeSummary");
function ir(e) {
  return e.length > 50;
}
t(ir, "isTooLongForDefaultValueSummary");
function y(e, r) {
  return e === r ? { summary: e } : { summary: e, detail: r };
}
t(y, "createSummaryValue");
var fg = /* @__PURE__ */ t((e) => e.replace(/\\r\\n/g, "\\n"), "normalizeNewlines");

// src/docs-tools/argTypes/docgen/flow/createDefaultValue.ts
function ri(e, r) {
  if (e != null) {
    let { value: n } = e;
    if (!E(n))
      return ir(n) ? y(r?.name, n) : y(n);
  }
  return null;
}
t(ri, "createDefaultValue");

// src/docs-tools/argTypes/docgen/flow/createType.ts
function ti({ name: e, value: r, elements: n, raw: o }) {
  return r ?? (n != null ? n.map(ti).join(" | ") : o ?? e);
}
t(ti, "generateUnionElement");
function mg({ name: e, raw: r, elements: n }) {
  return n != null ? y(n.map(ti).join(" | ")) : r != null ? y(r.replace(/^\|\s*/, "")) : y(e);
}
t(mg, "generateUnion");
function gg({ type: e, raw: r }) {
  return r != null ? y(r) : y(e);
}
t(gg, "generateFuncSignature");
function yg({ type: e, raw: r }) {
  return r != null ? Te(r) ? y(e, r) : y(r) : y(e);
}
t(yg, "generateObjectSignature");
function dg(e) {
  let { type: r } = e;
  return r === "object" ? yg(e) : gg(e);
}
t(dg, "generateSignature");
function hg({ name: e, raw: r }) {
  return r != null ? Te(r) ? y(e, r) : y(r) : y(e);
}
t(hg, "generateDefault");
function ni(e) {
  if (e == null)
    return null;
  switch (e.name) {
    case "union":
      return mg(e);
    case "signature":
      return dg(e);
    default:
      return hg(e);
  }
}
t(ni, "createType");

// src/docs-tools/argTypes/docgen/flow/createPropDef.ts
var oi = /* @__PURE__ */ t((e, r) => {
  let { flowType: n, description: o, required: a, defaultValue: s } = r;
  return {
    name: e,
    type: ni(n),
    required: a,
    description: o,
    defaultValue: ri(s ?? null, n ?? null)
  };
}, "createFlowPropDef");

// src/docs-tools/argTypes/docgen/typeScript/createDefaultValue.ts
function ai({ defaultValue: e }) {
  if (e != null) {
    let { value: r } = e;
    if (!E(r))
      return y(r);
  }
  return null;
}
t(ai, "createDefaultValue");

// src/docs-tools/argTypes/docgen/typeScript/createType.ts
function ii({ tsType: e, required: r }) {
  if (e == null)
    return null;
  let n = e.name;
  return r || (n = n.replace(" | undefined", "")), y(
    ["Array", "Record", "signature"].includes(e.name) ? e.raw : n
  );
}
t(ii, "createType");

// src/docs-tools/argTypes/docgen/typeScript/createPropDef.ts
var si = /* @__PURE__ */ t((e, r) => {
  let { description: n, required: o } = r;
  return {
    name: e,
    type: ii(r),
    required: o,
    description: n,
    defaultValue: ai(r)
  };
}, "createTsPropDef");

// src/docs-tools/argTypes/docgen/createPropDef.ts
function Tg(e) {
  return e != null ? y(e.name) : null;
}
t(Tg, "createType");
function xg(e) {
  let { computed: r, func: n } = e;
  return typeof r > "u" && typeof n > "u";
}
t(xg, "isReactDocgenTypescript");
function bg(e) {
  return e ? e.name === "string" ? !0 : e.name === "enum" ? Array.isArray(e.value) && e.value.every(
    ({ value: r }) => typeof r == "string" && r[0] === '"' && r[r.length - 1] === '"'
  ) : !1 : !1;
}
t(bg, "isStringValued");
function vg(e, r) {
  if (e != null) {
    let { value: n } = e;
    if (!E(n))
      return xg(e) && bg(r) ? y(JSON.stringify(n)) : y(n);
  }
  return null;
}
t(vg, "createDefaultValue");
function ui(e, r, n) {
  let { description: o, required: a, defaultValue: s } = n;
  return {
    name: e,
    type: Tg(r),
    required: a,
    description: o,
    defaultValue: vg(s, r)
  };
}
t(ui, "createBasicPropDef");
function xe(e, r) {
  if (r?.includesJsDoc) {
    let { description: n, extractedTags: o } = r;
    n != null && (e.description = r.description);
    let a = {
      ...o,
      params: o?.params?.map(
        (s) => ({
          name: s.getPrettyName(),
          description: s.description
        })
      )
    };
    Object.values(a).filter(Boolean).length > 0 && (e.jsDocTags = a);
  }
  return e;
}
t(xe, "applyJsDocResult");
var Dg = /* @__PURE__ */ t((e, r, n) => {
  let o = ui(e, r.type, r);
  return o.sbType = re(r), xe(o, n);
}, "javaScriptFactory"), Pg = /* @__PURE__ */ t((e, r, n) => {
  let o = si(e, r);
  return o.sbType = re(r), xe(o, n);
}, "tsFactory"), qg = /* @__PURE__ */ t((e, r, n) => {
  let o = oi(e, r);
  return o.sbType = re(r), xe(o, n);
}, "flowFactory"), Sg = /* @__PURE__ */ t((e, r, n) => {
  let o = ui(e, { name: "unknown" }, r);
  return xe(o, n);
}, "unknownFactory"), sr = /* @__PURE__ */ t((e) => {
  switch (e) {
    case "JavaScript":
      return Dg;
    case "TypeScript":
      return Pg;
    case "Flow":
      return qg;
    default:
      return Sg;
  }
}, "getPropDefFactory");

// src/docs-tools/argTypes/docgen/extractDocgenProps.ts
var pi = /* @__PURE__ */ t((e) => e.type != null ? "JavaScript" : e.flowType != null ? "Flow" : e.tsType != null ? "TypeScript" : "Unknown",
"getTypeSystem"), ci = /* @__PURE__ */ t((e) => {
  let r = pi(e[0]), n = sr(r);
  return e.map((o) => {
    let a = o;
    return o.type?.elements && (a = {
      ...o,
      type: {
        ...o.type,
        value: o.type.elements
      }
    }), fi(a.name, a, r, n);
  });
}, "extractComponentSectionArray"), li = /* @__PURE__ */ t((e) => {
  let r = Object.keys(e), n = pi(e[r[0]]), o = sr(n);
  return r.map((a) => {
    let s = e[a];
    return s != null ? fi(a, s, n, o) : null;
  }).filter(Boolean);
}, "extractComponentSectionObject"), _g = /* @__PURE__ */ t((e, r) => {
  let n = $e(e, r);
  return Ke(n) ? Array.isArray(n) ? ci(n) : li(n) : [];
}, "extractComponentProps");
function fi(e, r, n, o) {
  let a = ar(r.description);
  return a.includesJsDoc && a.ignore ? null : {
    propDef: o(e, r, a),
    jsDocTags: a.extractedTags,
    docgenInfo: r,
    typeSystem: n
  };
}
t(fi, "extractProp");
function Og(e) {
  return e != null ? We(e) : "";
}
t(Og, "extractComponentDescription");

// src/docs-tools/argTypes/enhanceArgTypes.ts
var mi = require("@storybook/core/preview-api");
var wg = /* @__PURE__ */ t((e) => {
  let {
    component: r,
    argTypes: n,
    parameters: { docs: o = {} }
  } = e, { extractArgTypes: a } = o, s = a && r ? a(r) : {};
  return s ? (0, mi.combineParameters)(s, n) : n;
}, "enhanceArgTypes");

// src/docs-tools/shared.ts
var ur = "storybook/docs", Ag = `${ur}/panel`, Eg = "docs", jg = `${ur}/snippet-rendered`, gi = /* @__PURE__ */ ((o) => (o.AUTO = "auto", o.
CODE = "code", o.DYNAMIC = "dynamic", o))(gi || {});

// src/docs-tools/hasDocsOrControls.ts
var Cg = /(addons\/|addon-|addon-essentials\/)(docs|controls)/, Ig = /* @__PURE__ */ t((e) => e.presetsList?.some((r) => Cg.test(r.name)), "\
hasDocsOrControls");
