// @storybook/core/dist/router/index.cjs
"use strict";
var uo = Object.create;
var Be = Object.defineProperty;
var lo = Object.getOwnPropertyDescriptor;
var co = Object.getOwnPropertyNames;
var so = Object.getPrototypeOf, fo = Object.prototype.hasOwnProperty;
var r = (t, e) => Be(t, "name", { value: e, configurable: !0 });
var V = (t, e) => () => (e || t((e = { exports: {} }).exports, e), e.exports), po = (t, e) => {
  for (var a in e)
    Be(t, a, { get: e[a], enumerable: !0 });
}, lr = (t, e, a, n) => {
  if (e && typeof e == "object" || typeof e == "function")
    for (let o of co(e))
      !fo.call(t, o) && o !== a && Be(t, o, { get: () => e[o], enumerable: !(n = lo(e, o)) || n.enumerable });
  return t;
};
var Re = (t, e, a) => (a = t != null ? uo(so(t)) : {}, lr(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  e || !t || !t.__esModule ? Be(a, "default", { value: t, enumerable: !0 }) : a,
  t
)), ho = (t) => lr(Be({}, "__esModule", { value: !0 }), t);

// ../node_modules/lodash/_freeGlobal.js
var pr = V((hl, fr) => {
  var yo = typeof global == "object" && global && global.Object === Object && global;
  fr.exports = yo;
});

// ../node_modules/lodash/_root.js
var yr = V((yl, hr) => {
  var mo = pr(), go = typeof self == "object" && self && self.Object === Object && self, vo = mo || go || Function("return this")();
  hr.exports = vo;
});

// ../node_modules/lodash/_Symbol.js
var Et = V((dl, dr) => {
  var bo = yr(), Po = bo.Symbol;
  dr.exports = Po;
});

// ../node_modules/lodash/_getRawTag.js
var br = V((ml, vr) => {
  var mr = Et(), gr = Object.prototype, wo = gr.hasOwnProperty, So = gr.toString, We = mr ? mr.toStringTag : void 0;
  function Oo(t) {
    var e = wo.call(t, We), a = t[We];
    try {
      t[We] = void 0;
      var n = !0;
    } catch {
    }
    var o = So.call(t);
    return n && (e ? t[We] = a : delete t[We]), o;
  }
  r(Oo, "getRawTag");
  vr.exports = Oo;
});

// ../node_modules/lodash/_objectToString.js
var wr = V((vl, Pr) => {
  var Eo = Object.prototype, Ao = Eo.toString;
  function Ro(t) {
    return Ao.call(t);
  }
  r(Ro, "objectToString");
  Pr.exports = Ro;
});

// ../node_modules/lodash/_baseGetTag.js
var Ar = V((Pl, Er) => {
  var Sr = Et(), xo = br(), No = wr(), Io = "[object Null]", Lo = "[object Undefined]", Or = Sr ? Sr.toStringTag : void 0;
  function Mo(t) {
    return t == null ? t === void 0 ? Lo : Io : Or && Or in Object(t) ? xo(t) : No(t);
  }
  r(Mo, "baseGetTag");
  Er.exports = Mo;
});

// ../node_modules/lodash/_overArg.js
var xr = V((Sl, Rr) => {
  function Co(t, e) {
    return function(a) {
      return t(e(a));
    };
  }
  r(Co, "overArg");
  Rr.exports = Co;
});

// ../node_modules/lodash/_getPrototype.js
var Ir = V((El, Nr) => {
  var _o = xr(), To = _o(Object.getPrototypeOf, Object);
  Nr.exports = To;
});

// ../node_modules/lodash/isObjectLike.js
var Mr = V((Al, Lr) => {
  function jo(t) {
    return t != null && typeof t == "object";
  }
  r(jo, "isObjectLike");
  Lr.exports = jo;
});

// ../node_modules/lodash/isPlainObject.js
var Tr = V((xl, _r) => {
  var Fo = Ar(), ko = Ir(), $o = Mr(), Do = "[object Object]", qo = Function.prototype, Bo = Object.prototype, Cr = qo.toString, Wo = Bo.hasOwnProperty,
  Uo = Cr.call(Object);
  function Ho(t) {
    if (!$o(t) || Fo(t) != Do)
      return !1;
    var e = ko(t);
    if (e === null)
      return !0;
    var a = Wo.call(e, "constructor") && e.constructor;
    return typeof a == "function" && a instanceof a && Cr.call(a) == Uo;
  }
  r(Ho, "isPlainObject");
  _r.exports = Ho;
});

// ../node_modules/map-or-similar/src/similar.js
var Fr = V((Il, jr) => {
  function we() {
    return this.list = [], this.lastItem = void 0, this.size = 0, this;
  }
  r(we, "Similar");
  we.prototype.get = function(t) {
    var e;
    if (this.lastItem && this.isEqual(this.lastItem.key, t))
      return this.lastItem.val;
    if (e = this.indexOf(t), e >= 0)
      return this.lastItem = this.list[e], this.list[e].val;
  };
  we.prototype.set = function(t, e) {
    var a;
    return this.lastItem && this.isEqual(this.lastItem.key, t) ? (this.lastItem.val = e, this) : (a = this.indexOf(t), a >= 0 ? (this.lastItem =
    this.list[a], this.list[a].val = e, this) : (this.lastItem = { key: t, val: e }, this.list.push(this.lastItem), this.size++, this));
  };
  we.prototype.delete = function(t) {
    var e;
    if (this.lastItem && this.isEqual(this.lastItem.key, t) && (this.lastItem = void 0), e = this.indexOf(t), e >= 0)
      return this.size--, this.list.splice(e, 1)[0];
  };
  we.prototype.has = function(t) {
    var e;
    return this.lastItem && this.isEqual(this.lastItem.key, t) ? !0 : (e = this.indexOf(t), e >= 0 ? (this.lastItem = this.list[e], !0) : !1);
  };
  we.prototype.forEach = function(t, e) {
    var a;
    for (a = 0; a < this.size; a++)
      t.call(e || this, this.list[a].val, this.list[a].key, this);
  };
  we.prototype.indexOf = function(t) {
    var e;
    for (e = 0; e < this.size; e++)
      if (this.isEqual(this.list[e].key, t))
        return e;
    return -1;
  };
  we.prototype.isEqual = function(t, e) {
    return t === e || t !== t && e !== e;
  };
  jr.exports = we;
});

// ../node_modules/map-or-similar/src/map-or-similar.js
var $r = V((Ml, kr) => {
  kr.exports = function(t) {
    if (typeof Map != "function" || t) {
      var e = Fr();
      return new e();
    } else
      return /* @__PURE__ */ new Map();
  };
});

// ../node_modules/memoizerific/src/memoizerific.js
var Br = V((Cl, qr) => {
  var Dr = $r();
  qr.exports = function(t) {
    var e = new Dr(process.env.FORCE_SIMILAR_INSTEAD_OF_MAP === "true"), a = [];
    return function(n) {
      var o = /* @__PURE__ */ r(function() {
        var i = e, l, w, v = arguments.length - 1, I = Array(v + 1), R = !0, m;
        if ((o.numArgs || o.numArgs === 0) && o.numArgs !== v + 1)
          throw new Error("Memoizerific functions should always be called with the same number of arguments");
        for (m = 0; m < v; m++) {
          if (I[m] = {
            cacheItem: i,
            arg: arguments[m]
          }, i.has(arguments[m])) {
            i = i.get(arguments[m]);
            continue;
          }
          R = !1, l = new Dr(process.env.FORCE_SIMILAR_INSTEAD_OF_MAP === "true"), i.set(arguments[m], l), i = l;
        }
        return R && (i.has(arguments[v]) ? w = i.get(arguments[v]) : R = !1), R || (w = n.apply(null, arguments), i.set(arguments[v], w)), t >
        0 && (I[v] = {
          cacheItem: i,
          arg: arguments[v]
        }, R ? zo(a, I) : a.push(I), a.length > t && Go(a.shift())), o.wasMemoized = R, o.numArgs = v + 1, w;
      }, "memoizerific");
      return o.limit = t, o.wasMemoized = !1, o.cache = e, o.lru = a, o;
    };
  };
  function zo(t, e) {
    var a = t.length, n = e.length, o, i, l;
    for (i = 0; i < a; i++) {
      for (o = !0, l = 0; l < n; l++)
        if (!Vo(t[i][l].arg, e[l].arg)) {
          o = !1;
          break;
        }
      if (o)
        break;
    }
    t.push(t.splice(i, 1)[0]);
  }
  r(zo, "moveToMostRecentLru");
  function Go(t) {
    var e = t.length, a = t[e - 1], n, o;
    for (a.cacheItem.delete(a.arg), o = e - 2; o >= 0 && (a = t[o], n = a.cacheItem.get(a.arg), !n || !n.size); o--)
      a.cacheItem.delete(a.arg);
  }
  r(Go, "removeCachedResult");
  function Vo(t, e) {
    return t === e || t !== t && e !== e;
  }
  r(Vo, "isEqual");
});

// ../node_modules/es-errors/index.js
var Ur = V((Tl, Wr) => {
  "use strict";
  Wr.exports = Error;
});

// ../node_modules/es-errors/eval.js
var zr = V((jl, Hr) => {
  "use strict";
  Hr.exports = EvalError;
});

// ../node_modules/es-errors/range.js
var Vr = V((Fl, Gr) => {
  "use strict";
  Gr.exports = RangeError;
});

// ../node_modules/es-errors/ref.js
var Qr = V((kl, Kr) => {
  "use strict";
  Kr.exports = ReferenceError;
});

// ../node_modules/es-errors/syntax.js
var At = V(($l, Jr) => {
  "use strict";
  Jr.exports = SyntaxError;
});

// ../node_modules/es-errors/type.js
var Ce = V((Dl, Yr) => {
  "use strict";
  Yr.exports = TypeError;
});

// ../node_modules/es-errors/uri.js
var Zr = V((ql, Xr) => {
  "use strict";
  Xr.exports = URIError;
});

// ../node_modules/has-symbols/shams.js
var tn = V((Bl, en) => {
  "use strict";
  en.exports = /* @__PURE__ */ r(function() {
    if (typeof Symbol != "function" || typeof Object.getOwnPropertySymbols != "function")
      return !1;
    if (typeof Symbol.iterator == "symbol")
      return !0;
    var e = {}, a = Symbol("test"), n = Object(a);
    if (typeof a == "string" || Object.prototype.toString.call(a) !== "[object Symbol]" || Object.prototype.toString.call(n) !== "[object Sy\
mbol]")
      return !1;
    var o = 42;
    e[a] = o;
    for (a in e)
      return !1;
    if (typeof Object.keys == "function" && Object.keys(e).length !== 0 || typeof Object.getOwnPropertyNames == "function" && Object.getOwnPropertyNames(
    e).length !== 0)
      return !1;
    var i = Object.getOwnPropertySymbols(e);
    if (i.length !== 1 || i[0] !== a || !Object.prototype.propertyIsEnumerable.call(e, a))
      return !1;
    if (typeof Object.getOwnPropertyDescriptor == "function") {
      var l = Object.getOwnPropertyDescriptor(e, a);
      if (l.value !== o || l.enumerable !== !0)
        return !1;
    }
    return !0;
  }, "hasSymbols");
});

// ../node_modules/has-symbols/index.js
var an = V((Ul, nn) => {
  "use strict";
  var rn = typeof Symbol < "u" && Symbol, Ko = tn();
  nn.exports = /* @__PURE__ */ r(function() {
    return typeof rn != "function" || typeof Symbol != "function" || typeof rn("foo") != "symbol" || typeof Symbol("bar") != "symbol" ? !1 :
    Ko();
  }, "hasNativeSymbols");
});

// ../node_modules/has-proto/index.js
var un = V((zl, on) => {
  "use strict";
  var Rt = {
    __proto__: null,
    foo: {}
  }, Qo = Object;
  on.exports = /* @__PURE__ */ r(function() {
    return { __proto__: Rt }.foo === Rt.foo && !(Rt instanceof Qo);
  }, "hasProto");
});

// ../node_modules/function-bind/implementation.js
var sn = V((Vl, cn) => {
  "use strict";
  var Jo = "Function.prototype.bind called on incompatible ", Yo = Object.prototype.toString, Xo = Math.max, Zo = "[object Function]", ln = /* @__PURE__ */ r(
  function(e, a) {
    for (var n = [], o = 0; o < e.length; o += 1)
      n[o] = e[o];
    for (var i = 0; i < a.length; i += 1)
      n[i + e.length] = a[i];
    return n;
  }, "concatty"), ei = /* @__PURE__ */ r(function(e, a) {
    for (var n = [], o = a || 0, i = 0; o < e.length; o += 1, i += 1)
      n[i] = e[o];
    return n;
  }, "slicy"), ti = /* @__PURE__ */ r(function(t, e) {
    for (var a = "", n = 0; n < t.length; n += 1)
      a += t[n], n + 1 < t.length && (a += e);
    return a;
  }, "joiny");
  cn.exports = /* @__PURE__ */ r(function(e) {
    var a = this;
    if (typeof a != "function" || Yo.apply(a) !== Zo)
      throw new TypeError(Jo + a);
    for (var n = ei(arguments, 1), o, i = /* @__PURE__ */ r(function() {
      if (this instanceof o) {
        var R = a.apply(
          this,
          ln(n, arguments)
        );
        return Object(R) === R ? R : this;
      }
      return a.apply(
        e,
        ln(n, arguments)
      );
    }, "binder"), l = Xo(0, a.length - n.length), w = [], v = 0; v < l; v++)
      w[v] = "$" + v;
    if (o = Function("binder", "return function (" + ti(w, ",") + "){ return binder.apply(this,arguments); }")(i), a.prototype) {
      var I = /* @__PURE__ */ r(function() {
      }, "Empty");
      I.prototype = a.prototype, o.prototype = new I(), I.prototype = null;
    }
    return o;
  }, "bind");
});

// ../node_modules/function-bind/index.js
var Xe = V((Ql, fn) => {
  "use strict";
  var ri = sn();
  fn.exports = Function.prototype.bind || ri;
});

// ../node_modules/hasown/index.js
var hn = V((Jl, pn) => {
  "use strict";
  var ni = Function.prototype.call, ai = Object.prototype.hasOwnProperty, oi = Xe();
  pn.exports = oi.call(ni, ai);
});

// ../node_modules/get-intrinsic/index.js
var Ie = V((Yl, vn) => {
  "use strict";
  var ee, ii = Ur(), ui = zr(), li = Vr(), ci = Qr(), Fe = At(), je = Ce(), si = Zr(), gn = Function, xt = /* @__PURE__ */ r(function(t) {
    try {
      return gn('"use strict"; return (' + t + ").constructor;")();
    } catch {
    }
  }, "getEvalledConstructor"), xe = Object.getOwnPropertyDescriptor;
  if (xe)
    try {
      xe({}, "");
    } catch {
      xe = null;
    }
  var Nt = /* @__PURE__ */ r(function() {
    throw new je();
  }, "throwTypeError"), fi = xe ? function() {
    try {
      return arguments.callee, Nt;
    } catch {
      try {
        return xe(arguments, "callee").get;
      } catch {
        return Nt;
      }
    }
  }() : Nt, _e = an()(), pi = un()(), ce = Object.getPrototypeOf || (pi ? function(t) {
    return t.__proto__;
  } : null), Te = {}, hi = typeof Uint8Array > "u" || !ce ? ee : ce(Uint8Array), Ne = {
    __proto__: null,
    "%AggregateError%": typeof AggregateError > "u" ? ee : AggregateError,
    "%Array%": Array,
    "%ArrayBuffer%": typeof ArrayBuffer > "u" ? ee : ArrayBuffer,
    "%ArrayIteratorPrototype%": _e && ce ? ce([][Symbol.iterator]()) : ee,
    "%AsyncFromSyncIteratorPrototype%": ee,
    "%AsyncFunction%": Te,
    "%AsyncGenerator%": Te,
    "%AsyncGeneratorFunction%": Te,
    "%AsyncIteratorPrototype%": Te,
    "%Atomics%": typeof Atomics > "u" ? ee : Atomics,
    "%BigInt%": typeof BigInt > "u" ? ee : BigInt,
    "%BigInt64Array%": typeof BigInt64Array > "u" ? ee : BigInt64Array,
    "%BigUint64Array%": typeof BigUint64Array > "u" ? ee : BigUint64Array,
    "%Boolean%": Boolean,
    "%DataView%": typeof DataView > "u" ? ee : DataView,
    "%Date%": Date,
    "%decodeURI%": decodeURI,
    "%decodeURIComponent%": decodeURIComponent,
    "%encodeURI%": encodeURI,
    "%encodeURIComponent%": encodeURIComponent,
    "%Error%": ii,
    "%eval%": eval,
    // eslint-disable-line no-eval
    "%EvalError%": ui,
    "%Float32Array%": typeof Float32Array > "u" ? ee : Float32Array,
    "%Float64Array%": typeof Float64Array > "u" ? ee : Float64Array,
    "%FinalizationRegistry%": typeof FinalizationRegistry > "u" ? ee : FinalizationRegistry,
    "%Function%": gn,
    "%GeneratorFunction%": Te,
    "%Int8Array%": typeof Int8Array > "u" ? ee : Int8Array,
    "%Int16Array%": typeof Int16Array > "u" ? ee : Int16Array,
    "%Int32Array%": typeof Int32Array > "u" ? ee : Int32Array,
    "%isFinite%": isFinite,
    "%isNaN%": isNaN,
    "%IteratorPrototype%": _e && ce ? ce(ce([][Symbol.iterator]())) : ee,
    "%JSON%": typeof JSON == "object" ? JSON : ee,
    "%Map%": typeof Map > "u" ? ee : Map,
    "%MapIteratorPrototype%": typeof Map > "u" || !_e || !ce ? ee : ce((/* @__PURE__ */ new Map())[Symbol.iterator]()),
    "%Math%": Math,
    "%Number%": Number,
    "%Object%": Object,
    "%parseFloat%": parseFloat,
    "%parseInt%": parseInt,
    "%Promise%": typeof Promise > "u" ? ee : Promise,
    "%Proxy%": typeof Proxy > "u" ? ee : Proxy,
    "%RangeError%": li,
    "%ReferenceError%": ci,
    "%Reflect%": typeof Reflect > "u" ? ee : Reflect,
    "%RegExp%": RegExp,
    "%Set%": typeof Set > "u" ? ee : Set,
    "%SetIteratorPrototype%": typeof Set > "u" || !_e || !ce ? ee : ce((/* @__PURE__ */ new Set())[Symbol.iterator]()),
    "%SharedArrayBuffer%": typeof SharedArrayBuffer > "u" ? ee : SharedArrayBuffer,
    "%String%": String,
    "%StringIteratorPrototype%": _e && ce ? ce(""[Symbol.iterator]()) : ee,
    "%Symbol%": _e ? Symbol : ee,
    "%SyntaxError%": Fe,
    "%ThrowTypeError%": fi,
    "%TypedArray%": hi,
    "%TypeError%": je,
    "%Uint8Array%": typeof Uint8Array > "u" ? ee : Uint8Array,
    "%Uint8ClampedArray%": typeof Uint8ClampedArray > "u" ? ee : Uint8ClampedArray,
    "%Uint16Array%": typeof Uint16Array > "u" ? ee : Uint16Array,
    "%Uint32Array%": typeof Uint32Array > "u" ? ee : Uint32Array,
    "%URIError%": si,
    "%WeakMap%": typeof WeakMap > "u" ? ee : WeakMap,
    "%WeakRef%": typeof WeakRef > "u" ? ee : WeakRef,
    "%WeakSet%": typeof WeakSet > "u" ? ee : WeakSet
  };
  if (ce)
    try {
      null.error;
    } catch (t) {
      yn = ce(ce(t)), Ne["%Error.prototype%"] = yn;
    }
  var yn, yi = /* @__PURE__ */ r(function t(e) {
    var a;
    if (e === "%AsyncFunction%")
      a = xt("async function () {}");
    else if (e === "%GeneratorFunction%")
      a = xt("function* () {}");
    else if (e === "%AsyncGeneratorFunction%")
      a = xt("async function* () {}");
    else if (e === "%AsyncGenerator%") {
      var n = t("%AsyncGeneratorFunction%");
      n && (a = n.prototype);
    } else if (e === "%AsyncIteratorPrototype%") {
      var o = t("%AsyncGenerator%");
      o && ce && (a = ce(o.prototype));
    }
    return Ne[e] = a, a;
  }, "doEval"), dn = {
    __proto__: null,
    "%ArrayBufferPrototype%": ["ArrayBuffer", "prototype"],
    "%ArrayPrototype%": ["Array", "prototype"],
    "%ArrayProto_entries%": ["Array", "prototype", "entries"],
    "%ArrayProto_forEach%": ["Array", "prototype", "forEach"],
    "%ArrayProto_keys%": ["Array", "prototype", "keys"],
    "%ArrayProto_values%": ["Array", "prototype", "values"],
    "%AsyncFunctionPrototype%": ["AsyncFunction", "prototype"],
    "%AsyncGenerator%": ["AsyncGeneratorFunction", "prototype"],
    "%AsyncGeneratorPrototype%": ["AsyncGeneratorFunction", "prototype", "prototype"],
    "%BooleanPrototype%": ["Boolean", "prototype"],
    "%DataViewPrototype%": ["DataView", "prototype"],
    "%DatePrototype%": ["Date", "prototype"],
    "%ErrorPrototype%": ["Error", "prototype"],
    "%EvalErrorPrototype%": ["EvalError", "prototype"],
    "%Float32ArrayPrototype%": ["Float32Array", "prototype"],
    "%Float64ArrayPrototype%": ["Float64Array", "prototype"],
    "%FunctionPrototype%": ["Function", "prototype"],
    "%Generator%": ["GeneratorFunction", "prototype"],
    "%GeneratorPrototype%": ["GeneratorFunction", "prototype", "prototype"],
    "%Int8ArrayPrototype%": ["Int8Array", "prototype"],
    "%Int16ArrayPrototype%": ["Int16Array", "prototype"],
    "%Int32ArrayPrototype%": ["Int32Array", "prototype"],
    "%JSONParse%": ["JSON", "parse"],
    "%JSONStringify%": ["JSON", "stringify"],
    "%MapPrototype%": ["Map", "prototype"],
    "%NumberPrototype%": ["Number", "prototype"],
    "%ObjectPrototype%": ["Object", "prototype"],
    "%ObjProto_toString%": ["Object", "prototype", "toString"],
    "%ObjProto_valueOf%": ["Object", "prototype", "valueOf"],
    "%PromisePrototype%": ["Promise", "prototype"],
    "%PromiseProto_then%": ["Promise", "prototype", "then"],
    "%Promise_all%": ["Promise", "all"],
    "%Promise_reject%": ["Promise", "reject"],
    "%Promise_resolve%": ["Promise", "resolve"],
    "%RangeErrorPrototype%": ["RangeError", "prototype"],
    "%ReferenceErrorPrototype%": ["ReferenceError", "prototype"],
    "%RegExpPrototype%": ["RegExp", "prototype"],
    "%SetPrototype%": ["Set", "prototype"],
    "%SharedArrayBufferPrototype%": ["SharedArrayBuffer", "prototype"],
    "%StringPrototype%": ["String", "prototype"],
    "%SymbolPrototype%": ["Symbol", "prototype"],
    "%SyntaxErrorPrototype%": ["SyntaxError", "prototype"],
    "%TypedArrayPrototype%": ["TypedArray", "prototype"],
    "%TypeErrorPrototype%": ["TypeError", "prototype"],
    "%Uint8ArrayPrototype%": ["Uint8Array", "prototype"],
    "%Uint8ClampedArrayPrototype%": ["Uint8ClampedArray", "prototype"],
    "%Uint16ArrayPrototype%": ["Uint16Array", "prototype"],
    "%Uint32ArrayPrototype%": ["Uint32Array", "prototype"],
    "%URIErrorPrototype%": ["URIError", "prototype"],
    "%WeakMapPrototype%": ["WeakMap", "prototype"],
    "%WeakSetPrototype%": ["WeakSet", "prototype"]
  }, Ue = Xe(), Ze = hn(), di = Ue.call(Function.call, Array.prototype.concat), mi = Ue.call(Function.apply, Array.prototype.splice), mn = Ue.
  call(Function.call, String.prototype.replace), et = Ue.call(Function.call, String.prototype.slice), gi = Ue.call(Function.call, RegExp.prototype.
  exec), vi = /[^%.[\]]+|\[(?:(-?\d+(?:\.\d+)?)|(["'])((?:(?!\2)[^\\]|\\.)*?)\2)\]|(?=(?:\.|\[\])(?:\.|\[\]|%$))/g, bi = /\\(\\)?/g, Pi = /* @__PURE__ */ r(
  function(e) {
    var a = et(e, 0, 1), n = et(e, -1);
    if (a === "%" && n !== "%")
      throw new Fe("invalid intrinsic syntax, expected closing `%`");
    if (n === "%" && a !== "%")
      throw new Fe("invalid intrinsic syntax, expected opening `%`");
    var o = [];
    return mn(e, vi, function(i, l, w, v) {
      o[o.length] = w ? mn(v, bi, "$1") : l || i;
    }), o;
  }, "stringToPath"), wi = /* @__PURE__ */ r(function(e, a) {
    var n = e, o;
    if (Ze(dn, n) && (o = dn[n], n = "%" + o[0] + "%"), Ze(Ne, n)) {
      var i = Ne[n];
      if (i === Te && (i = yi(n)), typeof i > "u" && !a)
        throw new je("intrinsic " + e + " exists, but is not available. Please file an issue!");
      return {
        alias: o,
        name: n,
        value: i
      };
    }
    throw new Fe("intrinsic " + e + " does not exist!");
  }, "getBaseIntrinsic");
  vn.exports = /* @__PURE__ */ r(function(e, a) {
    if (typeof e != "string" || e.length === 0)
      throw new je("intrinsic name must be a non-empty string");
    if (arguments.length > 1 && typeof a != "boolean")
      throw new je('"allowMissing" argument must be a boolean');
    if (gi(/^%?[^%]*%?$/, e) === null)
      throw new Fe("`%` may not be present anywhere but at the beginning and end of the intrinsic name");
    var n = Pi(e), o = n.length > 0 ? n[0] : "", i = wi("%" + o + "%", a), l = i.name, w = i.value, v = !1, I = i.alias;
    I && (o = I[0], mi(n, di([0, 1], I)));
    for (var R = 1, m = !0; R < n.length; R += 1) {
      var u = n[R], S = et(u, 0, 1), O = et(u, -1);
      if ((S === '"' || S === "'" || S === "`" || O === '"' || O === "'" || O === "`") && S !== O)
        throw new Fe("property names with quotes must have matching quotes");
      if ((u === "constructor" || !m) && (v = !0), o += "." + u, l = "%" + o + "%", Ze(Ne, l))
        w = Ne[l];
      else if (w != null) {
        if (!(u in w)) {
          if (!a)
            throw new je("base intrinsic for " + e + " exists, but the property is not available.");
          return;
        }
        if (xe && R + 1 >= n.length) {
          var $ = xe(w, u);
          m = !!$, m && "get" in $ && !("originalValue" in $.get) ? w = $.get : w = w[u];
        } else
          m = Ze(w, u), w = w[u];
        m && !v && (Ne[l] = w);
      }
    }
    return w;
  }, "GetIntrinsic");
});

// ../node_modules/es-define-property/index.js
var rt = V((Zl, bn) => {
  "use strict";
  var Si = Ie(), tt = Si("%Object.defineProperty%", !0) || !1;
  if (tt)
    try {
      tt({}, "a", { value: 1 });
    } catch {
      tt = !1;
    }
  bn.exports = tt;
});

// ../node_modules/gopd/index.js
var It = V((ec, Pn) => {
  "use strict";
  var Oi = Ie(), nt = Oi("%Object.getOwnPropertyDescriptor%", !0);
  if (nt)
    try {
      nt([], "length");
    } catch {
      nt = null;
    }
  Pn.exports = nt;
});

// ../node_modules/define-data-property/index.js
var En = V((tc, On) => {
  "use strict";
  var wn = rt(), Ei = At(), ke = Ce(), Sn = It();
  On.exports = /* @__PURE__ */ r(function(e, a, n) {
    if (!e || typeof e != "object" && typeof e != "function")
      throw new ke("`obj` must be an object or a function`");
    if (typeof a != "string" && typeof a != "symbol")
      throw new ke("`property` must be a string or a symbol`");
    if (arguments.length > 3 && typeof arguments[3] != "boolean" && arguments[3] !== null)
      throw new ke("`nonEnumerable`, if provided, must be a boolean or null");
    if (arguments.length > 4 && typeof arguments[4] != "boolean" && arguments[4] !== null)
      throw new ke("`nonWritable`, if provided, must be a boolean or null");
    if (arguments.length > 5 && typeof arguments[5] != "boolean" && arguments[5] !== null)
      throw new ke("`nonConfigurable`, if provided, must be a boolean or null");
    if (arguments.length > 6 && typeof arguments[6] != "boolean")
      throw new ke("`loose`, if provided, must be a boolean");
    var o = arguments.length > 3 ? arguments[3] : null, i = arguments.length > 4 ? arguments[4] : null, l = arguments.length > 5 ? arguments[5] :
    null, w = arguments.length > 6 ? arguments[6] : !1, v = !!Sn && Sn(e, a);
    if (wn)
      wn(e, a, {
        configurable: l === null && v ? v.configurable : !l,
        enumerable: o === null && v ? v.enumerable : !o,
        value: n,
        writable: i === null && v ? v.writable : !i
      });
    else if (w || !o && !i && !l)
      e[a] = n;
    else
      throw new Ei("This environment does not support defining a property as non-configurable, non-writable, or non-enumerable.");
  }, "defineDataProperty");
});

// ../node_modules/has-property-descriptors/index.js
var xn = V((nc, Rn) => {
  "use strict";
  var Lt = rt(), An = /* @__PURE__ */ r(function() {
    return !!Lt;
  }, "hasPropertyDescriptors");
  An.hasArrayLengthDefineBug = /* @__PURE__ */ r(function() {
    if (!Lt)
      return null;
    try {
      return Lt([], "length", { value: 1 }).length !== 1;
    } catch {
      return !0;
    }
  }, "hasArrayLengthDefineBug");
  Rn.exports = An;
});

// ../node_modules/set-function-length/index.js
var Cn = V((oc, Mn) => {
  "use strict";
  var Ai = Ie(), Nn = En(), Ri = xn()(), In = It(), Ln = Ce(), xi = Ai("%Math.floor%");
  Mn.exports = /* @__PURE__ */ r(function(e, a) {
    if (typeof e != "function")
      throw new Ln("`fn` is not a function");
    if (typeof a != "number" || a < 0 || a > 4294967295 || xi(a) !== a)
      throw new Ln("`length` must be a positive 32-bit integer");
    var n = arguments.length > 2 && !!arguments[2], o = !0, i = !0;
    if ("length" in e && In) {
      var l = In(e, "length");
      l && !l.configurable && (o = !1), l && !l.writable && (i = !1);
    }
    return (o || i || !n) && (Ri ? Nn(
      /** @type {Parameters<define>[0]} */
      e,
      "length",
      a,
      !0,
      !0
    ) : Nn(
      /** @type {Parameters<define>[0]} */
      e,
      "length",
      a
    )), e;
  }, "setFunctionLength");
});

// ../node_modules/call-bind/index.js
var $n = V((uc, at) => {
  "use strict";
  var Mt = Xe(), ot = Ie(), Ni = Cn(), Ii = Ce(), jn = ot("%Function.prototype.apply%"), Fn = ot("%Function.prototype.call%"), kn = ot("%Ref\
lect.apply%", !0) || Mt.call(Fn, jn), _n = rt(), Li = ot("%Math.max%");
  at.exports = /* @__PURE__ */ r(function(e) {
    if (typeof e != "function")
      throw new Ii("a function is required");
    var a = kn(Mt, Fn, arguments);
    return Ni(
      a,
      1 + Li(0, e.length - (arguments.length - 1)),
      !0
    );
  }, "callBind");
  var Tn = /* @__PURE__ */ r(function() {
    return kn(Mt, jn, arguments);
  }, "applyBind");
  _n ? _n(at.exports, "apply", { value: Tn }) : at.exports.apply = Tn;
});

// ../node_modules/call-bind/callBound.js
var Wn = V((cc, Bn) => {
  "use strict";
  var Dn = Ie(), qn = $n(), Mi = qn(Dn("String.prototype.indexOf"));
  Bn.exports = /* @__PURE__ */ r(function(e, a) {
    var n = Dn(e, !!a);
    return typeof n == "function" && Mi(e, ".prototype.") > -1 ? qn(n) : n;
  }, "callBoundIntrinsic");
});

// ../node_modules/object-inspect/util.inspect.js
var Hn = V((fc, Un) => {
  Un.exports = require("util").inspect;
});

// ../node_modules/object-inspect/index.js
var sa = V((pc, ca) => {
  var Bt = typeof Map == "function" && Map.prototype, Ct = Object.getOwnPropertyDescriptor && Bt ? Object.getOwnPropertyDescriptor(Map.prototype,
  "size") : null, ut = Bt && Ct && typeof Ct.get == "function" ? Ct.get : null, zn = Bt && Map.prototype.forEach, Wt = typeof Set == "functi\
on" && Set.prototype, _t = Object.getOwnPropertyDescriptor && Wt ? Object.getOwnPropertyDescriptor(Set.prototype, "size") : null, lt = Wt &&
  _t && typeof _t.get == "function" ? _t.get : null, Gn = Wt && Set.prototype.forEach, Ci = typeof WeakMap == "function" && WeakMap.prototype,
  ze = Ci ? WeakMap.prototype.has : null, _i = typeof WeakSet == "function" && WeakSet.prototype, Ge = _i ? WeakSet.prototype.has : null, Ti = typeof WeakRef ==
  "function" && WeakRef.prototype, Vn = Ti ? WeakRef.prototype.deref : null, ji = Boolean.prototype.valueOf, Fi = Object.prototype.toString,
  ki = Function.prototype.toString, $i = String.prototype.match, Ut = String.prototype.slice, Oe = String.prototype.replace, Di = String.prototype.
  toUpperCase, Kn = String.prototype.toLowerCase, na = RegExp.prototype.test, Qn = Array.prototype.concat, me = Array.prototype.join, qi = Array.
  prototype.slice, Jn = Math.floor, Ft = typeof BigInt == "function" ? BigInt.prototype.valueOf : null, Tt = Object.getOwnPropertySymbols, kt = typeof Symbol ==
  "function" && typeof Symbol.iterator == "symbol" ? Symbol.prototype.toString : null, $e = typeof Symbol == "function" && typeof Symbol.iterator ==
  "object", se = typeof Symbol == "function" && Symbol.toStringTag && (typeof Symbol.toStringTag === $e || !0) ? Symbol.toStringTag : null, aa = Object.
  prototype.propertyIsEnumerable, Yn = (typeof Reflect == "function" ? Reflect.getPrototypeOf : Object.getPrototypeOf) || ([].__proto__ === Array.
  prototype ? function(t) {
    return t.__proto__;
  } : null);
  function Xn(t, e) {
    if (t === 1 / 0 || t === -1 / 0 || t !== t || t && t > -1e3 && t < 1e3 || na.call(/e/, e))
      return e;
    var a = /[0-9](?=(?:[0-9]{3})+(?![0-9]))/g;
    if (typeof t == "number") {
      var n = t < 0 ? -Jn(-t) : Jn(t);
      if (n !== t) {
        var o = String(n), i = Ut.call(e, o.length + 1);
        return Oe.call(o, a, "$&_") + "." + Oe.call(Oe.call(i, /([0-9]{3})/g, "$&_"), /_$/, "");
      }
    }
    return Oe.call(e, a, "$&_");
  }
  r(Xn, "addNumericSeparator");
  var $t = Hn(), Zn = $t.custom, ea = ia(Zn) ? Zn : null;
  ca.exports = /* @__PURE__ */ r(function t(e, a, n, o) {
    var i = a || {};
    if (Se(i, "quoteStyle") && i.quoteStyle !== "single" && i.quoteStyle !== "double")
      throw new TypeError('option "quoteStyle" must be "single" or "double"');
    if (Se(i, "maxStringLength") && (typeof i.maxStringLength == "number" ? i.maxStringLength < 0 && i.maxStringLength !== 1 / 0 : i.maxStringLength !==
    null))
      throw new TypeError('option "maxStringLength", if provided, must be a positive integer, Infinity, or `null`');
    var l = Se(i, "customInspect") ? i.customInspect : !0;
    if (typeof l != "boolean" && l !== "symbol")
      throw new TypeError("option \"customInspect\", if provided, must be `true`, `false`, or `'symbol'`");
    if (Se(i, "indent") && i.indent !== null && i.indent !== "	" && !(parseInt(i.indent, 10) === i.indent && i.indent > 0))
      throw new TypeError('option "indent" must be "\\t", an integer > 0, or `null`');
    if (Se(i, "numericSeparator") && typeof i.numericSeparator != "boolean")
      throw new TypeError('option "numericSeparator", if provided, must be `true` or `false`');
    var w = i.numericSeparator;
    if (typeof e > "u")
      return "undefined";
    if (e === null)
      return "null";
    if (typeof e == "boolean")
      return e ? "true" : "false";
    if (typeof e == "string")
      return la(e, i);
    if (typeof e == "number") {
      if (e === 0)
        return 1 / 0 / e > 0 ? "0" : "-0";
      var v = String(e);
      return w ? Xn(e, v) : v;
    }
    if (typeof e == "bigint") {
      var I = String(e) + "n";
      return w ? Xn(e, I) : I;
    }
    var R = typeof i.depth > "u" ? 5 : i.depth;
    if (typeof n > "u" && (n = 0), n >= R && R > 0 && typeof e == "object")
      return Dt(e) ? "[Array]" : "[Object]";
    var m = au(i, n);
    if (typeof o > "u")
      o = [];
    else if (ua(o, e) >= 0)
      return "[Circular]";
    function u(N, E, B) {
      if (E && (o = qi.call(o), o.push(E)), B) {
        var c = {
          depth: i.depth
        };
        return Se(i, "quoteStyle") && (c.quoteStyle = i.quoteStyle), t(N, c, n + 1, o);
      }
      return t(N, i, n + 1, o);
    }
    if (r(u, "inspect"), typeof e == "function" && !ta(e)) {
      var S = Qi(e), O = it(e, u);
      return "[Function" + (S ? ": " + S : " (anonymous)") + "]" + (O.length > 0 ? " { " + me.call(O, ", ") + " }" : "");
    }
    if (ia(e)) {
      var $ = $e ? Oe.call(String(e), /^(Symbol\(.*\))_[^)]*$/, "$1") : kt.call(e);
      return typeof e == "object" && !$e ? He($) : $;
    }
    if (tu(e)) {
      for (var h = "<" + Kn.call(String(e.nodeName)), s = e.attributes || [], b = 0; b < s.length; b++)
        h += " " + s[b].name + "=" + oa(Bi(s[b].value), "double", i);
      return h += ">", e.childNodes && e.childNodes.length && (h += "..."), h += "</" + Kn.call(String(e.nodeName)) + ">", h;
    }
    if (Dt(e)) {
      if (e.length === 0)
        return "[]";
      var f = it(e, u);
      return m && !nu(f) ? "[" + qt(f, m) + "]" : "[ " + me.call(f, ", ") + " ]";
    }
    if (Ui(e)) {
      var L = it(e, u);
      return !("cause" in Error.prototype) && "cause" in e && !aa.call(e, "cause") ? "{ [" + String(e) + "] " + me.call(Qn.call("[cause]: " +
      u(e.cause), L), ", ") + " }" : L.length === 0 ? "[" + String(e) + "]" : "{ [" + String(e) + "] " + me.call(L, ", ") + " }";
    }
    if (typeof e == "object" && l) {
      if (ea && typeof e[ea] == "function" && $t)
        return $t(e, { depth: R - n });
      if (l !== "symbol" && typeof e.inspect == "function")
        return e.inspect();
    }
    if (Ji(e)) {
      var _ = [];
      return zn && zn.call(e, function(N, E) {
        _.push(u(E, e, !0) + " => " + u(N, e));
      }), ra("Map", ut.call(e), _, m);
    }
    if (Zi(e)) {
      var j = [];
      return Gn && Gn.call(e, function(N) {
        j.push(u(N, e));
      }), ra("Set", lt.call(e), j, m);
    }
    if (Yi(e))
      return jt("WeakMap");
    if (eu(e))
      return jt("WeakSet");
    if (Xi(e))
      return jt("WeakRef");
    if (zi(e))
      return He(u(Number(e)));
    if (Vi(e))
      return He(u(Ft.call(e)));
    if (Gi(e))
      return He(ji.call(e));
    if (Hi(e))
      return He(u(String(e)));
    if (typeof window < "u" && e === window)
      return "{ [object Window] }";
    if (e === global)
      return "{ [object globalThis] }";
    if (!Wi(e) && !ta(e)) {
      var W = it(e, u), U = Yn ? Yn(e) === Object.prototype : e instanceof Object || e.constructor === Object, J = e instanceof Object ? "" :
      "null prototype", F = !U && se && Object(e) === e && se in e ? Ut.call(Ee(e), 8, -1) : J ? "Object" : "", K = U || typeof e.constructor !=
      "function" ? "" : e.constructor.name ? e.constructor.name + " " : "", G = K + (F || J ? "[" + me.call(Qn.call([], F || [], J || []), "\
: ") + "] " : "");
      return W.length === 0 ? G + "{}" : m ? G + "{" + qt(W, m) + "}" : G + "{ " + me.call(W, ", ") + " }";
    }
    return String(e);
  }, "inspect_");
  function oa(t, e, a) {
    var n = (a.quoteStyle || e) === "double" ? '"' : "'";
    return n + t + n;
  }
  r(oa, "wrapQuotes");
  function Bi(t) {
    return Oe.call(String(t), /"/g, "&quot;");
  }
  r(Bi, "quote");
  function Dt(t) {
    return Ee(t) === "[object Array]" && (!se || !(typeof t == "object" && se in t));
  }
  r(Dt, "isArray");
  function Wi(t) {
    return Ee(t) === "[object Date]" && (!se || !(typeof t == "object" && se in t));
  }
  r(Wi, "isDate");
  function ta(t) {
    return Ee(t) === "[object RegExp]" && (!se || !(typeof t == "object" && se in t));
  }
  r(ta, "isRegExp");
  function Ui(t) {
    return Ee(t) === "[object Error]" && (!se || !(typeof t == "object" && se in t));
  }
  r(Ui, "isError");
  function Hi(t) {
    return Ee(t) === "[object String]" && (!se || !(typeof t == "object" && se in t));
  }
  r(Hi, "isString");
  function zi(t) {
    return Ee(t) === "[object Number]" && (!se || !(typeof t == "object" && se in t));
  }
  r(zi, "isNumber");
  function Gi(t) {
    return Ee(t) === "[object Boolean]" && (!se || !(typeof t == "object" && se in t));
  }
  r(Gi, "isBoolean");
  function ia(t) {
    if ($e)
      return t && typeof t == "object" && t instanceof Symbol;
    if (typeof t == "symbol")
      return !0;
    if (!t || typeof t != "object" || !kt)
      return !1;
    try {
      return kt.call(t), !0;
    } catch {
    }
    return !1;
  }
  r(ia, "isSymbol");
  function Vi(t) {
    if (!t || typeof t != "object" || !Ft)
      return !1;
    try {
      return Ft.call(t), !0;
    } catch {
    }
    return !1;
  }
  r(Vi, "isBigInt");
  var Ki = Object.prototype.hasOwnProperty || function(t) {
    return t in this;
  };
  function Se(t, e) {
    return Ki.call(t, e);
  }
  r(Se, "has");
  function Ee(t) {
    return Fi.call(t);
  }
  r(Ee, "toStr");
  function Qi(t) {
    if (t.name)
      return t.name;
    var e = $i.call(ki.call(t), /^function\s*([\w$]+)/);
    return e ? e[1] : null;
  }
  r(Qi, "nameOf");
  function ua(t, e) {
    if (t.indexOf)
      return t.indexOf(e);
    for (var a = 0, n = t.length; a < n; a++)
      if (t[a] === e)
        return a;
    return -1;
  }
  r(ua, "indexOf");
  function Ji(t) {
    if (!ut || !t || typeof t != "object")
      return !1;
    try {
      ut.call(t);
      try {
        lt.call(t);
      } catch {
        return !0;
      }
      return t instanceof Map;
    } catch {
    }
    return !1;
  }
  r(Ji, "isMap");
  function Yi(t) {
    if (!ze || !t || typeof t != "object")
      return !1;
    try {
      ze.call(t, ze);
      try {
        Ge.call(t, Ge);
      } catch {
        return !0;
      }
      return t instanceof WeakMap;
    } catch {
    }
    return !1;
  }
  r(Yi, "isWeakMap");
  function Xi(t) {
    if (!Vn || !t || typeof t != "object")
      return !1;
    try {
      return Vn.call(t), !0;
    } catch {
    }
    return !1;
  }
  r(Xi, "isWeakRef");
  function Zi(t) {
    if (!lt || !t || typeof t != "object")
      return !1;
    try {
      lt.call(t);
      try {
        ut.call(t);
      } catch {
        return !0;
      }
      return t instanceof Set;
    } catch {
    }
    return !1;
  }
  r(Zi, "isSet");
  function eu(t) {
    if (!Ge || !t || typeof t != "object")
      return !1;
    try {
      Ge.call(t, Ge);
      try {
        ze.call(t, ze);
      } catch {
        return !0;
      }
      return t instanceof WeakSet;
    } catch {
    }
    return !1;
  }
  r(eu, "isWeakSet");
  function tu(t) {
    return !t || typeof t != "object" ? !1 : typeof HTMLElement < "u" && t instanceof HTMLElement ? !0 : typeof t.nodeName == "string" && typeof t.
    getAttribute == "function";
  }
  r(tu, "isElement");
  function la(t, e) {
    if (t.length > e.maxStringLength) {
      var a = t.length - e.maxStringLength, n = "... " + a + " more character" + (a > 1 ? "s" : "");
      return la(Ut.call(t, 0, e.maxStringLength), e) + n;
    }
    var o = Oe.call(Oe.call(t, /(['\\])/g, "\\$1"), /[\x00-\x1f]/g, ru);
    return oa(o, "single", e);
  }
  r(la, "inspectString");
  function ru(t) {
    var e = t.charCodeAt(0), a = {
      8: "b",
      9: "t",
      10: "n",
      12: "f",
      13: "r"
    }[e];
    return a ? "\\" + a : "\\x" + (e < 16 ? "0" : "") + Di.call(e.toString(16));
  }
  r(ru, "lowbyte");
  function He(t) {
    return "Object(" + t + ")";
  }
  r(He, "markBoxed");
  function jt(t) {
    return t + " { ? }";
  }
  r(jt, "weakCollectionOf");
  function ra(t, e, a, n) {
    var o = n ? qt(a, n) : me.call(a, ", ");
    return t + " (" + e + ") {" + o + "}";
  }
  r(ra, "collectionOf");
  function nu(t) {
    for (var e = 0; e < t.length; e++)
      if (ua(t[e], `
`) >= 0)
        return !1;
    return !0;
  }
  r(nu, "singleLineValues");
  function au(t, e) {
    var a;
    if (t.indent === "	")
      a = "	";
    else if (typeof t.indent == "number" && t.indent > 0)
      a = me.call(Array(t.indent + 1), " ");
    else
      return null;
    return {
      base: a,
      prev: me.call(Array(e + 1), a)
    };
  }
  r(au, "getIndent");
  function qt(t, e) {
    if (t.length === 0)
      return "";
    var a = `
` + e.prev + e.base;
    return a + me.call(t, "," + a) + `
` + e.prev;
  }
  r(qt, "indentedJoin");
  function it(t, e) {
    var a = Dt(t), n = [];
    if (a) {
      n.length = t.length;
      for (var o = 0; o < t.length; o++)
        n[o] = Se(t, o) ? e(t[o], t) : "";
    }
    var i = typeof Tt == "function" ? Tt(t) : [], l;
    if ($e) {
      l = {};
      for (var w = 0; w < i.length; w++)
        l["$" + i[w]] = i[w];
    }
    for (var v in t)
      Se(t, v) && (a && String(Number(v)) === v && v < t.length || $e && l["$" + v] instanceof Symbol || (na.call(/[^\w$]/, v) ? n.push(e(v,
      t) + ": " + e(t[v], t)) : n.push(v + ": " + e(t[v], t))));
    if (typeof Tt == "function")
      for (var I = 0; I < i.length; I++)
        aa.call(t, i[I]) && n.push("[" + e(i[I]) + "]: " + e(t[i[I]], t));
    return n;
  }
  r(it, "arrObjKeys");
});

// ../node_modules/side-channel/index.js
var ha = V((yc, pa) => {
  "use strict";
  var fa = Ie(), De = Wn(), ou = sa(), iu = Ce(), ct = fa("%WeakMap%", !0), st = fa("%Map%", !0), uu = De("WeakMap.prototype.get", !0), lu = De(
  "WeakMap.prototype.set", !0), cu = De("WeakMap.prototype.has", !0), su = De("Map.prototype.get", !0), fu = De("Map.prototype.set", !0), pu = De(
  "Map.prototype.has", !0), Ht = /* @__PURE__ */ r(function(t, e) {
    for (var a = t, n; (n = a.next) !== null; a = n)
      if (n.key === e)
        return a.next = n.next, n.next = /** @type {NonNullable<typeof list.next>} */
        t.next, t.next = n, n;
  }, "listGetNode"), hu = /* @__PURE__ */ r(function(t, e) {
    var a = Ht(t, e);
    return a && a.value;
  }, "listGet"), yu = /* @__PURE__ */ r(function(t, e, a) {
    var n = Ht(t, e);
    n ? n.value = a : t.next = /** @type {import('.').ListNode<typeof value>} */
    {
      // eslint-disable-line no-param-reassign, no-extra-parens
      key: e,
      next: t.next,
      value: a
    };
  }, "listSet"), du = /* @__PURE__ */ r(function(t, e) {
    return !!Ht(t, e);
  }, "listHas");
  pa.exports = /* @__PURE__ */ r(function() {
    var e, a, n, o = {
      assert: /* @__PURE__ */ r(function(i) {
        if (!o.has(i))
          throw new iu("Side channel does not contain " + ou(i));
      }, "assert"),
      get: /* @__PURE__ */ r(function(i) {
        if (ct && i && (typeof i == "object" || typeof i == "function")) {
          if (e)
            return uu(e, i);
        } else if (st) {
          if (a)
            return su(a, i);
        } else if (n)
          return hu(n, i);
      }, "get"),
      has: /* @__PURE__ */ r(function(i) {
        if (ct && i && (typeof i == "object" || typeof i == "function")) {
          if (e)
            return cu(e, i);
        } else if (st) {
          if (a)
            return pu(a, i);
        } else if (n)
          return du(n, i);
        return !1;
      }, "has"),
      set: /* @__PURE__ */ r(function(i, l) {
        ct && i && (typeof i == "object" || typeof i == "function") ? (e || (e = new ct()), lu(e, i, l)) : st ? (a || (a = new st()), fu(a, i,
        l)) : (n || (n = { key: {}, next: null }), yu(n, i, l));
      }, "set")
    };
    return o;
  }, "getSideChannel");
});

// ../node_modules/qs/lib/formats.js
var ft = V((mc, ya) => {
  "use strict";
  var mu = String.prototype.replace, gu = /%20/g, zt = {
    RFC1738: "RFC1738",
    RFC3986: "RFC3986"
  };
  ya.exports = {
    default: zt.RFC3986,
    formatters: {
      RFC1738: /* @__PURE__ */ r(function(t) {
        return mu.call(t, gu, "+");
      }, "RFC1738"),
      RFC3986: /* @__PURE__ */ r(function(t) {
        return String(t);
      }, "RFC3986")
    },
    RFC1738: zt.RFC1738,
    RFC3986: zt.RFC3986
  };
});

// ../node_modules/qs/lib/utils.js
var Kt = V((vc, ma) => {
  "use strict";
  var vu = ft(), Gt = Object.prototype.hasOwnProperty, Le = Array.isArray, ge = function() {
    for (var t = [], e = 0; e < 256; ++e)
      t.push("%" + ((e < 16 ? "0" : "") + e.toString(16)).toUpperCase());
    return t;
  }(), bu = /* @__PURE__ */ r(function(e) {
    for (; e.length > 1; ) {
      var a = e.pop(), n = a.obj[a.prop];
      if (Le(n)) {
        for (var o = [], i = 0; i < n.length; ++i)
          typeof n[i] < "u" && o.push(n[i]);
        a.obj[a.prop] = o;
      }
    }
  }, "compactQueue"), da = /* @__PURE__ */ r(function(e, a) {
    for (var n = a && a.plainObjects ? /* @__PURE__ */ Object.create(null) : {}, o = 0; o < e.length; ++o)
      typeof e[o] < "u" && (n[o] = e[o]);
    return n;
  }, "arrayToObject"), Pu = /* @__PURE__ */ r(function t(e, a, n) {
    if (!a)
      return e;
    if (typeof a != "object") {
      if (Le(e))
        e.push(a);
      else if (e && typeof e == "object")
        (n && (n.plainObjects || n.allowPrototypes) || !Gt.call(Object.prototype, a)) && (e[a] = !0);
      else
        return [e, a];
      return e;
    }
    if (!e || typeof e != "object")
      return [e].concat(a);
    var o = e;
    return Le(e) && !Le(a) && (o = da(e, n)), Le(e) && Le(a) ? (a.forEach(function(i, l) {
      if (Gt.call(e, l)) {
        var w = e[l];
        w && typeof w == "object" && i && typeof i == "object" ? e[l] = t(w, i, n) : e.push(i);
      } else
        e[l] = i;
    }), e) : Object.keys(a).reduce(function(i, l) {
      var w = a[l];
      return Gt.call(i, l) ? i[l] = t(i[l], w, n) : i[l] = w, i;
    }, o);
  }, "merge"), wu = /* @__PURE__ */ r(function(e, a) {
    return Object.keys(a).reduce(function(n, o) {
      return n[o] = a[o], n;
    }, e);
  }, "assignSingleSource"), Su = /* @__PURE__ */ r(function(t, e, a) {
    var n = t.replace(/\+/g, " ");
    if (a === "iso-8859-1")
      return n.replace(/%[0-9a-f]{2}/gi, unescape);
    try {
      return decodeURIComponent(n);
    } catch {
      return n;
    }
  }, "decode"), Vt = 1024, Ou = /* @__PURE__ */ r(function(e, a, n, o, i) {
    if (e.length === 0)
      return e;
    var l = e;
    if (typeof e == "symbol" ? l = Symbol.prototype.toString.call(e) : typeof e != "string" && (l = String(e)), n === "iso-8859-1")
      return escape(l).replace(/%u[0-9a-f]{4}/gi, function(S) {
        return "%26%23" + parseInt(S.slice(2), 16) + "%3B";
      });
    for (var w = "", v = 0; v < l.length; v += Vt) {
      for (var I = l.length >= Vt ? l.slice(v, v + Vt) : l, R = [], m = 0; m < I.length; ++m) {
        var u = I.charCodeAt(m);
        if (u === 45 || u === 46 || u === 95 || u === 126 || u >= 48 && u <= 57 || u >= 65 && u <= 90 || u >= 97 && u <= 122 || i === vu.RFC1738 &&
        (u === 40 || u === 41)) {
          R[R.length] = I.charAt(m);
          continue;
        }
        if (u < 128) {
          R[R.length] = ge[u];
          continue;
        }
        if (u < 2048) {
          R[R.length] = ge[192 | u >> 6] + ge[128 | u & 63];
          continue;
        }
        if (u < 55296 || u >= 57344) {
          R[R.length] = ge[224 | u >> 12] + ge[128 | u >> 6 & 63] + ge[128 | u & 63];
          continue;
        }
        m += 1, u = 65536 + ((u & 1023) << 10 | I.charCodeAt(m) & 1023), R[R.length] = ge[240 | u >> 18] + ge[128 | u >> 12 & 63] + ge[128 |
        u >> 6 & 63] + ge[128 | u & 63];
      }
      w += R.join("");
    }
    return w;
  }, "encode"), Eu = /* @__PURE__ */ r(function(e) {
    for (var a = [{ obj: { o: e }, prop: "o" }], n = [], o = 0; o < a.length; ++o)
      for (var i = a[o], l = i.obj[i.prop], w = Object.keys(l), v = 0; v < w.length; ++v) {
        var I = w[v], R = l[I];
        typeof R == "object" && R !== null && n.indexOf(R) === -1 && (a.push({ obj: l, prop: I }), n.push(R));
      }
    return bu(a), e;
  }, "compact"), Au = /* @__PURE__ */ r(function(e) {
    return Object.prototype.toString.call(e) === "[object RegExp]";
  }, "isRegExp"), Ru = /* @__PURE__ */ r(function(e) {
    return !e || typeof e != "object" ? !1 : !!(e.constructor && e.constructor.isBuffer && e.constructor.isBuffer(e));
  }, "isBuffer"), xu = /* @__PURE__ */ r(function(e, a) {
    return [].concat(e, a);
  }, "combine"), Nu = /* @__PURE__ */ r(function(e, a) {
    if (Le(e)) {
      for (var n = [], o = 0; o < e.length; o += 1)
        n.push(a(e[o]));
      return n;
    }
    return a(e);
  }, "maybeMap");
  ma.exports = {
    arrayToObject: da,
    assign: wu,
    combine: xu,
    compact: Eu,
    decode: Su,
    encode: Ou,
    isBuffer: Ru,
    isRegExp: Au,
    maybeMap: Nu,
    merge: Pu
  };
});

// ../node_modules/qs/lib/stringify.js
var Sa = V((Pc, wa) => {
  "use strict";
  var va = ha(), pt = Kt(), Ve = ft(), Iu = Object.prototype.hasOwnProperty, ba = {
    brackets: /* @__PURE__ */ r(function(e) {
      return e + "[]";
    }, "brackets"),
    comma: "comma",
    indices: /* @__PURE__ */ r(function(e, a) {
      return e + "[" + a + "]";
    }, "indices"),
    repeat: /* @__PURE__ */ r(function(e) {
      return e;
    }, "repeat")
  }, ve = Array.isArray, Lu = Array.prototype.push, Pa = /* @__PURE__ */ r(function(t, e) {
    Lu.apply(t, ve(e) ? e : [e]);
  }, "pushToArray"), Mu = Date.prototype.toISOString, ga = Ve.default, ie = {
    addQueryPrefix: !1,
    allowDots: !1,
    allowEmptyArrays: !1,
    arrayFormat: "indices",
    charset: "utf-8",
    charsetSentinel: !1,
    delimiter: "&",
    encode: !0,
    encodeDotInKeys: !1,
    encoder: pt.encode,
    encodeValuesOnly: !1,
    format: ga,
    formatter: Ve.formatters[ga],
    // deprecated
    indices: !1,
    serializeDate: /* @__PURE__ */ r(function(e) {
      return Mu.call(e);
    }, "serializeDate"),
    skipNulls: !1,
    strictNullHandling: !1
  }, Cu = /* @__PURE__ */ r(function(e) {
    return typeof e == "string" || typeof e == "number" || typeof e == "boolean" || typeof e == "symbol" || typeof e == "bigint";
  }, "isNonNullishPrimitive"), Qt = {}, _u = /* @__PURE__ */ r(function t(e, a, n, o, i, l, w, v, I, R, m, u, S, O, $, h, s, b) {
    for (var f = e, L = b, _ = 0, j = !1; (L = L.get(Qt)) !== void 0 && !j; ) {
      var W = L.get(e);
      if (_ += 1, typeof W < "u") {
        if (W === _)
          throw new RangeError("Cyclic object value");
        j = !0;
      }
      typeof L.get(Qt) > "u" && (_ = 0);
    }
    if (typeof R == "function" ? f = R(a, f) : f instanceof Date ? f = S(f) : n === "comma" && ve(f) && (f = pt.maybeMap(f, function(P) {
      return P instanceof Date ? S(P) : P;
    })), f === null) {
      if (l)
        return I && !h ? I(a, ie.encoder, s, "key", O) : a;
      f = "";
    }
    if (Cu(f) || pt.isBuffer(f)) {
      if (I) {
        var U = h ? a : I(a, ie.encoder, s, "key", O);
        return [$(U) + "=" + $(I(f, ie.encoder, s, "value", O))];
      }
      return [$(a) + "=" + $(String(f))];
    }
    var J = [];
    if (typeof f > "u")
      return J;
    var F;
    if (n === "comma" && ve(f))
      h && I && (f = pt.maybeMap(f, I)), F = [{ value: f.length > 0 ? f.join(",") || null : void 0 }];
    else if (ve(R))
      F = R;
    else {
      var K = Object.keys(f);
      F = m ? K.sort(m) : K;
    }
    var G = v ? a.replace(/\./g, "%2E") : a, N = o && ve(f) && f.length === 1 ? G + "[]" : G;
    if (i && ve(f) && f.length === 0)
      return N + "[]";
    for (var E = 0; E < F.length; ++E) {
      var B = F[E], c = typeof B == "object" && typeof B.value < "u" ? B.value : f[B];
      if (!(w && c === null)) {
        var p = u && v ? B.replace(/\./g, "%2E") : B, A = ve(f) ? typeof n == "function" ? n(N, p) : N : N + (u ? "." + p : "[" + p + "]");
        b.set(e, _);
        var g = va();
        g.set(Qt, b), Pa(J, t(
          c,
          A,
          n,
          o,
          i,
          l,
          w,
          v,
          n === "comma" && h && ve(f) ? null : I,
          R,
          m,
          u,
          S,
          O,
          $,
          h,
          s,
          g
        ));
      }
    }
    return J;
  }, "stringify"), Tu = /* @__PURE__ */ r(function(e) {
    if (!e)
      return ie;
    if (typeof e.allowEmptyArrays < "u" && typeof e.allowEmptyArrays != "boolean")
      throw new TypeError("`allowEmptyArrays` option can only be `true` or `false`, when provided");
    if (typeof e.encodeDotInKeys < "u" && typeof e.encodeDotInKeys != "boolean")
      throw new TypeError("`encodeDotInKeys` option can only be `true` or `false`, when provided");
    if (e.encoder !== null && typeof e.encoder < "u" && typeof e.encoder != "function")
      throw new TypeError("Encoder has to be a function.");
    var a = e.charset || ie.charset;
    if (typeof e.charset < "u" && e.charset !== "utf-8" && e.charset !== "iso-8859-1")
      throw new TypeError("The charset option must be either utf-8, iso-8859-1, or undefined");
    var n = Ve.default;
    if (typeof e.format < "u") {
      if (!Iu.call(Ve.formatters, e.format))
        throw new TypeError("Unknown format option provided.");
      n = e.format;
    }
    var o = Ve.formatters[n], i = ie.filter;
    (typeof e.filter == "function" || ve(e.filter)) && (i = e.filter);
    var l;
    if (e.arrayFormat in ba ? l = e.arrayFormat : "indices" in e ? l = e.indices ? "indices" : "repeat" : l = ie.arrayFormat, "commaRoundTri\
p" in e && typeof e.commaRoundTrip != "boolean")
      throw new TypeError("`commaRoundTrip` must be a boolean, or absent");
    var w = typeof e.allowDots > "u" ? e.encodeDotInKeys === !0 ? !0 : ie.allowDots : !!e.allowDots;
    return {
      addQueryPrefix: typeof e.addQueryPrefix == "boolean" ? e.addQueryPrefix : ie.addQueryPrefix,
      allowDots: w,
      allowEmptyArrays: typeof e.allowEmptyArrays == "boolean" ? !!e.allowEmptyArrays : ie.allowEmptyArrays,
      arrayFormat: l,
      charset: a,
      charsetSentinel: typeof e.charsetSentinel == "boolean" ? e.charsetSentinel : ie.charsetSentinel,
      commaRoundTrip: e.commaRoundTrip,
      delimiter: typeof e.delimiter > "u" ? ie.delimiter : e.delimiter,
      encode: typeof e.encode == "boolean" ? e.encode : ie.encode,
      encodeDotInKeys: typeof e.encodeDotInKeys == "boolean" ? e.encodeDotInKeys : ie.encodeDotInKeys,
      encoder: typeof e.encoder == "function" ? e.encoder : ie.encoder,
      encodeValuesOnly: typeof e.encodeValuesOnly == "boolean" ? e.encodeValuesOnly : ie.encodeValuesOnly,
      filter: i,
      format: n,
      formatter: o,
      serializeDate: typeof e.serializeDate == "function" ? e.serializeDate : ie.serializeDate,
      skipNulls: typeof e.skipNulls == "boolean" ? e.skipNulls : ie.skipNulls,
      sort: typeof e.sort == "function" ? e.sort : null,
      strictNullHandling: typeof e.strictNullHandling == "boolean" ? e.strictNullHandling : ie.strictNullHandling
    };
  }, "normalizeStringifyOptions");
  wa.exports = function(t, e) {
    var a = t, n = Tu(e), o, i;
    typeof n.filter == "function" ? (i = n.filter, a = i("", a)) : ve(n.filter) && (i = n.filter, o = i);
    var l = [];
    if (typeof a != "object" || a === null)
      return "";
    var w = ba[n.arrayFormat], v = w === "comma" && n.commaRoundTrip;
    o || (o = Object.keys(a)), n.sort && o.sort(n.sort);
    for (var I = va(), R = 0; R < o.length; ++R) {
      var m = o[R];
      n.skipNulls && a[m] === null || Pa(l, _u(
        a[m],
        m,
        w,
        v,
        n.allowEmptyArrays,
        n.strictNullHandling,
        n.skipNulls,
        n.encodeDotInKeys,
        n.encode ? n.encoder : null,
        n.filter,
        n.sort,
        n.allowDots,
        n.serializeDate,
        n.format,
        n.formatter,
        n.encodeValuesOnly,
        n.charset,
        I
      ));
    }
    var u = l.join(n.delimiter), S = n.addQueryPrefix === !0 ? "?" : "";
    return n.charsetSentinel && (n.charset === "iso-8859-1" ? S += "utf8=%26%2310003%3B&" : S += "utf8=%E2%9C%93&"), u.length > 0 ? S + u : "";
  };
});

// ../node_modules/qs/lib/parse.js
var Aa = V((Sc, Ea) => {
  "use strict";
  var qe = Kt(), Jt = Object.prototype.hasOwnProperty, ju = Array.isArray, oe = {
    allowDots: !1,
    allowEmptyArrays: !1,
    allowPrototypes: !1,
    allowSparse: !1,
    arrayLimit: 20,
    charset: "utf-8",
    charsetSentinel: !1,
    comma: !1,
    decodeDotInKeys: !1,
    decoder: qe.decode,
    delimiter: "&",
    depth: 5,
    duplicates: "combine",
    ignoreQueryPrefix: !1,
    interpretNumericEntities: !1,
    parameterLimit: 1e3,
    parseArrays: !0,
    plainObjects: !1,
    strictNullHandling: !1
  }, Fu = /* @__PURE__ */ r(function(t) {
    return t.replace(/&#(\d+);/g, function(e, a) {
      return String.fromCharCode(parseInt(a, 10));
    });
  }, "interpretNumericEntities"), Oa = /* @__PURE__ */ r(function(t, e) {
    return t && typeof t == "string" && e.comma && t.indexOf(",") > -1 ? t.split(",") : t;
  }, "parseArrayValue"), ku = "utf8=%26%2310003%3B", $u = "utf8=%E2%9C%93", Du = /* @__PURE__ */ r(function(e, a) {
    var n = { __proto__: null }, o = a.ignoreQueryPrefix ? e.replace(/^\?/, "") : e;
    o = o.replace(/%5B/gi, "[").replace(/%5D/gi, "]");
    var i = a.parameterLimit === 1 / 0 ? void 0 : a.parameterLimit, l = o.split(a.delimiter, i), w = -1, v, I = a.charset;
    if (a.charsetSentinel)
      for (v = 0; v < l.length; ++v)
        l[v].indexOf("utf8=") === 0 && (l[v] === $u ? I = "utf-8" : l[v] === ku && (I = "iso-8859-1"), w = v, v = l.length);
    for (v = 0; v < l.length; ++v)
      if (v !== w) {
        var R = l[v], m = R.indexOf("]="), u = m === -1 ? R.indexOf("=") : m + 1, S, O;
        u === -1 ? (S = a.decoder(R, oe.decoder, I, "key"), O = a.strictNullHandling ? null : "") : (S = a.decoder(R.slice(0, u), oe.decoder,
        I, "key"), O = qe.maybeMap(
          Oa(R.slice(u + 1), a),
          function(h) {
            return a.decoder(h, oe.decoder, I, "value");
          }
        )), O && a.interpretNumericEntities && I === "iso-8859-1" && (O = Fu(O)), R.indexOf("[]=") > -1 && (O = ju(O) ? [O] : O);
        var $ = Jt.call(n, S);
        $ && a.duplicates === "combine" ? n[S] = qe.combine(n[S], O) : (!$ || a.duplicates === "last") && (n[S] = O);
      }
    return n;
  }, "parseQueryStringValues"), qu = /* @__PURE__ */ r(function(t, e, a, n) {
    for (var o = n ? e : Oa(e, a), i = t.length - 1; i >= 0; --i) {
      var l, w = t[i];
      if (w === "[]" && a.parseArrays)
        l = a.allowEmptyArrays && (o === "" || a.strictNullHandling && o === null) ? [] : [].concat(o);
      else {
        l = a.plainObjects ? /* @__PURE__ */ Object.create(null) : {};
        var v = w.charAt(0) === "[" && w.charAt(w.length - 1) === "]" ? w.slice(1, -1) : w, I = a.decodeDotInKeys ? v.replace(/%2E/g, ".") :
        v, R = parseInt(I, 10);
        !a.parseArrays && I === "" ? l = { 0: o } : !isNaN(R) && w !== I && String(R) === I && R >= 0 && a.parseArrays && R <= a.arrayLimit ?
        (l = [], l[R] = o) : I !== "__proto__" && (l[I] = o);
      }
      o = l;
    }
    return o;
  }, "parseObject"), Bu = /* @__PURE__ */ r(function(e, a, n, o) {
    if (e) {
      var i = n.allowDots ? e.replace(/\.([^.[]+)/g, "[$1]") : e, l = /(\[[^[\]]*])/, w = /(\[[^[\]]*])/g, v = n.depth > 0 && l.exec(i), I = v ?
      i.slice(0, v.index) : i, R = [];
      if (I) {
        if (!n.plainObjects && Jt.call(Object.prototype, I) && !n.allowPrototypes)
          return;
        R.push(I);
      }
      for (var m = 0; n.depth > 0 && (v = w.exec(i)) !== null && m < n.depth; ) {
        if (m += 1, !n.plainObjects && Jt.call(Object.prototype, v[1].slice(1, -1)) && !n.allowPrototypes)
          return;
        R.push(v[1]);
      }
      return v && R.push("[" + i.slice(v.index) + "]"), qu(R, a, n, o);
    }
  }, "parseQueryStringKeys"), Wu = /* @__PURE__ */ r(function(e) {
    if (!e)
      return oe;
    if (typeof e.allowEmptyArrays < "u" && typeof e.allowEmptyArrays != "boolean")
      throw new TypeError("`allowEmptyArrays` option can only be `true` or `false`, when provided");
    if (typeof e.decodeDotInKeys < "u" && typeof e.decodeDotInKeys != "boolean")
      throw new TypeError("`decodeDotInKeys` option can only be `true` or `false`, when provided");
    if (e.decoder !== null && typeof e.decoder < "u" && typeof e.decoder != "function")
      throw new TypeError("Decoder has to be a function.");
    if (typeof e.charset < "u" && e.charset !== "utf-8" && e.charset !== "iso-8859-1")
      throw new TypeError("The charset option must be either utf-8, iso-8859-1, or undefined");
    var a = typeof e.charset > "u" ? oe.charset : e.charset, n = typeof e.duplicates > "u" ? oe.duplicates : e.duplicates;
    if (n !== "combine" && n !== "first" && n !== "last")
      throw new TypeError("The duplicates option must be either combine, first, or last");
    var o = typeof e.allowDots > "u" ? e.decodeDotInKeys === !0 ? !0 : oe.allowDots : !!e.allowDots;
    return {
      allowDots: o,
      allowEmptyArrays: typeof e.allowEmptyArrays == "boolean" ? !!e.allowEmptyArrays : oe.allowEmptyArrays,
      allowPrototypes: typeof e.allowPrototypes == "boolean" ? e.allowPrototypes : oe.allowPrototypes,
      allowSparse: typeof e.allowSparse == "boolean" ? e.allowSparse : oe.allowSparse,
      arrayLimit: typeof e.arrayLimit == "number" ? e.arrayLimit : oe.arrayLimit,
      charset: a,
      charsetSentinel: typeof e.charsetSentinel == "boolean" ? e.charsetSentinel : oe.charsetSentinel,
      comma: typeof e.comma == "boolean" ? e.comma : oe.comma,
      decodeDotInKeys: typeof e.decodeDotInKeys == "boolean" ? e.decodeDotInKeys : oe.decodeDotInKeys,
      decoder: typeof e.decoder == "function" ? e.decoder : oe.decoder,
      delimiter: typeof e.delimiter == "string" || qe.isRegExp(e.delimiter) ? e.delimiter : oe.delimiter,
      // eslint-disable-next-line no-implicit-coercion, no-extra-parens
      depth: typeof e.depth == "number" || e.depth === !1 ? +e.depth : oe.depth,
      duplicates: n,
      ignoreQueryPrefix: e.ignoreQueryPrefix === !0,
      interpretNumericEntities: typeof e.interpretNumericEntities == "boolean" ? e.interpretNumericEntities : oe.interpretNumericEntities,
      parameterLimit: typeof e.parameterLimit == "number" ? e.parameterLimit : oe.parameterLimit,
      parseArrays: e.parseArrays !== !1,
      plainObjects: typeof e.plainObjects == "boolean" ? e.plainObjects : oe.plainObjects,
      strictNullHandling: typeof e.strictNullHandling == "boolean" ? e.strictNullHandling : oe.strictNullHandling
    };
  }, "normalizeParseOptions");
  Ea.exports = function(t, e) {
    var a = Wu(e);
    if (t === "" || t === null || typeof t > "u")
      return a.plainObjects ? /* @__PURE__ */ Object.create(null) : {};
    for (var n = typeof t == "string" ? Du(t, a) : t, o = a.plainObjects ? /* @__PURE__ */ Object.create(null) : {}, i = Object.keys(n), l = 0; l <
    i.length; ++l) {
      var w = i[l], v = Bu(w, n[w], a, typeof t == "string");
      o = qe.merge(o, v, a);
    }
    return a.allowSparse === !0 ? o : qe.compact(o);
  };
});

// ../node_modules/qs/lib/index.js
var xa = V((Ec, Ra) => {
  "use strict";
  var Uu = Sa(), Hu = Aa(), zu = ft();
  Ra.exports = {
    formats: zu,
    parse: Hu,
    stringify: Uu
  };
});

// ../node_modules/ts-dedent/dist/index.js
var Ia = V((Ke) => {
  "use strict";
  Object.defineProperty(Ke, "__esModule", { value: !0 });
  Ke.dedent = void 0;
  function Na(t) {
    for (var e = [], a = 1; a < arguments.length; a++)
      e[a - 1] = arguments[a];
    var n = Array.from(typeof t == "string" ? [t] : t);
    n[n.length - 1] = n[n.length - 1].replace(/\r?\n([\t ]*)$/, "");
    var o = n.reduce(function(w, v) {
      var I = v.match(/\n([\t ]+|(?!\s).)/g);
      return I ? w.concat(I.map(function(R) {
        var m, u;
        return (u = (m = R.match(/[\t ]/g)) === null || m === void 0 ? void 0 : m.length) !== null && u !== void 0 ? u : 0;
      })) : w;
    }, []);
    if (o.length) {
      var i = new RegExp(`
[	 ]{` + Math.min.apply(Math, o) + "}", "g");
      n = n.map(function(w) {
        return w.replace(i, `
`);
      });
    }
    n[0] = n[0].replace(/^\r?\n/, "");
    var l = n[0];
    return e.forEach(function(w, v) {
      var I = l.match(/(?:^|\n)( *)$/), R = I ? I[1] : "", m = w;
      typeof w == "string" && w.includes(`
`) && (m = String(w).split(`
`).map(function(u, S) {
        return S === 0 ? u : "" + R + u;
      }).join(`
`)), l += m + n[v + 1];
    }), l;
  }
  r(Na, "dedent");
  Ke.dedent = Na;
  Ke.default = Na;
});

// ../node_modules/@storybook/global/dist/index.js
var ka = V((Lc, Fa) => {
  "use strict";
  var tr = Object.defineProperty, Xu = Object.getOwnPropertyDescriptor, Zu = Object.getOwnPropertyNames, el = Object.prototype.hasOwnProperty,
  tl = /* @__PURE__ */ r((t, e) => {
    for (var a in e)
      tr(t, a, { get: e[a], enumerable: !0 });
  }, "__export"), rl = /* @__PURE__ */ r((t, e, a, n) => {
    if (e && typeof e == "object" || typeof e == "function")
      for (let o of Zu(e))
        !el.call(t, o) && o !== a && tr(t, o, { get: /* @__PURE__ */ r(() => e[o], "get"), enumerable: !(n = Xu(e, o)) || n.enumerable });
    return t;
  }, "__copyProps"), nl = /* @__PURE__ */ r((t) => rl(tr({}, "__esModule", { value: !0 }), t), "__toCommonJS"), ja = {};
  tl(ja, {
    global: /* @__PURE__ */ r(() => al, "global")
  });
  Fa.exports = nl(ja);
  var al = (() => {
    let t;
    return typeof window < "u" ? t = window : typeof globalThis < "u" ? t = globalThis : typeof global < "u" ? t = global : typeof self < "u" ?
    t = self : t = {}, t;
  })();
});

// ../node_modules/history/umd/history.production.min.js
var Da = V((gt, $a) => {
  (function(t, e) {
    typeof gt == "object" && typeof $a < "u" ? e(gt) : typeof define == "function" && define.amd ? define(["exports"], e) : e((t = typeof globalThis <
    "u" ? globalThis : t || self).HistoryLibrary = {});
  })(gt, function(t) {
    "use strict";
    function e() {
      return (e = Object.assign || function(m) {
        for (var u = 1; u < arguments.length; u++) {
          var S = arguments[u];
          for (var O in S) Object.prototype.hasOwnProperty.call(S, O) && (m[O] = S[O]);
        }
        return m;
      }).apply(this, arguments);
    }
    r(e, "n");
    var a;
    t.Action = void 0, (a = t.Action || (t.Action = {})).Pop = "POP", a.Push = "PUSH", a.Replace = "REPLACE";
    var n = "beforeunload", o = "popstate";
    function i(m, u, S) {
      return Math.min(Math.max(m, u), S);
    }
    r(i, "a");
    function l(m) {
      m.preventDefault(), m.returnValue = "";
    }
    r(l, "i");
    function w() {
      var m = [];
      return { get length() {
        return m.length;
      }, push: /* @__PURE__ */ r(function(u) {
        return m.push(u), function() {
          m = m.filter(function(S) {
            return S !== u;
          });
        };
      }, "push"), call: /* @__PURE__ */ r(function(u) {
        m.forEach(function(S) {
          return S && S(u);
        });
      }, "call") };
    }
    r(w, "c");
    function v() {
      return Math.random().toString(36).substr(2, 8);
    }
    r(v, "u");
    function I(m) {
      var u = m.pathname, S = u === void 0 ? "/" : u, O = m.search, $ = O === void 0 ? "" : O, h = m.hash, s = h === void 0 ? "" : h;
      return $ && $ !== "?" && (S += $.charAt(0) === "?" ? $ : "?" + $), s && s !== "#" && (S += s.charAt(0) === "#" ? s : "#" + s), S;
    }
    r(I, "l");
    function R(m) {
      var u = {};
      if (m) {
        var S = m.indexOf("#");
        S >= 0 && (u.hash = m.substr(S), m = m.substr(0, S));
        var O = m.indexOf("?");
        O >= 0 && (u.search = m.substr(O), m = m.substr(0, O)), m && (u.pathname = m);
      }
      return u;
    }
    r(R, "f"), t.createBrowserHistory = function(m) {
      m === void 0 && (m = {});
      var u = m.window, S = u === void 0 ? document.defaultView : u, O = S.history;
      function $() {
        var N = S.location, E = N.pathname, B = N.search, c = N.hash, p = O.state || {};
        return [p.idx, { pathname: E, search: B, hash: c, state: p.usr || null, key: p.key || "default" }];
      }
      r($, "p");
      var h = null;
      S.addEventListener(o, function() {
        if (h) j.call(h), h = null;
        else {
          var N = t.Action.Pop, E = $(), B = E[0], c = E[1];
          if (j.length) {
            if (B != null) {
              var p = f - B;
              p && (h = { action: N, location: c, retry: /* @__PURE__ */ r(function() {
                G(-1 * p);
              }, "retry") }, G(p));
            }
          } else K(N);
        }
      });
      var s = t.Action.Pop, b = $(), f = b[0], L = b[1], _ = w(), j = w();
      function W(N) {
        return typeof N == "string" ? N : I(N);
      }
      r(W, "P");
      function U(N, E) {
        return E === void 0 && (E = null), e({ pathname: L.pathname, hash: "", search: "" }, typeof N == "string" ? R(N) : N, { state: E, key: v() });
      }
      r(U, "k");
      function J(N, E) {
        return [{ usr: N.state, key: N.key, idx: E }, W(N)];
      }
      r(J, "x");
      function F(N, E, B) {
        return !j.length || (j.call({ action: N, location: E, retry: B }), !1);
      }
      r(F, "w");
      function K(N) {
        s = N;
        var E = $();
        f = E[0], L = E[1], _.call({ action: s, location: L });
      }
      r(K, "E");
      function G(N) {
        O.go(N);
      }
      return r(G, "H"), f == null && (f = 0, O.replaceState(e({}, O.state, { idx: f }), "")), { get action() {
        return s;
      }, get location() {
        return L;
      }, createHref: W, push: /* @__PURE__ */ r(function N(E, B) {
        var c = t.Action.Push, p = U(E, B);
        if (F(c, p, function() {
          N(E, B);
        })) {
          var A = J(p, f + 1), g = A[0], P = A[1];
          try {
            O.pushState(g, "", P);
          } catch {
            S.location.assign(P);
          }
          K(c);
        }
      }, "n"), replace: /* @__PURE__ */ r(function N(E, B) {
        var c = t.Action.Replace, p = U(E, B);
        if (F(c, p, function() {
          N(E, B);
        })) {
          var A = J(p, f), g = A[0], P = A[1];
          O.replaceState(g, "", P), K(c);
        }
      }, "n"), go: G, back: /* @__PURE__ */ r(function() {
        G(-1);
      }, "back"), forward: /* @__PURE__ */ r(function() {
        G(1);
      }, "forward"), listen: /* @__PURE__ */ r(function(N) {
        return _.push(N);
      }, "listen"), block: /* @__PURE__ */ r(function(N) {
        var E = j.push(N);
        return j.length === 1 && S.addEventListener(n, l), function() {
          E(), j.length || S.removeEventListener(n, l);
        };
      }, "block") };
    }, t.createHashHistory = function(m) {
      m === void 0 && (m = {});
      var u = m.window, S = u === void 0 ? document.defaultView : u, O = S.history;
      function $() {
        var E = R(S.location.hash.substr(1)), B = E.pathname, c = B === void 0 ? "/" : B, p = E.search, A = p === void 0 ? "" : p, g = E.hash,
        P = g === void 0 ? "" : g, D = O.state || {};
        return [D.idx, { pathname: c, search: A, hash: P, state: D.usr || null, key: D.key || "default" }];
      }
      r($, "p");
      var h = null;
      function s() {
        if (h) W.call(h), h = null;
        else {
          var E = t.Action.Pop, B = $(), c = B[0], p = B[1];
          if (W.length) {
            if (c != null) {
              var A = L - c;
              A && (h = { action: E, location: p, retry: /* @__PURE__ */ r(function() {
                N(-1 * A);
              }, "retry") }, N(A));
            }
          } else G(E);
        }
      }
      r(s, "d"), S.addEventListener(o, s), S.addEventListener("hashchange", function() {
        I($()[1]) !== I(_) && s();
      });
      var b = t.Action.Pop, f = $(), L = f[0], _ = f[1], j = w(), W = w();
      function U(E) {
        return function() {
          var B = document.querySelector("base"), c = "";
          if (B && B.getAttribute("href")) {
            var p = S.location.href, A = p.indexOf("#");
            c = A === -1 ? p : p.slice(0, A);
          }
          return c;
        }() + "#" + (typeof E == "string" ? E : I(E));
      }
      r(U, "k");
      function J(E, B) {
        return B === void 0 && (B = null), e({ pathname: _.pathname, hash: "", search: "" }, typeof E == "string" ? R(E) : E, { state: B, key: v() });
      }
      r(J, "x");
      function F(E, B) {
        return [{ usr: E.state, key: E.key, idx: B }, U(E)];
      }
      r(F, "w");
      function K(E, B, c) {
        return !W.length || (W.call({ action: E, location: B, retry: c }), !1);
      }
      r(K, "E");
      function G(E) {
        b = E;
        var B = $();
        L = B[0], _ = B[1], j.call({ action: b, location: _ });
      }
      r(G, "H");
      function N(E) {
        O.go(E);
      }
      return r(N, "L"), L == null && (L = 0, O.replaceState(e({}, O.state, { idx: L }), "")), { get action() {
        return b;
      }, get location() {
        return _;
      }, createHref: U, push: /* @__PURE__ */ r(function E(B, c) {
        var p = t.Action.Push, A = J(B, c);
        if (K(p, A, function() {
          E(B, c);
        })) {
          var g = F(A, L + 1), P = g[0], D = g[1];
          try {
            O.pushState(P, "", D);
          } catch {
            S.location.assign(D);
          }
          G(p);
        }
      }, "n"), replace: /* @__PURE__ */ r(function E(B, c) {
        var p = t.Action.Replace, A = J(B, c);
        if (K(p, A, function() {
          E(B, c);
        })) {
          var g = F(A, L), P = g[0], D = g[1];
          O.replaceState(P, "", D), G(p);
        }
      }, "n"), go: N, back: /* @__PURE__ */ r(function() {
        N(-1);
      }, "back"), forward: /* @__PURE__ */ r(function() {
        N(1);
      }, "forward"), listen: /* @__PURE__ */ r(function(E) {
        return j.push(E);
      }, "listen"), block: /* @__PURE__ */ r(function(E) {
        var B = W.push(E);
        return W.length === 1 && S.addEventListener(n, l), function() {
          B(), W.length || S.removeEventListener(n, l);
        };
      }, "block") };
    }, t.createMemoryHistory = function(m) {
      m === void 0 && (m = {});
      var u = m, S = u.initialEntries, O = S === void 0 ? ["/"] : S, $ = u.initialIndex, h = O.map(function(F) {
        return e({ pathname: "/", search: "", hash: "", state: null, key: v() }, typeof F == "string" ? R(F) : F);
      }), s = i($ ?? h.length - 1, 0, h.length - 1), b = t.Action.Pop, f = h[s], L = w(), _ = w();
      function j(F, K) {
        return K === void 0 && (K = null), e({ pathname: f.pathname, search: "", hash: "" }, typeof F == "string" ? R(F) : F, { state: K, key: v() });
      }
      r(j, "m");
      function W(F, K, G) {
        return !_.length || (_.call({ action: F, location: K, retry: G }), !1);
      }
      r(W, "b");
      function U(F, K) {
        b = F, f = K, L.call({ action: b, location: f });
      }
      r(U, "A");
      function J(F) {
        var K = i(s + F, 0, h.length - 1), G = t.Action.Pop, N = h[K];
        W(G, N, function() {
          J(F);
        }) && (s = K, U(G, N));
      }
      return r(J, "P"), { get index() {
        return s;
      }, get action() {
        return b;
      }, get location() {
        return f;
      }, createHref: /* @__PURE__ */ r(function(F) {
        return typeof F == "string" ? F : I(F);
      }, "createHref"), push: /* @__PURE__ */ r(function F(K, G) {
        var N = t.Action.Push, E = j(K, G);
        W(N, E, function() {
          F(K, G);
        }) && (s += 1, h.splice(s, h.length, E), U(N, E));
      }, "n"), replace: /* @__PURE__ */ r(function F(K, G) {
        var N = t.Action.Replace, E = j(K, G);
        W(N, E, function() {
          F(K, G);
        }) && (h[s] = E, U(N, E));
      }, "n"), go: J, back: /* @__PURE__ */ r(function() {
        J(-1);
      }, "back"), forward: /* @__PURE__ */ r(function() {
        J(1);
      }, "forward"), listen: /* @__PURE__ */ r(function(F) {
        return L.push(F);
      }, "listen"), block: /* @__PURE__ */ r(function(F) {
        return _.push(F);
      }, "block") };
    }, t.createPath = I, t.parsePath = R, Object.defineProperty(t, "__esModule", { value: !0 });
  });
});

// ../node_modules/history/umd/history.development.js
var Ba = V((vt, qa) => {
  (function(t, e) {
    typeof vt == "object" && typeof qa < "u" ? e(vt) : typeof define == "function" && define.amd ? define(["exports"], e) : (t = typeof globalThis <
    "u" ? globalThis : t || self, e(t.HistoryLibrary = {}));
  })(vt, function(t) {
    "use strict";
    function e() {
      return e = Object.assign || function(h) {
        for (var s = 1; s < arguments.length; s++) {
          var b = arguments[s];
          for (var f in b)
            Object.prototype.hasOwnProperty.call(b, f) && (h[f] = b[f]);
        }
        return h;
      }, e.apply(this, arguments);
    }
    r(e, "_extends"), t.Action = void 0, function(h) {
      h.Pop = "POP", h.Push = "PUSH", h.Replace = "REPLACE";
    }(t.Action || (t.Action = {}));
    var a = /* @__PURE__ */ r(function(h) {
      return Object.freeze(h);
    }, "readOnly");
    function n(h, s) {
      if (!h) {
        typeof console < "u" && console.warn(s);
        try {
          throw new Error(s);
        } catch {
        }
      }
    }
    r(n, "warning");
    var o = "beforeunload", i = "hashchange", l = "popstate";
    function w(h) {
      h === void 0 && (h = {});
      var s = h, b = s.window, f = b === void 0 ? document.defaultView : b, L = f.history;
      function _() {
        var T = f.location, d = T.pathname, M = T.search, Q = T.hash, z = L.state || {};
        return [z.idx, a({
          pathname: d,
          search: M,
          hash: Q,
          state: z.usr || null,
          key: z.key || "default"
        })];
      }
      r(_, "getIndexAndLocation");
      var j = null;
      function W() {
        if (j)
          N.call(j), j = null;
        else {
          var T = t.Action.Pop, d = _(), M = d[0], Q = d[1];
          if (N.length)
            if (M != null) {
              var z = F - M;
              z && (j = {
                action: T,
                location: Q,
                retry: /* @__PURE__ */ r(function() {
                  D(z * -1);
                }, "retry")
              }, D(z));
            } else
              n(
                !1,
                // TODO: Write up a doc that explains our blocking strategy in
                // detail and link to it here so people can understand better what
                // is going on and how to avoid it.
                "You are trying to block a POP navigation to a location that was not created by the history library. The block will fail sil\
ently in production, but in general you should do all navigation with the history library (instead of using window.history.pushState directl\
y) to avoid this situation."
              );
          else
            A(T);
        }
      }
      r(W, "handlePop"), f.addEventListener(l, W);
      var U = t.Action.Pop, J = _(), F = J[0], K = J[1], G = u(), N = u();
      F == null && (F = 0, L.replaceState(e({}, L.state, {
        idx: F
      }), ""));
      function E(T) {
        return typeof T == "string" ? T : O(T);
      }
      r(E, "createHref");
      function B(T, d) {
        return d === void 0 && (d = null), a(e({
          pathname: K.pathname,
          hash: "",
          search: ""
        }, typeof T == "string" ? $(T) : T, {
          state: d,
          key: S()
        }));
      }
      r(B, "getNextLocation");
      function c(T, d) {
        return [{
          usr: T.state,
          key: T.key,
          idx: d
        }, E(T)];
      }
      r(c, "getHistoryStateAndUrl");
      function p(T, d, M) {
        return !N.length || (N.call({
          action: T,
          location: d,
          retry: M
        }), !1);
      }
      r(p, "allowTx");
      function A(T) {
        U = T;
        var d = _();
        F = d[0], K = d[1], G.call({
          action: U,
          location: K
        });
      }
      r(A, "applyTx");
      function g(T, d) {
        var M = t.Action.Push, Q = B(T, d);
        function z() {
          g(T, d);
        }
        if (r(z, "retry"), p(M, Q, z)) {
          var te = c(Q, F + 1), ae = te[0], ue = te[1];
          try {
            L.pushState(ae, "", ue);
          } catch {
            f.location.assign(ue);
          }
          A(M);
        }
      }
      r(g, "push");
      function P(T, d) {
        var M = t.Action.Replace, Q = B(T, d);
        function z() {
          P(T, d);
        }
        if (r(z, "retry"), p(M, Q, z)) {
          var te = c(Q, F), ae = te[0], ue = te[1];
          L.replaceState(ae, "", ue), A(M);
        }
      }
      r(P, "replace");
      function D(T) {
        L.go(T);
      }
      r(D, "go");
      var k = {
        get action() {
          return U;
        },
        get location() {
          return K;
        },
        createHref: E,
        push: g,
        replace: P,
        go: D,
        back: /* @__PURE__ */ r(function() {
          D(-1);
        }, "back"),
        forward: /* @__PURE__ */ r(function() {
          D(1);
        }, "forward"),
        listen: /* @__PURE__ */ r(function(d) {
          return G.push(d);
        }, "listen"),
        block: /* @__PURE__ */ r(function(d) {
          var M = N.push(d);
          return N.length === 1 && f.addEventListener(o, m), function() {
            M(), N.length || f.removeEventListener(o, m);
          };
        }, "block")
      };
      return k;
    }
    r(w, "createBrowserHistory");
    function v(h) {
      h === void 0 && (h = {});
      var s = h, b = s.window, f = b === void 0 ? document.defaultView : b, L = f.history;
      function _() {
        var d = $(f.location.hash.substr(1)), M = d.pathname, Q = M === void 0 ? "/" : M, z = d.search, te = z === void 0 ? "" : z, ae = d.hash,
        ue = ae === void 0 ? "" : ae, fe = L.state || {};
        return [fe.idx, a({
          pathname: Q,
          search: te,
          hash: ue,
          state: fe.usr || null,
          key: fe.key || "default"
        })];
      }
      r(_, "getIndexAndLocation");
      var j = null;
      function W() {
        if (j)
          N.call(j), j = null;
        else {
          var d = t.Action.Pop, M = _(), Q = M[0], z = M[1];
          if (N.length)
            if (Q != null) {
              var te = F - Q;
              te && (j = {
                action: d,
                location: z,
                retry: /* @__PURE__ */ r(function() {
                  k(te * -1);
                }, "retry")
              }, k(te));
            } else
              n(
                !1,
                // TODO: Write up a doc that explains our blocking strategy in
                // detail and link to it here so people can understand better
                // what is going on and how to avoid it.
                "You are trying to block a POP navigation to a location that was not created by the history library. The block will fail sil\
ently in production, but in general you should do all navigation with the history library (instead of using window.history.pushState directl\
y) to avoid this situation."
              );
          else
            g(d);
        }
      }
      r(W, "handlePop"), f.addEventListener(l, W), f.addEventListener(i, function() {
        var d = _(), M = d[1];
        O(M) !== O(K) && W();
      });
      var U = t.Action.Pop, J = _(), F = J[0], K = J[1], G = u(), N = u();
      F == null && (F = 0, L.replaceState(e({}, L.state, {
        idx: F
      }), ""));
      function E() {
        var d = document.querySelector("base"), M = "";
        if (d && d.getAttribute("href")) {
          var Q = f.location.href, z = Q.indexOf("#");
          M = z === -1 ? Q : Q.slice(0, z);
        }
        return M;
      }
      r(E, "getBaseHref");
      function B(d) {
        return E() + "#" + (typeof d == "string" ? d : O(d));
      }
      r(B, "createHref");
      function c(d, M) {
        return M === void 0 && (M = null), a(e({
          pathname: K.pathname,
          hash: "",
          search: ""
        }, typeof d == "string" ? $(d) : d, {
          state: M,
          key: S()
        }));
      }
      r(c, "getNextLocation");
      function p(d, M) {
        return [{
          usr: d.state,
          key: d.key,
          idx: M
        }, B(d)];
      }
      r(p, "getHistoryStateAndUrl");
      function A(d, M, Q) {
        return !N.length || (N.call({
          action: d,
          location: M,
          retry: Q
        }), !1);
      }
      r(A, "allowTx");
      function g(d) {
        U = d;
        var M = _();
        F = M[0], K = M[1], G.call({
          action: U,
          location: K
        });
      }
      r(g, "applyTx");
      function P(d, M) {
        var Q = t.Action.Push, z = c(d, M);
        function te() {
          P(d, M);
        }
        if (r(te, "retry"), n(z.pathname.charAt(0) === "/", "Relative pathnames are not supported in hash history.push(" + JSON.stringify(d) +
        ")"), A(Q, z, te)) {
          var ae = p(z, F + 1), ue = ae[0], fe = ae[1];
          try {
            L.pushState(ue, "", fe);
          } catch {
            f.location.assign(fe);
          }
          g(Q);
        }
      }
      r(P, "push");
      function D(d, M) {
        var Q = t.Action.Replace, z = c(d, M);
        function te() {
          D(d, M);
        }
        if (r(te, "retry"), n(z.pathname.charAt(0) === "/", "Relative pathnames are not supported in hash history.replace(" + JSON.stringify(
        d) + ")"), A(Q, z, te)) {
          var ae = p(z, F), ue = ae[0], fe = ae[1];
          L.replaceState(ue, "", fe), g(Q);
        }
      }
      r(D, "replace");
      function k(d) {
        L.go(d);
      }
      r(k, "go");
      var T = {
        get action() {
          return U;
        },
        get location() {
          return K;
        },
        createHref: B,
        push: P,
        replace: D,
        go: k,
        back: /* @__PURE__ */ r(function() {
          k(-1);
        }, "back"),
        forward: /* @__PURE__ */ r(function() {
          k(1);
        }, "forward"),
        listen: /* @__PURE__ */ r(function(M) {
          return G.push(M);
        }, "listen"),
        block: /* @__PURE__ */ r(function(M) {
          var Q = N.push(M);
          return N.length === 1 && f.addEventListener(o, m), function() {
            Q(), N.length || f.removeEventListener(o, m);
          };
        }, "block")
      };
      return T;
    }
    r(v, "createHashHistory");
    function I(h) {
      h === void 0 && (h = {});
      var s = h, b = s.initialEntries, f = b === void 0 ? ["/"] : b, L = s.initialIndex, _ = f.map(function(g) {
        var P = a(e({
          pathname: "/",
          search: "",
          hash: "",
          state: null,
          key: S()
        }, typeof g == "string" ? $(g) : g));
        return n(P.pathname.charAt(0) === "/", "Relative pathnames are not supported in createMemoryHistory({ initialEntries }) (invalid ent\
ry: " + JSON.stringify(g) + ")"), P;
      }), j = R(L ?? _.length - 1, 0, _.length - 1), W = t.Action.Pop, U = _[j], J = u(), F = u();
      function K(g) {
        return typeof g == "string" ? g : O(g);
      }
      r(K, "createHref");
      function G(g, P) {
        return P === void 0 && (P = null), a(e({
          pathname: U.pathname,
          search: "",
          hash: ""
        }, typeof g == "string" ? $(g) : g, {
          state: P,
          key: S()
        }));
      }
      r(G, "getNextLocation");
      function N(g, P, D) {
        return !F.length || (F.call({
          action: g,
          location: P,
          retry: D
        }), !1);
      }
      r(N, "allowTx");
      function E(g, P) {
        W = g, U = P, J.call({
          action: W,
          location: U
        });
      }
      r(E, "applyTx");
      function B(g, P) {
        var D = t.Action.Push, k = G(g, P);
        function T() {
          B(g, P);
        }
        r(T, "retry"), n(U.pathname.charAt(0) === "/", "Relative pathnames are not supported in memory history.push(" + JSON.stringify(g) + "\
)"), N(D, k, T) && (j += 1, _.splice(j, _.length, k), E(D, k));
      }
      r(B, "push");
      function c(g, P) {
        var D = t.Action.Replace, k = G(g, P);
        function T() {
          c(g, P);
        }
        r(T, "retry"), n(U.pathname.charAt(0) === "/", "Relative pathnames are not supported in memory history.replace(" + JSON.stringify(g) +
        ")"), N(D, k, T) && (_[j] = k, E(D, k));
      }
      r(c, "replace");
      function p(g) {
        var P = R(j + g, 0, _.length - 1), D = t.Action.Pop, k = _[P];
        function T() {
          p(g);
        }
        r(T, "retry"), N(D, k, T) && (j = P, E(D, k));
      }
      r(p, "go");
      var A = {
        get index() {
          return j;
        },
        get action() {
          return W;
        },
        get location() {
          return U;
        },
        createHref: K,
        push: B,
        replace: c,
        go: p,
        back: /* @__PURE__ */ r(function() {
          p(-1);
        }, "back"),
        forward: /* @__PURE__ */ r(function() {
          p(1);
        }, "forward"),
        listen: /* @__PURE__ */ r(function(P) {
          return J.push(P);
        }, "listen"),
        block: /* @__PURE__ */ r(function(P) {
          return F.push(P);
        }, "block")
      };
      return A;
    }
    r(I, "createMemoryHistory");
    function R(h, s, b) {
      return Math.min(Math.max(h, s), b);
    }
    r(R, "clamp");
    function m(h) {
      h.preventDefault(), h.returnValue = "";
    }
    r(m, "promptBeforeUnload");
    function u() {
      var h = [];
      return {
        get length() {
          return h.length;
        },
        push: /* @__PURE__ */ r(function(b) {
          return h.push(b), function() {
            h = h.filter(function(f) {
              return f !== b;
            });
          };
        }, "push"),
        call: /* @__PURE__ */ r(function(b) {
          h.forEach(function(f) {
            return f && f(b);
          });
        }, "call")
      };
    }
    r(u, "createEvents");
    function S() {
      return Math.random().toString(36).substr(2, 8);
    }
    r(S, "createKey");
    function O(h) {
      var s = h.pathname, b = s === void 0 ? "/" : s, f = h.search, L = f === void 0 ? "" : f, _ = h.hash, j = _ === void 0 ? "" : _;
      return L && L !== "?" && (b += L.charAt(0) === "?" ? L : "?" + L), j && j !== "#" && (b += j.charAt(0) === "#" ? j : "#" + j), b;
    }
    r(O, "createPath");
    function $(h) {
      var s = {};
      if (h) {
        var b = h.indexOf("#");
        b >= 0 && (s.hash = h.substr(b), h = h.substr(0, b));
        var f = h.indexOf("?");
        f >= 0 && (s.search = h.substr(f), h = h.substr(0, f)), h && (s.pathname = h);
      }
      return s;
    }
    r($, "parsePath"), t.createBrowserHistory = w, t.createHashHistory = v, t.createMemoryHistory = I, t.createPath = O, t.parsePath = $, Object.
    defineProperty(t, "__esModule", { value: !0 });
  });
});

// ../node_modules/history/main.js
var Ye = V((Tc, rr) => {
  "use strict";
  process.env.NODE_ENV === "production" ? rr.exports = Da() : rr.exports = Ba();
});

// ../node_modules/react-router/umd/react-router.production.min.js
var Ua = V((bt, Wa) => {
  (function(t, e) {
    typeof bt == "object" && typeof Wa < "u" ? e(bt, require("react"), Ye()) : typeof define == "function" && define.amd ? define(["exports",
    "react", "history"], e) : e((t = t || self).ReactRouter = {}, t.React, t.HistoryLibrary);
  })(bt, function(t, e, a) {
    "use strict";
    function n(c, p) {
      if (!c) throw new Error(p);
    }
    r(n, "a");
    let o = e.createContext(null), i = e.createContext(null), l = e.createContext({ outlet: null, matches: [] });
    function w(c) {
      return S();
    }
    r(w, "s");
    function v(c) {
      n(!1);
    }
    r(v, "o");
    function I(c) {
      let { basename: p = "/", children: A = null, location: g, navigationType: P = a.Action.Pop, navigator: D, static: k = !1 } = c;
      R() && n(!1);
      let T = N(p), d = e.useMemo(() => ({ basename: T, navigator: D, static: k }), [T, D, k]);
      typeof g == "string" && (g = a.parsePath(g));
      let { pathname: M = "/", search: Q = "", hash: z = "", state: te = null, key: ae = "default" } = g, ue = e.useMemo(() => {
        let fe = K(M, T);
        return fe == null ? null : { pathname: fe, search: Q, hash: z, state: te, key: ae };
      }, [T, M, Q, z, te, ae]);
      return ue == null ? null : e.createElement(o.Provider, { value: d }, e.createElement(i.Provider, { children: A, value: { location: ue,
      navigationType: P } }));
    }
    r(I, "u");
    function R() {
      return e.useContext(i) != null;
    }
    r(R, "c");
    function m() {
      return R() || n(!1), e.useContext(i).location;
    }
    r(m, "h");
    function u() {
      R() || n(!1);
      let { basename: c, navigator: p } = e.useContext(o), { matches: A } = e.useContext(l), { pathname: g } = m(), P = JSON.stringify(A.map(
      (k) => k.pathnameBase)), D = e.useRef(!1);
      return e.useEffect(() => {
        D.current = !0;
      }), e.useCallback(function(k, T) {
        if (T === void 0 && (T = {}), !D.current) return;
        if (typeof k == "number") return void p.go(k);
        let d = F(k, JSON.parse(P), g);
        c !== "/" && (d.pathname = G([c, d.pathname])), (T.replace ? p.replace : p.push)(d, T.state);
      }, [c, p, P, g]);
    }
    r(u, "p");
    function S() {
      return e.useContext(l).outlet;
    }
    r(S, "f");
    function O(c) {
      let { matches: p } = e.useContext(l), { pathname: A } = m(), g = JSON.stringify(p.map((P) => P.pathnameBase));
      return e.useMemo(() => F(c, JSON.parse(g), A), [c, g, A]);
    }
    r(O, "m");
    function $(c, p) {
      R() || n(!1);
      let A, { matches: g } = e.useContext(l), P = g[g.length - 1], D = P ? P.params : {}, k = (P && P.pathname, P ? P.pathnameBase : "/"), T = (P &&
      P.route, m());
      if (p) {
        var d;
        let z = typeof p == "string" ? a.parsePath(p) : p;
        k === "/" || (d = z.pathname) != null && d.startsWith(k) || n(!1), A = z;
      } else A = T;
      let M = A.pathname || "/", Q = s(c, { pathname: k === "/" ? M : M.slice(k.length) || "/" });
      return W(Q && Q.map((z) => Object.assign({}, z, { params: Object.assign({}, D, z.params), pathname: G([k, z.pathname]), pathnameBase: z.
      pathnameBase === "/" ? k : G([k, z.pathnameBase]) })), g);
    }
    r($, "d");
    function h(c) {
      let p = [];
      return e.Children.forEach(c, (A) => {
        if (!e.isValidElement(A)) return;
        if (A.type === e.Fragment) return void p.push.apply(p, h(A.props.children));
        A.type !== v && n(!1);
        let g = { caseSensitive: A.props.caseSensitive, element: A.props.element, index: A.props.index, path: A.props.path };
        A.props.children && (g.children = h(A.props.children)), p.push(g);
      }), p;
    }
    r(h, "g");
    function s(c, p, A) {
      A === void 0 && (A = "/");
      let g = K((typeof p == "string" ? a.parsePath(p) : p).pathname || "/", A);
      if (g == null) return null;
      let P = b(c);
      (function(k) {
        k.sort((T, d) => T.score !== d.score ? d.score - T.score : function(M, Q) {
          return M.length === Q.length && M.slice(0, -1).every((z, te) => z === Q[te]) ? M[M.length - 1] - Q[Q.length - 1] : 0;
        }(T.routesMeta.map((M) => M.childrenIndex), d.routesMeta.map((M) => M.childrenIndex)));
      })(P);
      let D = null;
      for (let k = 0; D == null && k < P.length; ++k) D = j(P[k], c, g);
      return D;
    }
    r(s, "v");
    function b(c, p, A, g) {
      return p === void 0 && (p = []), A === void 0 && (A = []), g === void 0 && (g = ""), c.forEach((P, D) => {
        let k = { relativePath: P.path || "", caseSensitive: P.caseSensitive === !0, childrenIndex: D };
        k.relativePath.startsWith("/") && (k.relativePath.startsWith(g) || n(!1), k.relativePath = k.relativePath.slice(g.length));
        let T = G([g, k.relativePath]), d = A.concat(k);
        P.children && P.children.length > 0 && (P.index === !0 && n(!1), b(P.children, p, d, T)), (P.path != null || P.index) && p.push({ path: T,
        score: _(T, P.index), routesMeta: d });
      }), p;
    }
    r(b, "y");
    let f = /^:\w+$/, L = /* @__PURE__ */ r((c) => c === "*", "C");
    function _(c, p) {
      let A = c.split("/"), g = A.length;
      return A.some(L) && (g += -2), p && (g += 2), A.filter((P) => !L(P)).reduce((P, D) => P + (f.test(D) ? 3 : D === "" ? 1 : 10), g);
    }
    r(_, "P");
    function j(c, p, A) {
      let g = p, { routesMeta: P } = c, D = {}, k = "/", T = [];
      for (let d = 0; d < P.length; ++d) {
        let M = P[d], Q = d === P.length - 1, z = k === "/" ? A : A.slice(k.length) || "/", te = U({ path: M.relativePath, caseSensitive: M.
        caseSensitive, end: Q }, z);
        if (!te) return null;
        Object.assign(D, te.params);
        let ae = g[M.childrenIndex];
        T.push({ params: D, pathname: G([k, te.pathname]), pathnameBase: G([k, te.pathnameBase]), route: ae }), te.pathnameBase !== "/" && (k =
        G([k, te.pathnameBase])), g = ae.children;
      }
      return T;
    }
    r(j, "E");
    function W(c, p) {
      return p === void 0 && (p = []), c == null ? null : c.reduceRight((A, g, P) => e.createElement(l.Provider, { children: g.route.element !==
      void 0 ? g.route.element : e.createElement(w, null), value: { outlet: A, matches: p.concat(c.slice(0, P + 1)) } }), null);
    }
    r(W, "R");
    function U(c, p) {
      typeof c == "string" && (c = { path: c, caseSensitive: !1, end: !0 });
      let [A, g] = function(d, M, Q) {
        M === void 0 && (M = !1), Q === void 0 && (Q = !0);
        let z = [], te = "^" + d.replace(/\/*\*?$/, "").replace(/^\/*/, "/").replace(/[\\.*+^$?{}|()[\]]/g, "\\$&").replace(/:(\w+)/g, (ae, ue) => (z.
        push(ue), "([^\\/]+)"));
        return d.endsWith("*") ? (z.push("*"), te += d === "*" || d === "/*" ? "(.*)$" : "(?:\\/(.+)|\\/*)$") : te += Q ? "\\/*$" : "(?:\\b|$\
)", [new RegExp(te, M ? void 0 : "i"), z];
      }(c.path, c.caseSensitive, c.end), P = p.match(A);
      if (!P) return null;
      let D = P[0], k = D.replace(/(.)\/+$/, "$1"), T = P.slice(1);
      return { params: g.reduce((d, M, Q) => {
        if (M === "*") {
          let z = T[Q] || "";
          k = D.slice(0, D.length - z.length).replace(/(.)\/+$/, "$1");
        }
        return d[M] = function(z, te) {
          try {
            return decodeURIComponent(z);
          } catch {
            return z;
          }
        }(T[Q] || ""), d;
      }, {}), pathname: D, pathnameBase: k, pattern: c };
    }
    r(U, "S");
    function J(c, p) {
      p === void 0 && (p = "/");
      let { pathname: A, search: g = "", hash: P = "" } = typeof c == "string" ? a.parsePath(c) : c;
      return { pathname: A ? A.startsWith("/") ? A : function(k, T) {
        let d = T.replace(/\/+$/, "").split("/");
        return k.split("/").forEach((M) => {
          M === ".." ? d.length > 1 && d.pop() : M !== "." && d.push(M);
        }), d.length > 1 ? d.join("/") : "/";
      }(A, p) : p, search: E(g), hash: B(P) };
    }
    r(J, "b");
    function F(c, p, A) {
      let g, P = typeof c == "string" ? a.parsePath(c) : c, D = c === "" || P.pathname === "" ? "/" : P.pathname;
      if (D == null) g = A;
      else {
        let T = p.length - 1;
        if (D.startsWith("..")) {
          let d = D.split("/");
          for (; d[0] === ".."; ) d.shift(), T -= 1;
          P.pathname = d.join("/");
        }
        g = T >= 0 ? p[T] : "/";
      }
      let k = J(P, g);
      return D && D !== "/" && D.endsWith("/") && !k.pathname.endsWith("/") && (k.pathname += "/"), k;
    }
    r(F, "$");
    function K(c, p) {
      if (p === "/") return c;
      if (!c.toLowerCase().startsWith(p.toLowerCase())) return null;
      let A = c.charAt(p.length);
      return A && A !== "/" ? null : c.slice(p.length) || "/";
    }
    r(K, "M");
    let G = /* @__PURE__ */ r((c) => c.join("/").replace(/\/\/+/g, "/"), "W"), N = /* @__PURE__ */ r((c) => c.replace(/\/+$/, "").replace(/^\/*/,
    "/"), "B"), E = /* @__PURE__ */ r((c) => c && c !== "?" ? c.startsWith("?") ? c : "?" + c : "", "N"), B = /* @__PURE__ */ r((c) => c && c !==
    "#" ? c.startsWith("#") ? c : "#" + c : "", "O");
    t.MemoryRouter = function(c) {
      let { basename: p, children: A, initialEntries: g, initialIndex: P } = c, D = e.useRef();
      D.current == null && (D.current = a.createMemoryHistory({ initialEntries: g, initialIndex: P }));
      let k = D.current, [T, d] = e.useState({ action: k.action, location: k.location });
      return e.useLayoutEffect(() => k.listen(d), [k]), e.createElement(I, { basename: p, children: A, location: T.location, navigationType: T.
      action, navigator: k });
    }, t.Navigate = function(c) {
      let { to: p, replace: A, state: g } = c;
      R() || n(!1);
      let P = u();
      return e.useEffect(() => {
        P(p, { replace: A, state: g });
      }), null;
    }, t.Outlet = w, t.Route = v, t.Router = I, t.Routes = function(c) {
      let { children: p, location: A } = c;
      return $(h(p), A);
    }, t.UNSAFE_LocationContext = i, t.UNSAFE_NavigationContext = o, t.UNSAFE_RouteContext = l, t.createRoutesFromChildren = h, t.generatePath =
    function(c, p) {
      return p === void 0 && (p = {}), c.replace(/:(\w+)/g, (A, g) => (p[g] == null && n(!1), p[g])).replace(/\/*\*$/, (A) => p["*"] == null ?
      "" : p["*"].replace(/^\/*/, "/"));
    }, t.matchPath = U, t.matchRoutes = s, t.renderMatches = function(c) {
      return W(c);
    }, t.resolvePath = J, t.useHref = function(c) {
      R() || n(!1);
      let { basename: p, navigator: A } = e.useContext(o), { hash: g, pathname: P, search: D } = O(c), k = P;
      if (p !== "/") {
        let T = function(M) {
          return M === "" || M.pathname === "" ? "/" : typeof M == "string" ? a.parsePath(M).pathname : M.pathname;
        }(c), d = T != null && T.endsWith("/");
        k = P === "/" ? p + (d ? "/" : "") : G([p, P]);
      }
      return A.createHref({ pathname: k, search: D, hash: g });
    }, t.useInRouterContext = R, t.useLocation = m, t.useMatch = function(c) {
      return R() || n(!1), U(c, m().pathname);
    }, t.useNavigate = u, t.useNavigationType = function() {
      return e.useContext(i).navigationType;
    }, t.useOutlet = S, t.useParams = function() {
      let { matches: c } = e.useContext(l), p = c[c.length - 1];
      return p ? p.params : {};
    }, t.useResolvedPath = O, t.useRoutes = $, Object.defineProperty(t, "__esModule", { value: !0 });
  });
});

// ../node_modules/react-router/umd/react-router.development.js
var za = V((Pt, Ha) => {
  (function(t, e) {
    typeof Pt == "object" && typeof Ha < "u" ? e(Pt, require("react"), Ye()) : typeof define == "function" && define.amd ? define(["exports",
    "react", "history"], e) : (t = t || self, e(t.ReactRouter = {}, t.React, t.HistoryLibrary));
  })(Pt, function(t, e, a) {
    "use strict";
    function n(y, x) {
      if (!y) throw new Error(x);
    }
    r(n, "invariant");
    function o(y, x) {
      if (!y) {
        typeof console < "u" && console.warn(x);
        try {
          throw new Error(x);
        } catch {
        }
      }
    }
    r(o, "warning");
    let i = {};
    function l(y, x, C) {
      !x && !i[y] && (i[y] = !0, o(!1, C));
    }
    r(l, "warningOnce");
    let w = /* @__PURE__ */ e.createContext(null);
    w.displayName = "Navigation";
    let v = /* @__PURE__ */ e.createContext(null);
    v.displayName = "Location";
    let I = /* @__PURE__ */ e.createContext({
      outlet: null,
      matches: []
    });
    I.displayName = "Route";
    function R(y) {
      let {
        basename: x,
        children: C,
        initialEntries: q,
        initialIndex: H
      } = y, Z = e.useRef();
      Z.current == null && (Z.current = a.createMemoryHistory({
        initialEntries: q,
        initialIndex: H
      }));
      let Y = Z.current, [X, re] = e.useState({
        action: Y.action,
        location: Y.location
      });
      return e.useLayoutEffect(() => Y.listen(re), [Y]), /* @__PURE__ */ e.createElement(O, {
        basename: x,
        children: C,
        location: X.location,
        navigationType: X.action,
        navigator: Y
      });
    }
    r(R, "MemoryRouter");
    function m(y) {
      let {
        to: x,
        replace: C,
        state: q
      } = y;
      s() || n(
        !1,
        // TODO: This error is probably because they somehow have 2 versions of
        // the router loaded. We can help them understand how to avoid that.
        "<Navigate> may be used only in the context of a <Router> component."
      ), o(!e.useContext(w).static, "<Navigate> must not be used on the initial render in a <StaticRouter>. This is a no-op, but you should \
modify your code so the <Navigate> is only ever rendered in response to some user interaction or state change.");
      let H = _();
      return e.useEffect(() => {
        H(x, {
          replace: C,
          state: q
        });
      }), null;
    }
    r(m, "Navigate");
    function u(y) {
      return j();
    }
    r(u, "Outlet");
    function S(y) {
      n(!1, "A <Route> is only ever to be used as the child of <Routes> element, never rendered directly. Please wrap your <Route> in a <Rou\
tes>.");
    }
    r(S, "Route");
    function O(y) {
      let {
        basename: x = "/",
        children: C = null,
        location: q,
        navigationType: H = a.Action.Pop,
        navigator: Z,
        static: Y = !1
      } = y;
      s() && n(!1, "You cannot render a <Router> inside another <Router>. You should never have more than one in your app.");
      let X = ao(x), re = e.useMemo(() => ({
        basename: X,
        navigator: Z,
        static: Y
      }), [X, Z, Y]);
      typeof q == "string" && (q = a.parsePath(q));
      let {
        pathname: ne = "/",
        search: he = "",
        hash: de = "",
        state: pe = null,
        key: ye = "default"
      } = q, le = e.useMemo(() => {
        let ur = ir(ne, X);
        return ur == null ? null : {
          pathname: ur,
          search: he,
          hash: de,
          state: pe,
          key: ye
        };
      }, [X, ne, he, de, pe, ye]);
      return o(le != null, '<Router basename="' + X + '"> is not able to match the URL ' + ('"' + ne + he + de + '" because it does not star\
t with the ') + "basename, so the <Router> won't render anything."), le == null ? null : /* @__PURE__ */ e.createElement(w.Provider, {
        value: re
      }, /* @__PURE__ */ e.createElement(v.Provider, {
        children: C,
        value: {
          location: le,
          navigationType: H
        }
      }));
    }
    r(O, "Router");
    function $(y) {
      let {
        children: x,
        location: C
      } = y;
      return J(F(x), C);
    }
    r($, "Routes");
    function h(y) {
      s() || n(
        !1,
        // TODO: This error is probably because they somehow have 2 versions of the
        // router loaded. We can help them understand how to avoid that.
        "useHref() may be used only in the context of a <Router> component."
      );
      let {
        basename: x,
        navigator: C
      } = e.useContext(w), {
        hash: q,
        pathname: H,
        search: Z
      } = U(y), Y = H;
      if (x !== "/") {
        let X = no(y), re = X != null && X.endsWith("/");
        Y = H === "/" ? x + (re ? "/" : "") : Pe([x, H]);
      }
      return C.createHref({
        pathname: Y,
        search: Z,
        hash: q
      });
    }
    r(h, "useHref");
    function s() {
      return e.useContext(v) != null;
    }
    r(s, "useInRouterContext");
    function b() {
      return s() || n(
        !1,
        // TODO: This error is probably because they somehow have 2 versions of the
        // router loaded. We can help them understand how to avoid that.
        "useLocation() may be used only in the context of a <Router> component."
      ), e.useContext(v).location;
    }
    r(b, "useLocation");
    function f() {
      return e.useContext(v).navigationType;
    }
    r(f, "useNavigationType");
    function L(y) {
      return s() || n(
        !1,
        // TODO: This error is probably because they somehow have 2 versions of the
        // router loaded. We can help them understand how to avoid that.
        "useMatch() may be used only in the context of a <Router> component."
      ), z(y, b().pathname);
    }
    r(L, "useMatch");
    function _() {
      s() || n(
        !1,
        // TODO: This error is probably because they somehow have 2 versions of the
        // router loaded. We can help them understand how to avoid that.
        "useNavigate() may be used only in the context of a <Router> component."
      );
      let {
        basename: y,
        navigator: x
      } = e.useContext(w), {
        matches: C
      } = e.useContext(I), {
        pathname: q
      } = b(), H = JSON.stringify(C.map((X) => X.pathnameBase)), Z = e.useRef(!1);
      return e.useEffect(() => {
        Z.current = !0;
      }), e.useCallback(function(X, re) {
        if (re === void 0 && (re = {}), o(Z.current, "You should call navigate() in a React.useEffect(), not when your component is first re\
ndered."), !Z.current) return;
        if (typeof X == "number") {
          x.go(X);
          return;
        }
        let ne = Ot(X, JSON.parse(H), q);
        y !== "/" && (ne.pathname = Pe([y, ne.pathname])), (re.replace ? x.replace : x.push)(ne, re.state);
      }, [y, x, H, q]);
    }
    r(_, "useNavigate");
    function j() {
      return e.useContext(I).outlet;
    }
    r(j, "useOutlet");
    function W() {
      let {
        matches: y
      } = e.useContext(I), x = y[y.length - 1];
      return x ? x.params : {};
    }
    r(W, "useParams");
    function U(y) {
      let {
        matches: x
      } = e.useContext(I), {
        pathname: C
      } = b(), q = JSON.stringify(x.map((H) => H.pathnameBase));
      return e.useMemo(() => Ot(y, JSON.parse(q), C), [y, q, C]);
    }
    r(U, "useResolvedPath");
    function J(y, x) {
      s() || n(
        !1,
        // TODO: This error is probably because they somehow have 2 versions of the
        // router loaded. We can help them understand how to avoid that.
        "useRoutes() may be used only in the context of a <Router> component."
      );
      let {
        matches: C
      } = e.useContext(I), q = C[C.length - 1], H = q ? q.params : {}, Z = q ? q.pathname : "/", Y = q ? q.pathnameBase : "/", X = q && q.route;
      {
        let le = X && X.path || "";
        l(Z, !X || le.endsWith("*"), "You rendered descendant <Routes> (or called `useRoutes()`) at " + ('"' + Z + '" (under <Route path="' +
        le + '">) but the ') + `parent route path has no trailing "*". This means if you navigate deeper, the parent won't match anymore and\
 therefore the child routes will never render.

` + ('Please change the parent <Route path="' + le + '"> to <Route ') + ('path="' + le + '/*">.'));
      }
      let re = b(), ne;
      if (x) {
        var he;
        let le = typeof x == "string" ? a.parsePath(x) : x;
        Y === "/" || (he = le.pathname) != null && he.startsWith(Y) || n(!1, "When overriding the location using `<Routes location>` or `use\
Routes(routes, location)`, the location pathname must begin with the portion of the URL pathname that was " + ('matched by all parent routes\
. The current pathname base is "' + Y + '" ') + ('but pathname "' + le.pathname + '" was given in the `location` prop.')), ne = le;
      } else
        ne = re;
      let de = ne.pathname || "/", pe = Y === "/" ? de : de.slice(Y.length) || "/", ye = G(y, {
        pathname: pe
      });
      return o(X || ye != null, 'No routes matched location "' + ne.pathname + ne.search + ne.hash + '" '), o(ye == null || ye[ye.length - 1].
      route.element !== void 0, 'Matched leaf route at location "' + ne.pathname + ne.search + ne.hash + '" does not have an element. This m\
eans it will render an <Outlet /> with a null value by default resulting in an "empty" page.'), Q(ye && ye.map((le) => Object.assign({}, le,
      {
        params: Object.assign({}, H, le.params),
        pathname: Pe([Y, le.pathname]),
        pathnameBase: le.pathnameBase === "/" ? Y : Pe([Y, le.pathnameBase])
      })), C);
    }
    r(J, "useRoutes");
    function F(y) {
      let x = [];
      return e.Children.forEach(y, (C) => {
        if (!/* @__PURE__ */ e.isValidElement(C))
          return;
        if (C.type === e.Fragment) {
          x.push.apply(x, F(C.props.children));
          return;
        }
        C.type !== S && n(!1, "[" + (typeof C.type == "string" ? C.type : C.type.name) + "] is not a <Route> component. All component childr\
en of <Routes> must be a <Route> or <React.Fragment>");
        let q = {
          caseSensitive: C.props.caseSensitive,
          element: C.props.element,
          index: C.props.index,
          path: C.props.path
        };
        C.props.children && (q.children = F(C.props.children)), x.push(q);
      }), x;
    }
    r(F, "createRoutesFromChildren");
    function K(y, x) {
      return x === void 0 && (x = {}), y.replace(/:(\w+)/g, (C, q) => (x[q] == null && n(!1, 'Missing ":' + q + '" param'), x[q])).replace(/\/*\*$/,
      (C) => x["*"] == null ? "" : x["*"].replace(/^\/*/, "/"));
    }
    r(K, "generatePath");
    function G(y, x, C) {
      C === void 0 && (C = "/");
      let q = typeof x == "string" ? a.parsePath(x) : x, H = ir(q.pathname || "/", C);
      if (H == null)
        return null;
      let Z = N(y);
      E(Z);
      let Y = null;
      for (let X = 0; Y == null && X < Z.length; ++X)
        Y = d(Z[X], y, H);
      return Y;
    }
    r(G, "matchRoutes");
    function N(y, x, C, q) {
      return x === void 0 && (x = []), C === void 0 && (C = []), q === void 0 && (q = ""), y.forEach((H, Z) => {
        let Y = {
          relativePath: H.path || "",
          caseSensitive: H.caseSensitive === !0,
          childrenIndex: Z
        };
        Y.relativePath.startsWith("/") && (Y.relativePath.startsWith(q) || n(!1, 'Absolute route path "' + Y.relativePath + '" nested under \
path ' + ('"' + q + '" is not valid. An absolute child route path ') + "must start with the combined path of all its parent routes."), Y.relativePath =
        Y.relativePath.slice(q.length));
        let X = Pe([q, Y.relativePath]), re = C.concat(Y);
        H.children && H.children.length > 0 && (H.index === !0 && n(!1, "Index routes must not have child routes. Please remove " + ('all ch\
ild routes from route path "' + X + '".')), N(H.children, x, re, X)), !(H.path == null && !H.index) && x.push({
          path: X,
          score: k(X, H.index),
          routesMeta: re
        });
      }), x;
    }
    r(N, "flattenRoutes");
    function E(y) {
      y.sort((x, C) => x.score !== C.score ? C.score - x.score : T(x.routesMeta.map((q) => q.childrenIndex), C.routesMeta.map((q) => q.childrenIndex)));
    }
    r(E, "rankRouteBranches");
    let B = /^:\w+$/, c = 3, p = 2, A = 1, g = 10, P = -2, D = /* @__PURE__ */ r((y) => y === "*", "isSplat");
    function k(y, x) {
      let C = y.split("/"), q = C.length;
      return C.some(D) && (q += P), x && (q += p), C.filter((H) => !D(H)).reduce((H, Z) => H + (B.test(Z) ? c : Z === "" ? A : g), q);
    }
    r(k, "computeScore");
    function T(y, x) {
      return y.length === x.length && y.slice(0, -1).every((q, H) => q === x[H]) ? (
        // If two routes are siblings, we should try to match the earlier sibling
        // first. This allows people to have fine-grained control over the matching
        // behavior by simply putting routes with identical paths in the order they
        // want them tried.
        y[y.length - 1] - x[x.length - 1]
      ) : (
        // Otherwise, it doesn't really make sense to rank non-siblings by index,
        // so they sort equally.
        0
      );
    }
    r(T, "compareIndexes");
    function d(y, x, C) {
      let q = x, {
        routesMeta: H
      } = y, Z = {}, Y = "/", X = [];
      for (let re = 0; re < H.length; ++re) {
        let ne = H[re], he = re === H.length - 1, de = Y === "/" ? C : C.slice(Y.length) || "/", pe = z({
          path: ne.relativePath,
          caseSensitive: ne.caseSensitive,
          end: he
        }, de);
        if (!pe) return null;
        Object.assign(Z, pe.params);
        let ye = q[ne.childrenIndex];
        X.push({
          params: Z,
          pathname: Pe([Y, pe.pathname]),
          pathnameBase: Pe([Y, pe.pathnameBase]),
          route: ye
        }), pe.pathnameBase !== "/" && (Y = Pe([Y, pe.pathnameBase])), q = ye.children;
      }
      return X;
    }
    r(d, "matchRouteBranch");
    function M(y) {
      return Q(y);
    }
    r(M, "renderMatches");
    function Q(y, x) {
      return x === void 0 && (x = []), y == null ? null : y.reduceRight((C, q, H) => /* @__PURE__ */ e.createElement(I.Provider, {
        children: q.route.element !== void 0 ? q.route.element : /* @__PURE__ */ e.createElement(u, null),
        value: {
          outlet: C,
          matches: x.concat(y.slice(0, H + 1))
        }
      }), null);
    }
    r(Q, "_renderMatches");
    function z(y, x) {
      typeof y == "string" && (y = {
        path: y,
        caseSensitive: !1,
        end: !0
      });
      let [C, q] = te(y.path, y.caseSensitive, y.end), H = x.match(C);
      if (!H) return null;
      let Z = H[0], Y = Z.replace(/(.)\/+$/, "$1"), X = H.slice(1);
      return {
        params: q.reduce((ne, he, de) => {
          if (he === "*") {
            let pe = X[de] || "";
            Y = Z.slice(0, Z.length - pe.length).replace(/(.)\/+$/, "$1");
          }
          return ne[he] = ae(X[de] || "", he), ne;
        }, {}),
        pathname: Z,
        pathnameBase: Y,
        pattern: y
      };
    }
    r(z, "matchPath");
    function te(y, x, C) {
      x === void 0 && (x = !1), C === void 0 && (C = !0), o(y === "*" || !y.endsWith("*") || y.endsWith("/*"), 'Route path "' + y + '" will \
be treated as if it were ' + ('"' + y.replace(/\*$/, "/*") + '" because the `*` character must ') + "always follow a `/` in the pattern. To \
get rid of this warning, " + ('please change the route path to "' + y.replace(/\*$/, "/*") + '".'));
      let q = [], H = "^" + y.replace(/\/*\*?$/, "").replace(/^\/*/, "/").replace(/[\\.*+^$?{}|()[\]]/g, "\\$&").replace(/:(\w+)/g, (Y, X) => (q.
      push(X), "([^\\/]+)"));
      return y.endsWith("*") ? (q.push("*"), H += y === "*" || y === "/*" ? "(.*)$" : "(?:\\/(.+)|\\/*)$") : H += C ? "\\/*$" : (
        // Otherwise, at least match a word boundary. This restricts parent
        // routes to matching only their own words and nothing more, e.g. parent
        // route "/home" should not match "/home2".
        "(?:\\b|$)"
      ), [new RegExp(H, x ? void 0 : "i"), q];
    }
    r(te, "compilePath");
    function ae(y, x) {
      try {
        return decodeURIComponent(y);
      } catch (C) {
        return o(!1, 'The value for the URL param "' + x + '" will not be decoded because' + (' the string "' + y + '" is a malformed URL se\
gment. This is probably') + (" due to a bad percent encoding (" + C + ").")), y;
      }
    }
    r(ae, "safelyDecodeURIComponent");
    function ue(y, x) {
      x === void 0 && (x = "/");
      let {
        pathname: C,
        search: q = "",
        hash: H = ""
      } = typeof y == "string" ? a.parsePath(y) : y;
      return {
        pathname: C ? C.startsWith("/") ? C : fe(C, x) : x,
        search: oo(q),
        hash: io(H)
      };
    }
    r(ue, "resolvePath");
    function fe(y, x) {
      let C = x.replace(/\/+$/, "").split("/");
      return y.split("/").forEach((H) => {
        H === ".." ? C.length > 1 && C.pop() : H !== "." && C.push(H);
      }), C.length > 1 ? C.join("/") : "/";
    }
    r(fe, "resolvePathname");
    function Ot(y, x, C) {
      let q = typeof y == "string" ? a.parsePath(y) : y, H = y === "" || q.pathname === "" ? "/" : q.pathname, Z;
      if (H == null)
        Z = C;
      else {
        let X = x.length - 1;
        if (H.startsWith("..")) {
          let re = H.split("/");
          for (; re[0] === ".."; )
            re.shift(), X -= 1;
          q.pathname = re.join("/");
        }
        Z = X >= 0 ? x[X] : "/";
      }
      let Y = ue(q, Z);
      return H && H !== "/" && H.endsWith("/") && !Y.pathname.endsWith("/") && (Y.pathname += "/"), Y;
    }
    r(Ot, "resolveTo");
    function no(y) {
      return y === "" || y.pathname === "" ? "/" : typeof y == "string" ? a.parsePath(y).pathname : y.pathname;
    }
    r(no, "getToPathname");
    function ir(y, x) {
      if (x === "/") return y;
      if (!y.toLowerCase().startsWith(x.toLowerCase()))
        return null;
      let C = y.charAt(x.length);
      return C && C !== "/" ? null : y.slice(x.length) || "/";
    }
    r(ir, "stripBasename");
    let Pe = /* @__PURE__ */ r((y) => y.join("/").replace(/\/\/+/g, "/"), "joinPaths"), ao = /* @__PURE__ */ r((y) => y.replace(/\/+$/, "").
    replace(/^\/*/, "/"), "normalizePathname"), oo = /* @__PURE__ */ r((y) => !y || y === "?" ? "" : y.startsWith("?") ? y : "?" + y, "norma\
lizeSearch"), io = /* @__PURE__ */ r((y) => !y || y === "#" ? "" : y.startsWith("#") ? y : "#" + y, "normalizeHash");
    t.MemoryRouter = R, t.Navigate = m, t.Outlet = u, t.Route = S, t.Router = O, t.Routes = $, t.UNSAFE_LocationContext = v, t.UNSAFE_NavigationContext =
    w, t.UNSAFE_RouteContext = I, t.createRoutesFromChildren = F, t.generatePath = K, t.matchPath = z, t.matchRoutes = G, t.renderMatches = M,
    t.resolvePath = ue, t.useHref = h, t.useInRouterContext = s, t.useLocation = b, t.useMatch = L, t.useNavigate = _, t.useNavigationType =
    f, t.useOutlet = j, t.useParams = W, t.useResolvedPath = U, t.useRoutes = J, Object.defineProperty(t, "__esModule", { value: !0 });
  });
});

// ../node_modules/react-router/main.js
var nr = V((kc, Ga) => {
  "use strict";
  Ga.exports = process.env.NODE_ENV === "production" ? Ua() : za();
});

// ../node_modules/react-router-dom/umd/react-router-dom.production.min.js
var Ka = V((wt, Va) => {
  (function(t, e) {
    typeof wt == "object" && typeof Va < "u" ? e(wt, require("react"), Ye(), nr()) : typeof define == "function" && define.amd ? define(["ex\
ports", "react", "history", "react-router"], e) : e((t = t || self).ReactRouterDOM = {}, t.React, t.HistoryLibrary, t.ReactRouter);
  })(wt, function(t, e, a, n) {
    "use strict";
    function o() {
      return o = Object.assign || function(u) {
        for (var S = 1; S < arguments.length; S++) {
          var O = arguments[S];
          for (var $ in O) Object.prototype.hasOwnProperty.call(O, $) && (u[$] = O[$]);
        }
        return u;
      }, o.apply(this, arguments);
    }
    r(o, "a");
    function i(u, S) {
      if (u == null) return {};
      var O, $, h = {}, s = Object.keys(u);
      for ($ = 0; $ < s.length; $++) O = s[$], S.indexOf(O) >= 0 || (h[O] = u[O]);
      return h;
    }
    r(i, "o");
    let l = ["onClick", "reloadDocument", "replace", "state", "target", "to"], w = ["aria-current", "caseSensitive", "className", "end", "st\
yle", "to"], v = e.forwardRef(function(u, S) {
      let { onClick: O, reloadDocument: $, replace: h = !1, state: s, target: b, to: f } = u, L = i(u, l), _ = n.useHref(f), j = R(f, { replace: h,
      state: s, target: b });
      return e.createElement("a", o({}, L, { href: _, onClick: /* @__PURE__ */ r(function(W) {
        O && O(W), W.defaultPrevented || $ || j(W);
      }, "onClick"), ref: S, target: b }));
    }), I = e.forwardRef(function(u, S) {
      let { "aria-current": O = "page", caseSensitive: $ = !1, className: h = "", end: s = !1, style: b, to: f } = u, L = i(u, w), _ = n.useLocation(),
      j = n.useResolvedPath(f), W = _.pathname, U = j.pathname;
      $ || (W = W.toLowerCase(), U = U.toLowerCase());
      let J, F = W === U || !s && W.startsWith(U) && W.charAt(U.length) === "/", K = F ? O : void 0;
      J = typeof h == "function" ? h({ isActive: F }) : [h, F ? "active" : null].filter(Boolean).join(" ");
      let G = typeof b == "function" ? b({ isActive: F }) : b;
      return e.createElement(v, o({}, L, { "aria-current": K, className: J, ref: S, style: G, to: f }));
    });
    function R(u, S) {
      let { target: O, replace: $, state: h } = S === void 0 ? {} : S, s = n.useNavigate(), b = n.useLocation(), f = n.useResolvedPath(u);
      return e.useCallback((L) => {
        if (!(L.button !== 0 || O && O !== "_self" || function(_) {
          return !!(_.metaKey || _.altKey || _.ctrlKey || _.shiftKey);
        }(L))) {
          L.preventDefault();
          let _ = !!$ || a.createPath(b) === a.createPath(f);
          s(u, { replace: _, state: h });
        }
      }, [b, s, f, $, h, O, u]);
    }
    r(R, "l");
    function m(u) {
      return u === void 0 && (u = ""), new URLSearchParams(typeof u == "string" || Array.isArray(u) || u instanceof URLSearchParams ? u : Object.
      keys(u).reduce((S, O) => {
        let $ = u[O];
        return S.concat(Array.isArray($) ? $.map((h) => [O, h]) : [[O, $]]);
      }, []));
    }
    r(m, "f"), Object.defineProperty(t, "MemoryRouter", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.MemoryRouter;
    }, "get") }), Object.defineProperty(t, "Navigate", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.Navigate;
    }, "get") }), Object.defineProperty(t, "Outlet", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.Outlet;
    }, "get") }), Object.defineProperty(t, "Route", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.Route;
    }, "get") }), Object.defineProperty(t, "Router", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.Router;
    }, "get") }), Object.defineProperty(t, "Routes", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.Routes;
    }, "get") }), Object.defineProperty(t, "UNSAFE_LocationContext", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.UNSAFE_LocationContext;
    }, "get") }), Object.defineProperty(t, "UNSAFE_NavigationContext", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.UNSAFE_NavigationContext;
    }, "get") }), Object.defineProperty(t, "UNSAFE_RouteContext", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.UNSAFE_RouteContext;
    }, "get") }), Object.defineProperty(t, "createRoutesFromChildren", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.createRoutesFromChildren;
    }, "get") }), Object.defineProperty(t, "generatePath", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.generatePath;
    }, "get") }), Object.defineProperty(t, "matchPath", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.matchPath;
    }, "get") }), Object.defineProperty(t, "matchRoutes", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.matchRoutes;
    }, "get") }), Object.defineProperty(t, "renderMatches", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.renderMatches;
    }, "get") }), Object.defineProperty(t, "resolvePath", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.resolvePath;
    }, "get") }), Object.defineProperty(t, "useHref", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.useHref;
    }, "get") }), Object.defineProperty(t, "useInRouterContext", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.useInRouterContext;
    }, "get") }), Object.defineProperty(t, "useLocation", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.useLocation;
    }, "get") }), Object.defineProperty(t, "useMatch", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.useMatch;
    }, "get") }), Object.defineProperty(t, "useNavigate", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.useNavigate;
    }, "get") }), Object.defineProperty(t, "useNavigationType", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.useNavigationType;
    }, "get") }), Object.defineProperty(t, "useOutlet", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.useOutlet;
    }, "get") }), Object.defineProperty(t, "useParams", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.useParams;
    }, "get") }), Object.defineProperty(t, "useResolvedPath", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.useResolvedPath;
    }, "get") }), Object.defineProperty(t, "useRoutes", { enumerable: !0, get: /* @__PURE__ */ r(function() {
      return n.useRoutes;
    }, "get") }), t.BrowserRouter = function(u) {
      let { basename: S, children: O, window: $ } = u, h = e.useRef();
      h.current == null && (h.current = a.createBrowserHistory({ window: $ }));
      let s = h.current, [b, f] = e.useState({ action: s.action, location: s.location });
      return e.useLayoutEffect(() => s.listen(f), [s]), e.createElement(n.Router, { basename: S, children: O, location: b.location, navigationType: b.
      action, navigator: s });
    }, t.HashRouter = function(u) {
      let { basename: S, children: O, window: $ } = u, h = e.useRef();
      h.current == null && (h.current = a.createHashHistory({ window: $ }));
      let s = h.current, [b, f] = e.useState({ action: s.action, location: s.location });
      return e.useLayoutEffect(() => s.listen(f), [s]), e.createElement(n.Router, { basename: S, children: O, location: b.location, navigationType: b.
      action, navigator: s });
    }, t.Link = v, t.NavLink = I, t.createSearchParams = m, t.useLinkClickHandler = R, t.useSearchParams = function(u) {
      let S = e.useRef(m(u)), O = n.useLocation(), $ = e.useMemo(() => {
        let s = m(O.search);
        for (let b of S.current.keys()) s.has(b) || S.current.getAll(b).forEach((f) => {
          s.append(b, f);
        });
        return s;
      }, [O.search]), h = n.useNavigate();
      return [$, e.useCallback((s, b) => {
        h("?" + m(s), b);
      }, [h])];
    }, Object.defineProperty(t, "__esModule", { value: !0 });
  });
});

// ../node_modules/react-router-dom/umd/react-router-dom.development.js
var Ja = V((St, Qa) => {
  (function(t, e) {
    typeof St == "object" && typeof Qa < "u" ? e(St, require("react"), Ye(), nr()) : typeof define == "function" && define.amd ? define(["ex\
ports", "react", "history", "react-router"], e) : (t = t || self, e(t.ReactRouterDOM = {}, t.React, t.HistoryLibrary, t.ReactRouter));
  })(St, function(t, e, a, n) {
    "use strict";
    function o() {
      return o = Object.assign || function(s) {
        for (var b = 1; b < arguments.length; b++) {
          var f = arguments[b];
          for (var L in f)
            Object.prototype.hasOwnProperty.call(f, L) && (s[L] = f[L]);
        }
        return s;
      }, o.apply(this, arguments);
    }
    r(o, "_extends");
    function i(s, b) {
      if (s == null) return {};
      var f = {}, L = Object.keys(s), _, j;
      for (j = 0; j < L.length; j++)
        _ = L[j], !(b.indexOf(_) >= 0) && (f[_] = s[_]);
      return f;
    }
    r(i, "_objectWithoutPropertiesLoose");
    let l = ["onClick", "reloadDocument", "replace", "state", "target", "to"], w = ["aria-current", "caseSensitive", "className", "end", "st\
yle", "to"];
    function v(s, b) {
      if (!s) {
        typeof console < "u" && console.warn(b);
        try {
          throw new Error(b);
        } catch {
        }
      }
    }
    r(v, "warning");
    function I(s) {
      let {
        basename: b,
        children: f,
        window: L
      } = s, _ = e.useRef();
      _.current == null && (_.current = a.createBrowserHistory({
        window: L
      }));
      let j = _.current, [W, U] = e.useState({
        action: j.action,
        location: j.location
      });
      return e.useLayoutEffect(() => j.listen(U), [j]), /* @__PURE__ */ e.createElement(n.Router, {
        basename: b,
        children: f,
        location: W.location,
        navigationType: W.action,
        navigator: j
      });
    }
    r(I, "BrowserRouter");
    function R(s) {
      let {
        basename: b,
        children: f,
        window: L
      } = s, _ = e.useRef();
      _.current == null && (_.current = a.createHashHistory({
        window: L
      }));
      let j = _.current, [W, U] = e.useState({
        action: j.action,
        location: j.location
      });
      return e.useLayoutEffect(() => j.listen(U), [j]), /* @__PURE__ */ e.createElement(n.Router, {
        basename: b,
        children: f,
        location: W.location,
        navigationType: W.action,
        navigator: j
      });
    }
    r(R, "HashRouter");
    function m(s) {
      return !!(s.metaKey || s.altKey || s.ctrlKey || s.shiftKey);
    }
    r(m, "isModifiedEvent");
    let u = /* @__PURE__ */ e.forwardRef(/* @__PURE__ */ r(function(b, f) {
      let {
        onClick: L,
        reloadDocument: _,
        replace: j = !1,
        state: W,
        target: U,
        to: J
      } = b, F = i(b, l), K = n.useHref(J), G = O(J, {
        replace: j,
        state: W,
        target: U
      });
      function N(E) {
        L && L(E), !E.defaultPrevented && !_ && G(E);
      }
      return r(N, "handleClick"), // eslint-disable-next-line jsx-a11y/anchor-has-content
      /* @__PURE__ */ e.createElement("a", o({}, F, {
        href: K,
        onClick: N,
        ref: f,
        target: U
      }));
    }, "LinkWithRef"));
    u.displayName = "Link";
    let S = /* @__PURE__ */ e.forwardRef(/* @__PURE__ */ r(function(b, f) {
      let {
        "aria-current": L = "page",
        caseSensitive: _ = !1,
        className: j = "",
        end: W = !1,
        style: U,
        to: J
      } = b, F = i(b, w), K = n.useLocation(), G = n.useResolvedPath(J), N = K.pathname, E = G.pathname;
      _ || (N = N.toLowerCase(), E = E.toLowerCase());
      let B = N === E || !W && N.startsWith(E) && N.charAt(E.length) === "/", c = B ? L : void 0, p;
      typeof j == "function" ? p = j({
        isActive: B
      }) : p = [j, B ? "active" : null].filter(Boolean).join(" ");
      let A = typeof U == "function" ? U({
        isActive: B
      }) : U;
      return /* @__PURE__ */ e.createElement(u, o({}, F, {
        "aria-current": c,
        className: p,
        ref: f,
        style: A,
        to: J
      }));
    }, "NavLinkWithRef"));
    S.displayName = "NavLink";
    function O(s, b) {
      let {
        target: f,
        replace: L,
        state: _
      } = b === void 0 ? {} : b, j = n.useNavigate(), W = n.useLocation(), U = n.useResolvedPath(s);
      return e.useCallback((J) => {
        if (J.button === 0 && // Ignore everything but left clicks
        (!f || f === "_self") && // Let browser handle "target=_blank" etc.
        !m(J)) {
          J.preventDefault();
          let F = !!L || a.createPath(W) === a.createPath(U);
          j(s, {
            replace: F,
            state: _
          });
        }
      }, [W, j, U, L, _, f, s]);
    }
    r(O, "useLinkClickHandler");
    function $(s) {
      v(typeof URLSearchParams < "u", "You cannot use the `useSearchParams` hook in a browser that does not support the URLSearchParams API.\
 If you need to support Internet Explorer 11, we recommend you load a polyfill such as https://github.com/ungap/url-search-params\n\nIf you're\
 unsure how to load polyfills, we recommend you check out https://polyfill.io/v3/ which provides some recommendations about how to load poly\
fills only for users that need them, instead of for every user.");
      let b = e.useRef(h(s)), f = n.useLocation(), L = e.useMemo(() => {
        let W = h(f.search);
        for (let U of b.current.keys())
          W.has(U) || b.current.getAll(U).forEach((J) => {
            W.append(U, J);
          });
        return W;
      }, [f.search]), _ = n.useNavigate(), j = e.useCallback((W, U) => {
        _("?" + h(W), U);
      }, [_]);
      return [L, j];
    }
    r($, "useSearchParams");
    function h(s) {
      return s === void 0 && (s = ""), new URLSearchParams(typeof s == "string" || Array.isArray(s) || s instanceof URLSearchParams ? s : Object.
      keys(s).reduce((b, f) => {
        let L = s[f];
        return b.concat(Array.isArray(L) ? L.map((_) => [f, _]) : [[f, L]]);
      }, []));
    }
    r(h, "createSearchParams"), Object.defineProperty(t, "MemoryRouter", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.MemoryRouter;
      }, "get")
    }), Object.defineProperty(t, "Navigate", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.Navigate;
      }, "get")
    }), Object.defineProperty(t, "Outlet", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.Outlet;
      }, "get")
    }), Object.defineProperty(t, "Route", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.Route;
      }, "get")
    }), Object.defineProperty(t, "Router", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.Router;
      }, "get")
    }), Object.defineProperty(t, "Routes", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.Routes;
      }, "get")
    }), Object.defineProperty(t, "UNSAFE_LocationContext", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.UNSAFE_LocationContext;
      }, "get")
    }), Object.defineProperty(t, "UNSAFE_NavigationContext", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.UNSAFE_NavigationContext;
      }, "get")
    }), Object.defineProperty(t, "UNSAFE_RouteContext", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.UNSAFE_RouteContext;
      }, "get")
    }), Object.defineProperty(t, "createRoutesFromChildren", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.createRoutesFromChildren;
      }, "get")
    }), Object.defineProperty(t, "generatePath", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.generatePath;
      }, "get")
    }), Object.defineProperty(t, "matchPath", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.matchPath;
      }, "get")
    }), Object.defineProperty(t, "matchRoutes", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.matchRoutes;
      }, "get")
    }), Object.defineProperty(t, "renderMatches", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.renderMatches;
      }, "get")
    }), Object.defineProperty(t, "resolvePath", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.resolvePath;
      }, "get")
    }), Object.defineProperty(t, "useHref", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.useHref;
      }, "get")
    }), Object.defineProperty(t, "useInRouterContext", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.useInRouterContext;
      }, "get")
    }), Object.defineProperty(t, "useLocation", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.useLocation;
      }, "get")
    }), Object.defineProperty(t, "useMatch", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.useMatch;
      }, "get")
    }), Object.defineProperty(t, "useNavigate", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.useNavigate;
      }, "get")
    }), Object.defineProperty(t, "useNavigationType", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.useNavigationType;
      }, "get")
    }), Object.defineProperty(t, "useOutlet", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.useOutlet;
      }, "get")
    }), Object.defineProperty(t, "useParams", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.useParams;
      }, "get")
    }), Object.defineProperty(t, "useResolvedPath", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.useResolvedPath;
      }, "get")
    }), Object.defineProperty(t, "useRoutes", {
      enumerable: !0,
      get: /* @__PURE__ */ r(function() {
        return n.useRoutes;
      }, "get")
    }), t.BrowserRouter = I, t.HashRouter = R, t.Link = u, t.NavLink = S, t.createSearchParams = h, t.useLinkClickHandler = O, t.useSearchParams =
    $, Object.defineProperty(t, "__esModule", { value: !0 });
  });
});

// ../node_modules/react-router-dom/main.js
var Xa = V((qc, Ya) => {
  "use strict";
  Ya.exports = process.env.NODE_ENV === "production" ? Ka() : Ja();
});

// src/router/index.ts
var cl = {};
po(cl, {
  BaseLocationProvider: () => ll,
  DEEPLY_EQUAL: () => Qe,
  Link: () => to,
  Location: () => ar,
  LocationProvider: () => ul,
  Match: () => or,
  Route: () => ro,
  buildArgsParam: () => Qu,
  deepDiff: () => ht,
  getMatch: () => er,
  parsePath: () => Zt,
  queryFromLocation: () => Ju,
  queryFromString: () => mt,
  stringifyQuery: () => Yu,
  useNavigate: () => il
});
module.exports = ho(cl);

// src/router/utils.ts
var Ma = require("@storybook/core/client-logger");

// ../node_modules/dequal/dist/index.mjs
var cr = Object.prototype.hasOwnProperty;
function sr(t, e, a) {
  for (a of t.keys())
    if (Me(a, e)) return a;
}
r(sr, "find");
function Me(t, e) {
  var a, n, o;
  if (t === e) return !0;
  if (t && e && (a = t.constructor) === e.constructor) {
    if (a === Date) return t.getTime() === e.getTime();
    if (a === RegExp) return t.toString() === e.toString();
    if (a === Array) {
      if ((n = t.length) === e.length)
        for (; n-- && Me(t[n], e[n]); ) ;
      return n === -1;
    }
    if (a === Set) {
      if (t.size !== e.size)
        return !1;
      for (n of t)
        if (o = n, o && typeof o == "object" && (o = sr(e, o), !o) || !e.has(o)) return !1;
      return !0;
    }
    if (a === Map) {
      if (t.size !== e.size)
        return !1;
      for (n of t)
        if (o = n[0], o && typeof o == "object" && (o = sr(e, o), !o) || !Me(n[1], e.get(o)))
          return !1;
      return !0;
    }
    if (a === ArrayBuffer)
      t = new Uint8Array(t), e = new Uint8Array(e);
    else if (a === DataView) {
      if ((n = t.byteLength) === e.byteLength)
        for (; n-- && t.getInt8(n) === e.getInt8(n); ) ;
      return n === -1;
    }
    if (ArrayBuffer.isView(t)) {
      if ((n = t.byteLength) === e.byteLength)
        for (; n-- && t[n] === e[n]; ) ;
      return n === -1;
    }
    if (!a || typeof t == "object") {
      n = 0;
      for (a in t)
        if (cr.call(t, a) && ++n && !cr.call(e, a) || !(a in e) || !Me(t[a], e[a])) return !1;
      return Object.keys(e).length === n;
    }
  }
  return t !== t && e !== e;
}
r(Me, "dequal");

// src/router/utils.ts
var Je = Re(Tr(), 1), yt = Re(Br(), 1), dt = Re(xa(), 1), Ca = Re(Ia(), 1);
var Gu = /\/([^/]+)\/(?:(.*)_)?([^/]+)?/, Zt = (0, yt.default)(1e3)((t) => {
  let e = {
    viewMode: void 0,
    storyId: void 0,
    refId: void 0
  };
  if (t) {
    let [, a, n, o] = t.toLowerCase().match(Gu) || [];
    a && Object.assign(e, {
      viewMode: a,
      storyId: o,
      refId: n
    });
  }
  return e;
}), Qe = Symbol("Deeply equal"), ht = /* @__PURE__ */ r((t, e) => {
  if (typeof t != typeof e)
    return e;
  if (Me(t, e))
    return Qe;
  if (Array.isArray(t) && Array.isArray(e)) {
    let a = e.reduce((n, o, i) => {
      let l = ht(t[i], o);
      return l !== Qe && (n[i] = l), n;
    }, new Array(e.length));
    return e.length >= t.length ? a : a.concat(new Array(t.length - e.length).fill(void 0));
  }
  return (0, Je.default)(t) && (0, Je.default)(e) ? Object.keys({ ...t, ...e }).reduce((a, n) => {
    let o = ht(t?.[n], e?.[n]);
    return o === Qe ? a : Object.assign(a, { [n]: o });
  }, {}) : e;
}, "deepDiff"), La = /^[a-zA-Z0-9 _-]*$/, Vu = /^-?[0-9]+(\.[0-9]+)?$/, _a = /^#([a-f0-9]{3,4}|[a-f0-9]{6}|[a-f0-9]{8})$/i, Ta = /^(rgba?|hsla?)\(([0-9]{1,3}),\s?([0-9]{1,3})%?,\s?([0-9]{1,3})%?,?\s?([0-9](\.[0-9]{1,2})?)?\)$/i,
Yt = /* @__PURE__ */ r((t = "", e) => t === null || t === "" || !La.test(t) ? !1 : e == null || e instanceof Date || typeof e == "number" ||
typeof e == "boolean" ? !0 : typeof e == "string" ? La.test(e) || Vu.test(e) || _a.test(e) || Ta.test(e) : Array.isArray(e) ? e.every((a) => Yt(
t, a)) : (0, Je.default)(e) ? Object.entries(e).every(([a, n]) => Yt(a, n)) : !1, "validateArgs"), Xt = /* @__PURE__ */ r((t) => t === void 0 ?
"!undefined" : t === null ? "!null" : typeof t == "string" ? _a.test(t) ? `!hex(${t.slice(1)})` : Ta.test(t) ? `!${t.replace(/[\s%]/g, "")}` :
t : typeof t == "boolean" ? `!${t}` : Array.isArray(t) ? t.map(Xt) : (0, Je.default)(t) ? Object.entries(t).reduce(
  (e, [a, n]) => Object.assign(e, { [a]: Xt(n) }),
  {}
) : t, "encodeSpecialValues"), Ku = {
  encode: !1,
  // we handle URL encoding ourselves
  delimiter: ";",
  // we don't actually create multiple query params
  allowDots: !0,
  // encode objects using dot notation: obj.key=val
  format: "RFC1738",
  // encode spaces using the + sign
  serializeDate: /* @__PURE__ */ r((t) => `!date(${t.toISOString()})`, "serializeDate")
}, Qu = /* @__PURE__ */ r((t, e) => {
  let a = ht(t, e);
  if (!a || a === Qe)
    return "";
  let n = Object.entries(a).reduce((o, [i, l]) => Yt(i, l) ? Object.assign(o, { [i]: l }) : (Ma.once.warn(Ca.dedent`
      Omitted potentially unsafe URL args.

      More info: https://storybook.js.org/docs/react/writing-stories/args#setting-args-through-the-url
    `), o), {});
  return dt.default.stringify(Xt(n), Ku).replace(/ /g, "+").split(";").map((o) => o.replace("=", ":")).join(";");
}, "buildArgsParam"), mt = (0, yt.default)(1e3)(
  (t) => t !== void 0 ? dt.default.parse(t, { ignoreQueryPrefix: !0 }) : {}
), Ju = /* @__PURE__ */ r((t) => mt(t.search), "queryFromLocation"), Yu = /* @__PURE__ */ r((t) => dt.default.stringify(t, { addQueryPrefix: !0,
encode: !1 }), "stringifyQuery"), er = (0, yt.default)(1e3)((t, e, a = !0) => {
  if (a) {
    if (typeof e != "string")
      throw new Error("startsWith only works with string targets");
    return t && t.startsWith(e) ? { path: t } : null;
  }
  let n = typeof e == "string" && t === e, o = t && e && t.match(e);
  return n || o ? { path: t } : null;
});

// src/router/router.tsx
var Ae = Re(require("react"), 1), Za = Re(ka(), 1), be = Re(Xa(), 1);
var { document: eo } = Za.global, ol = /* @__PURE__ */ r(() => `${eo.location.pathname}?`, "getBase"), il = /* @__PURE__ */ r(() => {
  let t = be.useNavigate();
  return (0, Ae.useCallback)((e, { plain: a, ...n } = {}) => {
    if (typeof e == "string" && e.startsWith("#")) {
      eo.location.hash = e;
      return;
    }
    if (typeof e == "string") {
      let o = a ? e : `?path=${e}`;
      return t(o, n);
    }
    if (typeof e == "number")
      return t(e);
  }, []);
}, "useNavigate"), to = /* @__PURE__ */ r(({ to: t, children: e, ...a }) => /* @__PURE__ */ Ae.default.createElement(be.Link, { to: `${ol()}\
path=${t}`, ...a }, e), "Link");
to.displayName = "QueryLink";
var ar = /* @__PURE__ */ r(({ children: t }) => {
  let e = be.useLocation(), { path: a, singleStory: n } = mt(e.search), { viewMode: o, storyId: i, refId: l } = Zt(a);
  return /* @__PURE__ */ Ae.default.createElement(Ae.default.Fragment, null, t({
    path: a || "/",
    location: e,
    viewMode: o,
    storyId: i,
    refId: l,
    singleStory: n === "true"
  }));
}, "Location");
ar.displayName = "QueryLocation";
function or({
  children: t,
  path: e,
  startsWith: a = !1
}) {
  return /* @__PURE__ */ Ae.default.createElement(ar, null, ({ path: n, ...o }) => t({
    match: er(n, e, a),
    ...o
  }));
}
r(or, "Match");
or.displayName = "QueryMatch";
function ro(t) {
  let { children: e, ...a } = t;
  return a.startsWith === void 0 && (a.startsWith = !1), /* @__PURE__ */ Ae.default.createElement(or, { ...a }, ({ match: o }) => o ? e : null);
}
r(ro, "Route");
ro.displayName = "Route";
var ul = /* @__PURE__ */ r((...t) => be.BrowserRouter(...t), "LocationProvider"), ll = /* @__PURE__ */ r((...t) => be.Router(...t), "BaseLoc\
ationProvider");
